// @adaptavant/eds-core/dist/alert-8749ff07.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = require('./index-7955fb8e.cjs.prod.js');
var tw = require('./tw-9929267e.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var warningIcon = require('./warning-icon-81fd0dc0.cjs.prod.js');
var errorIcon = require('./error-icon-892410c8.cjs.prod.js');
var informationIcon = require('./information-icon-8a7c46f6.cjs.prod.js');
var checkIcon = require('./check-icon-19cfc34c.cjs.prod.js');
var track = require('./track-f65e0bbd.cjs.prod.js');
var align_child_to_text = require('./align-child-to-text-c75908d4.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');
require('react');
require('./styles-30814c9d.cjs.prod.js');

const getAlertStyles = ({
  variant
}) => ({
  root: tw.tw`
		${alertVariantStylesLookup[variant]}
		gap-2
		px-3
		py-2
		rounded-8px
		text-body-12
	`,
  icon: tw.tw`
		${alertVariantFillStylesLookup[variant]}
	`
});
const alertVariantStylesLookup = {
  caution: tw.tw`
		bg-caution-secondary 
		text-caution-secondary
	`,
  critical: tw.tw`
		bg-critical-secondary 
		text-critical-secondary
	`,
  info: tw.tw`
		bg-canvas-secondary 
		text-primary
	`,
  positive: tw.tw`
		bg-positive-secondary 
		text-positive-secondary
	`
};
const alertVariantFillStylesLookup = {
  caution: tw.tw`
		fill-caution
	`,
  critical: tw.tw`
		fill-critical
	`,
  info: tw.tw`
		fill-primary
	`,
  positive: tw.tw`
		fill-positive
	`
};

/**
 * Alert component displays important information to the user.
 *
 * @description It captures the user's attention without disrupting their task.
 */

function Alert({
  children,
  className = '',
  classNames = {},
  style = {},
  styles = {},
  variant = 'info',
  ...consumerProps
}) {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getAlertStyles({
      variant
    })
  });
  const Icon = variantToIconMap[variant];
  return /*#__PURE__*/jsxRuntime.jsx(track.Track, {
    ...consumerProps,
    ...styleProps('root'),
    children: children,
    railEnd: undefined,
    railStart: /*#__PURE__*/jsxRuntime.jsx(align_child_to_text.AlignChildToText, {
      children: /*#__PURE__*/jsxRuntime.jsx(Icon, {
        ...styleProps('icon'),
        size: "16"
      })
    }),
    role: "alert",
    verticalAlign: "top"
  });
}
const variantToIconMap = {
  caution: warningIcon.WarningIcon,
  critical: errorIcon.ErrorIcon,
  info: informationIcon.InformationIcon,
  positive: checkIcon.CheckIcon
};

exports.Alert = Alert;
