// @adaptavant/eds-core/dist/align-child-to-text-b4992541.cjs.dev.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var index = require('./index-3bdbdb42.cjs.dev.js');
var zeroWidthSpace = require('./zero-width-space-a1290647.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-88b49271.cjs.dev.js');
require('./tw-merge-70bc959c.cjs.dev.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js');

/**
 * AlignChildToText
 *
 * @description a component that uses a zero-width space to vertically align a
 * child element such as an icon with a line of text. This is especially useful
 * when text needs to wrap.
 *
 * @example
 * <Text>
 * 	<AlignChildToText>
 * 		<MapPinIcon />
 * 	</AlignChildToText>
 * 	{address}
 * </Text>
 */
const AlignChildToText = /*#__PURE__*/React.forwardRef(function AlignChildToText({
  as = 'span',
  children,
  className = '',
  classNames = {},
  style = {},
  styles = {},
  ...consumerProps
}, forwardedRef) {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: {
      root: `
				inline-flex
				items-center
			`,
      child: `
				flex
				h-0
				items-center
			`
    }
  });
  return /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
    ...consumerProps,
    as: as,
    ref: forwardedRef,
    ...styleProps('root'),
    children: [/*#__PURE__*/jsxRuntime.jsx(zeroWidthSpace.ZeroWidthSpace, {}), /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      as: "span",
      ...styleProps('child'),
      children: children
    })]
  });
});
AlignChildToText.displayName = 'AlignChildToText';

exports.AlignChildToText = AlignChildToText;
