// @adaptavant/eds-core/dist/align-child-to-text-c6521616.esm.js
'use client';
import { forwardRef } from 'react';
import { g as getStyleProps } from './index-dd10a77e.esm.js';
import { Z as ZeroWidthSpace } from './zero-width-space-1914c889.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import { B as Box } from './box-a5dda214.esm.js';
import './tw-merge-91a8b6ee.esm.js';
import 'tailwind-merge';
import '../tokens/dist/adaptavant-eds-core-tokens.esm.js';

/**
 * AlignChildToText
 *
 * @description a component that uses a zero-width space to vertically align a
 * child element such as an icon with a line of text. This is especially useful
 * when text needs to wrap.
 *
 * @example
 * <Text>
 * 	<AlignChildToText>
 * 		<MapPinIcon />
 * 	</AlignChildToText>
 * 	{address}
 * </Text>
 */
const AlignChildToText = /*#__PURE__*/forwardRef(function AlignChildToText({
  as = 'span',
  children,
  className = '',
  classNames = {},
  style = {},
  styles = {},
  ...consumerProps
}, forwardedRef) {
  const styleProps = getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: {
      root: `
				inline-flex
				items-center
			`,
      child: `
				flex
				h-0
				items-center
			`
    }
  });
  return /*#__PURE__*/jsxs(Box, {
    ...consumerProps,
    as: as,
    ref: forwardedRef,
    ...styleProps('root'),
    children: [/*#__PURE__*/jsx(ZeroWidthSpace, {}), /*#__PURE__*/jsx(Box, {
      as: "span",
      ...styleProps('child'),
      children: children
    })]
  });
});
AlignChildToText.displayName = 'AlignChildToText';

export { AlignChildToText };
