// @adaptavant/eds-core/dist/calendar-bd043357.cjs.dev.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var focus = require('@react-aria/focus');
var dateFns = require('date-fns');
var React = require('react');
var reactDayPicker = require('react-day-picker');
var chevronRightIcon = require('./chevron-right-icon-ea8bd9cb.cjs.dev.js');
var dropdownDownIcon = require('./dropdown-down-icon-e861b656.cjs.dev.js');
var dropdownUpIcon = require('./dropdown-up-icon-9612d99c.cjs.dev.js');
var goTodayIcon = require('./go-today-icon-121becec.cjs.dev.js');
var index = require('./index-3bdbdb42.cjs.dev.js');
var composeId = require('./compose-id-8f0cde18.cjs.dev.js');
var styles = require('./styles-554c7d0d.cjs.dev.js');
var tw = require('./tw-cabb7e4c.cjs.dev.js');
var dateFnsTz = require('date-fns-tz');
var twMerge = require('./tw-merge-70bc959c.cjs.dev.js');
var locales = require('date-fns/locale');
var jsxRuntime = require('react/jsx-runtime');
var text = require('./text-b4d8b4ef.cjs.dev.js');
var heading = require('./heading-2f0d515d.cjs.dev.js');
var box = require('./box-88b49271.cjs.dev.js');
var iconButton = require('./icon-button-705aec14.cjs.dev.js');
var track = require('./track-1da464e0.cjs.dev.js');
var button = require('./button-4b229f71.cjs.dev.js');
var tooltip = require('./tooltip-da97c0fc.cjs.dev.js');
require('./styles-32c512c4.cjs.dev.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js');
require('./use-button-2241e53a.cjs.dev.js');
require('./styles-8337486f.cjs.dev.js');
require('./loading-902bdbe2.cjs.dev.js');
require('./grow-your-brand-icon-6c89c797.cjs.dev.js');
require('./setmore-icon-45794869.cjs.dev.js');
require('./translations-433e453a.cjs.dev.js');
require('./config-cec3759c.cjs.dev.js');

function _interopNamespace(e) {
	if (e && e.__esModule) return e;
	var n = Object.create(null);
	if (e) {
		Object.keys(e).forEach(function (k) {
			if (k !== 'default') {
				var d = Object.getOwnPropertyDescriptor(e, k);
				Object.defineProperty(n, k, d.get ? d : {
					enumerable: true,
					get: function () { return e[k]; }
				});
			}
		});
	}
	n["default"] = e;
	return Object.freeze(n);
}

var locales__namespace = /*#__PURE__*/_interopNamespace(locales);

/**
 * Uses react-day-picker hooks to provide month picker functionality.
 * @returns An object which contains the month picker state and methods.
 *
 */
function useMonthPicker() {
  const {
    locale
  } = reactDayPicker.useDayPicker();
  const focusManager = focus.useFocusManager();

  // Get the month captions for the year navigation
  const monthLabels = React.useMemo(() => {
    return Array.from({
      length: 12
    }, (_, i) => {
      const date = new Date(0, i); // Set the year to 0 as default coz the months are always going to be the same. Alternative approach is to use new Date().getFullYear().
      return {
        short: date.toLocaleString(locale.code, {
          month: 'short'
        }),
        full: date.toLocaleString(locale.code, {
          month: 'long'
        })
      };
    });
  }, [locale]);

  // KeyDown Event Handler
  const onKeyDown = e => {
    switch (e.key) {
      case 'ArrowRight':
      case 'ArrowDown':
        focusManager.focusNext({
          wrap: true
        });
        break;
      case 'ArrowLeft':
      case 'ArrowUp':
        focusManager.focusPrevious({
          wrap: true
        });
        break;
    }
  };
  return {
    monthLabels,
    onKeyDown: onKeyDown
  };
}

/**
 * A Hook to determine the render states for a single month in a calendar UI.
 * @returns An object containing the render states for the month.
 */
function useMonthRender({
  monthIndex,
  monthLabel
}) {
  // ----------- Current Month -----------------
  const {
    today,
    fromDate,
    toDate
  } = reactDayPicker.useDayPicker();
  const {
    currentMonth
  } = reactDayPicker.useNavigation();
  const currentYear = dateFns.getYear(currentMonth);
  let isCurrentMonth = false;
  if (currentYear === dateFns.getYear(today) && monthIndex === dateFns.getMonth(today)) isCurrentMonth = true;

  // ----------- Selected Month -----------------
  const {
    selectedMonth
  } = useCalendarContext();
  const renderedMonth = new Date(currentYear, monthIndex);
  let isSelected = false;
  if (renderedMonth && selectedMonth && dateFns.isSameMonth(renderedMonth, selectedMonth)) isSelected = true;

  // ----------- Disabled Month -----------------
  const fromMonth = fromDate ? dateFns.getMonth(fromDate) : undefined;
  const toMonth = toDate ? dateFns.getMonth(toDate) : undefined;
  const fromYear = fromDate ? dateFns.getYear(fromDate) : undefined;
  const toYear = toDate ? dateFns.getYear(toDate) : undefined;
  let isDisabled = false;
  if (fromYear && fromMonth) if (currentYear === fromYear && monthIndex < fromMonth) isDisabled = true;
  if (toMonth && toYear) if (currentYear === toYear && monthIndex > toMonth) isDisabled = true;

  // ----------- Selected Month: Screen Reader Announcement -----------------
  const [selectedMonthAnnouncement, setSelectedMonthAnnouncement] = React.useState('');
  React.useEffect(() => {
    if (isSelected) {
      setSelectedMonthAnnouncement(`${monthLabel.full} ${currentYear} selected`);
    }
  }, [monthLabel, currentYear, isSelected]);
  return {
    isCurrentMonth,
    isSelected,
    isDisabled,
    selectedMonthAnnouncement
  };
}

/**
 * Hook to generate or retrieve a unique ID for the calendar month picker menu.
 *
 * @returns An object containing `menuId` & `triggerId`.
 */
function useMonthPickerId() {
  const id = React.useId();
  return {
    /**
     * ID used to associate the month picker menu with the trigger button.
     */
    menuId: composeId.composeId('menu', id),
    triggerId: composeId.composeId('trigger', id)
  };
}

const getCalendarStyles = ({
  size,
  monthView
}) => ({
  root: tw.tw`
		relative
	`,
  caption: tw.tw`
		mb-2
		w-full
	`,
  captionIconEnd: tw.tw``,
  captionLabel: tw.tw`
		${calendarCaptionTextStylesLookup[size]}
		font-stronger
		inline
		text-primary
	`,
  captionWrapper: tw.tw`
		${calendarCaptionWrapperStylesMap[size]}	
	`,
  captionYear: tw.tw`
		${calendarCaptionTextStylesLookup[size]}	
		font-stronger
		inline
		text-secondary
	`,
  nav: tw.tw`
		flex
		gap-1
		items-center
	`,
  navButtonPrevious: tw.tw`
		aria-disabled:cursor-not-allowed
	`,
  navButtonNext: tw.tw`
		aria-disabled:cursor-not-allowed
	`,
  table: tw.tw`
		border-collapse
	`,
  tableBody: tw.tw`
		${size === 'standard' ? 'gap-2' : ''}
		flex
		flex-col
	`,
  head: tw.tw``,
  headRow: tw.tw`
		${rowStylesLookup[size]}
		${monthView ? 'invisible' : ''}
	`,
  headCell: tw.tw`
		${headCellStyleMap[size]}
		[border:none]
		flex
		items-center
		justify-between
		text-center
	`,
  headCellContent: tw.tw`
		text-body-12
		text-secondary
		w-full
	`,
  row: tw.tw`
		${rowStylesLookup[size]}
		${monthView ? 'invisible' : ''}
	`,
  monthGrid: tw.tw`
		${monthGridStylesLookup[size]}
		absolute
		bottom-0
		grid
		grid-cols-3
		h-[calc(100%-44px)]
		inset-x-0
		top-11 
		w-full
		z-10
	`
});
const getCalendarDayStyles = ({
  isDisabled,
  isOutside,
  isSelected,
  isToday,
  monthView,
  size
}) => ({
  cell: tw.tw`
		[border:none]
		[font-size:0px]
		w-full
	`,
  cellButton: tw.tw`
		${styles.buttonBaseStyles}
		${cellButtonStylesLookup({
    size,
    isDisabled,
    isOutside,
    isSelected,
    isToday,
    isMonthView: monthView
  })}
		[border:none]
		focus-visible:focus-ring
		font-regular
		relative
		rounded-6px
		w-full
	`,
  cellText: tw.tw`
		${cellTextStylesLookup[size]}
		${!isOutside && isToday ? 'font-strong' : ''}
		-inset-1
		absolute
		flex
		items-center
		justify-center
		capitalize
	`
});
const calendarCaptionWrapperStylesMap = {
  small: tw.tw`px-2`,
  standard: tw.tw`px-3`
};
const calendarCaptionTextStylesLookup = {
  small: tw.tw`text-body-12`,
  standard: tw.tw`text-body-16`
};
const rowStylesLookup = {
  small: tw.tw`
		flex
		gap-1
		items-center
		justify-between
	`,
  standard: tw.tw`
		flex
		gap-1.5
		items-center
		justify-between
	`
};
const headCellStyleMap = {
  small: tw.tw`h-6 w-[22px]`,
  standard: tw.tw`h-10 w-9`
};
const cellTextStylesLookup = {
  small: tw.tw`text-body-12`,
  standard: tw.tw`text-body-14`
};
const monthGridStylesLookup = {
  small: tw.tw`px-2 gap-x-4 gap-y-5 grid-rows-[repeat(4,24px)]`,
  standard: tw.tw`px-3.5 gap-x-5 gap-y-7 grid-rows-[repeat(4,36px)]`
};
const cellButtonStylesLookup = ({
  size,
  isDisabled,
  isSelected,
  isToday,
  isOutside
}) => {
  const commonStylesLookup = {
    small: tw.tw`h-6 my-0.5 w-6`,
    standard: tw.tw`h-9 my-0.5 w-9`
  };
  function conditionalStylesLookup() {
    if (isDisabled) {
      return tw.tw`
				${isToday ? 'bg-neutral-disabled' : ''}
				aria-disabled:cursor-not-allowed
				text-disabled
			`;
    }
    if (isToday) {
      return tw.tw`
				active:[box-shadow:0px_0px_0px_1.5px_var(--border-primary)]
				bg-inverse
				hover:[box-shadow:0px_0px_0px_1.5px_var(--border-hover)]
				hover:bg-inverse-hover
				text-onPrimary
			`;
    }
    if (isSelected) {
      return tw.tw`
				[box-shadow:0px_0px_0px_1.5px_var(--border-primary)]
			`;
    }
    if (isOutside) {
      return tw.tw`
				${size !== 'small' ? 'bg-neutral-secondary' : ''}
				hover:[box-shadow:0px_0px_0px_1.5px_var(--border-hover)]
				text-secondary
			`;
    }
    return tw.tw`
			${size !== 'small' ? 'bg-neutral-secondary' : ''}
			active:[box-shadow:0px_0px_0px_1.5px_var(--border-primary)]
			hover:[box-shadow:0px_0px_0px_1.5px_var(--border-hover)]
			text-primary
		`;
  }
  return tw.tw`
		${commonStylesLookup[size]}
		${conditionalStylesLookup()}
	`;
};

/**
 * Hook to generate values for Calendar component context.
 * @returns The values for the Calendar component context: size, labels, styleProps.
 */

function useCalendar({
  className = '',
  classNames = {},
  copyLabels,
  size,
  style = {},
  styles = {},
  onVisibleYearChange,
  hasMonthPicker
}) {
  const [monthView, setMonthView] = React.useState(false);
  const [selectedMonth, setSelectedMonth] = React.useState(undefined);
  const {
    menuId,
    triggerId
  } = useMonthPickerId();

  // A11y props to associate the trigger with the month picker menu.
  const monthPickerProps = {
    triggerProps: {
      'aria-haspopup': 'grid',
      'aria-expanded': monthView ? true : false,
      'aria-controls': menuId,
      id: triggerId
    },
    menuProps: {
      id: menuId,
      'aria-labelledby': triggerId
    }
  };
  return {
    labels: copyLabels,
    styleProps: index.getStyleProps({
      className,
      classNames,
      style,
      styles,
      tw: getCalendarStyles({
        size,
        monthView
      })
    }),
    size,
    monthView,
    setMonthView,
    selectedMonth,
    setSelectedMonth,
    monthPickerProps,
    onVisibleYearChange,
    hasMonthPicker
  };
}

/** Calendar context definition */

const CalendarContext = /*#__PURE__*/React.createContext(undefined);
const CalendarProvider = CalendarContext.Provider;

/**
 * Hook to access the Calendar component context.
 * @returns The values for the Calendar component context: size, labels, styleProps.
 */
function useCalendarContext() {
  const context = React.useContext(CalendarContext);
  if (context === undefined) {
    throw new Error('useCalendarContext must be used within a CalendarStylesContext');
  }
  return context;
}

/**
 * Hook to generate values for CalendarDay component styles.
 * @returns The values for the CalendarDay: styleProps.
 */

function useCalendarDayStyles({
  className = '',
  classNames = {},
  style = {},
  styles = {},
  isDisabled,
  isOutside,
  isSelected,
  isToday,
  size,
  ...config
}) {
  const {
    monthView
  } = useCalendarContext();
  return index.getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getCalendarDayStyles({
      ...config,
      size,
      isDisabled,
      isOutside,
      isSelected,
      isToday,
      monthView
    })
  });
}

/** Calendar day context definition */

const CalendarDayStylesContext = /*#__PURE__*/React.createContext(undefined);
const CalendarDayStylesProvider = CalendarDayStylesContext.Provider;

/**
 * Hook to access the CalendarDay component styles context.
 * @returns The values for the CalendarDay component styles: styleProps.
 */
function useCalendarDayStylesContext() {
  const context = React.useContext(CalendarDayStylesContext);
  if (context === undefined) {
    throw new Error('useCalendarDayStylesContext must be used within a CalendarDayStylesContext');
  }
  return context;
}

/**
 * Adjusts a date with the given time zone offset.
 *
 * @param date - The date to adjust.
 * @param tzIdentifier - The time zone identifier to adjust the date with.
 * If not provided, the system's time zone will be used.
 * @returns The adjusted date according to the provided time zone identifier.
 */
function adjustDateWithTimeZoneOffset(date, tzIdentifier) {
  const systemTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
  const utcAdjustedDate = dateFnsTz.zonedTimeToUtc(date, systemTimeZone);
  const zonedDate = dateFnsTz.utcToZonedTime(utcAdjustedDate, tzIdentifier);
  return zonedDate;
}

/**
 * Formats a date to a human-readable string.
 *
 * @param date - The date to format.
 * @param locale - The locale to format the date with.
 *
 * @returns The formatted date string.
 *
 * @example
 * ```tsx
 * humanReadableDate(new Date('2024-10-16'), 'en-US');
 * ```
 *
 * @output
 *
 * Tuesday, 16 October 2024
 *
 * */
function getHumanReadableDate(date, locale) {
  return dateFnsTz.format(date, 'EEEE, d LLLL yyyy', {
    locale
  });
}

/**
 * Merges class names and styles with multiple parameters.
 *
 * The properties from each parameter are combined, with the last parameter taking the highest precedence.
 *
 * @param styleProps - An array of objects containing `className` and `style` properties.
 *
 * @example
 * ```ts
 * const { className, style } = getMergedStyleProps(
 * 	{ className: 'class1', style: { color: 'red' } }, // Base styles
 * 	{ className: 'class2', style: { backgroundColor: 'blue' } }, // Internal custom styles
 * 	{ className: 'class3', style: { fontSize: '16px', margin: '10px' } } // Consumer styles
 * );
 * ```
 * @output
 *
 * ```ts
 * {
 * 	className: 'class1 class2 class3',
 * 	style: { color: 'red', backgroundColor: 'blue', fontSize: '16px', margin: '10px' }
 * };
 * ```
 *
 *
 * @returns
 * ```ts
 * {{ className: string; style: React.CSSProperties }}
 * ```
 */
function getMergedStyleProps(...styleProps) {
  let mergedClassName = '';
  let mergedStyle = {};
  if (styleProps.length === 0) {
    throw new Error('At least one parameter must be provided.');
  }
  for (let i = 0; i < styleProps.length; i++) {
    const props = styleProps[i];
    if (props) {
      // Merge class names using twMerge for conflict resolution
      mergedClassName = twMerge.twMerge(mergedClassName, props.className);
      // Spread styles to merge without overwriting previous ones
      mergedStyle = {
        ...mergedStyle,
        ...props.style
      };
    }
  }
  return {
    className: mergedClassName,
    style: mergedStyle
  };
}
getMergedStyleProps({
  className: 'class1',
  style: {
    color: 'red'
  }
}, {
  className: 'class1',
  style: {
    color: 'red'
  }
}, {
  className: 'class1',
  style: {
    color: 'red'
  }
});

const defaultLanguage = 'en';

/**
 * Represents the mapping of supported language codes with their corresponding `date-fns` locale objects.
 *
 * This allows `date-fns` to format dates according to different locales.
 */
const supportedDateFnsLocales = {
  ar: locales__namespace.ar,
  bg: locales__namespace.bg,
  cs: locales__namespace.cs,
  da: locales__namespace.da,
  de: locales__namespace.de,
  el: locales__namespace.el,
  en: locales__namespace.enUS,
  es: locales__namespace.es,
  et: locales__namespace.et,
  fi: locales__namespace.fi,
  fr: locales__namespace.fr,
  he: locales__namespace.he,
  hr: locales__namespace.hr,
  hu: locales__namespace.hu,
  is: locales__namespace.is,
  it: locales__namespace.it,
  ja: locales__namespace.ja,
  ko: locales__namespace.ko,
  lt: locales__namespace.lt,
  lv: locales__namespace.lv,
  nl: locales__namespace.nl,
  // Norwegian isn't supported, default to enUS
  no: locales__namespace.enUS,
  pl: locales__namespace.pl,
  pt: locales__namespace.pt,
  ro: locales__namespace.ro,
  ru: locales__namespace.ru,
  sl: locales__namespace.sl,
  sq: locales__namespace.sq,
  sr: locales__namespace.sr,
  sv: locales__namespace.sv,
  tr: locales__namespace.tr,
  uk: locales__namespace.uk,
  zh: locales__namespace.zhCN
};

/**
 * @description
 * A calendar component that allows users to select a single date.
 */

function Calendar({
  className = '',
  classNames = {},
  copyLabels = {
    disabledDayLabel: 'No availability',
    disableMonthLabel: 'No availability',
    goTodayLabel: 'Today',
    nextMonthLabel: 'Next month',
    nextYearLabel: 'Next year',
    previousMonthLabel: 'Previous month',
    previousYearLabel: 'Previous year'
  },
  isDateUnavailable,
  localeCode = defaultLanguage,
  maxValue,
  minValue,
  onValueChange,
  onVisibleMonthChange,
  size = 'standard',
  style = {},
  styles = {},
  tzModifier = Intl.DateTimeFormat().resolvedOptions().timeZone,
  value,
  visibleMonth,
  onVisibleYearChange,
  initialFocus,
  hasMonthPicker = true,
  ...consumerProps
}) {
  const {
    styleProps,
    labels,
    monthView,
    setMonthView,
    selectedMonth,
    setSelectedMonth,
    monthPickerProps
  } = useCalendar({
    className,
    classNames,
    copyLabels,
    size,
    style,
    styles,
    onVisibleYearChange,
    hasMonthPicker
  });
  let today = new Date();
  const modifiedToday = adjustDateWithTimeZoneOffset(today, tzModifier);
  if (dateFns.isValid(modifiedToday)) {
    today = modifiedToday;
  } else {
    // eslint-disable-next-line no-console
    console.error(`${tzModifier} doesn't seem to be valid. Falling back to system time.`);
  }
  const tableStyles = styleProps('table');
  const tableBodyStyles = styleProps('tableBody');
  const rowStyles = styleProps('row');

  // Map or Fallback locale object according to the localeCode
  const locale = supportedDateFnsLocales.hasOwnProperty(localeCode) ? supportedDateFnsLocales[localeCode] : supportedDateFnsLocales[defaultLanguage];
  return /*#__PURE__*/jsxRuntime.jsx(CalendarProvider, {
    value: {
      styleProps,
      labels,
      size,
      monthView,
      setMonthView,
      selectedMonth,
      setSelectedMonth,
      monthPickerProps,
      onVisibleYearChange,
      hasMonthPicker
    },
    children: /*#__PURE__*/jsxRuntime.jsx(reactDayPicker.DayPicker, {
      ...consumerProps,
      // Utilizing DayPicker's style API for elements that are not directly accessible
      classNames: {
        row: rowStyles.className,
        table: tableStyles.className,
        tbody: tableBodyStyles.className,
        ...classNames
      },
      components: {
        Caption: props => Caption({
          ...props
        }),
        Head,
        Row: props => Row({
          ...props
        })
      },
      disabled: isDateUnavailable,
      fixedWeeks: true,
      fromDate: minValue,
      initialFocus: initialFocus,
      locale: locale,
      mode: "single",
      month: visibleMonth,
      onMonthChange: onVisibleMonthChange,
      onSelect: onValueChange,
      selected: value,
      showOutsideDays: true
      // Utilizing DayPicker's style API for elements that are not directly accessible
      ,
      styles: {
        row: rowStyles.style,
        table: tableStyles.style,
        tbody: tableBodyStyles.style,
        ...styles
      },
      toDate: maxValue,
      today: today,
      ...styleProps('root')
    })
  });
}
Calendar.displayName = 'Calendar';
function Caption({
  displayMonth,
  id
}) {
  const {
    locale
  } = reactDayPicker.useDayPicker();
  const {
    hasMonthPicker
  } = useCalendarContext();
  const {
    styleProps,
    monthView,
    setMonthView,
    size,
    monthPickerProps: {
      triggerProps
    }
  } = useCalendarContext();
  const captionIconEndStyles = styleProps('captionIconEnd');
  const captionLabelStyles = styleProps('captionLabel');
  const displayMonthName = dateFns.format(displayMonth, 'MMMM', {
    locale
  });
  const formattedDisplayMonthName = displayMonthName.charAt(0).toUpperCase() + displayMonthName.slice(1);
  return /*#__PURE__*/jsxRuntime.jsx(track.Track, {
    railEnd: /*#__PURE__*/jsxRuntime.jsx(Nav, {
      displayMonth: displayMonth
    }),
    ...styleProps('caption'),
    children: hasMonthPicker ? /*#__PURE__*/jsxRuntime.jsxs(button.Button, {
      classNames: {
        iconEnd: captionIconEndStyles.className,
        label: captionLabelStyles.className
      },
      iconEnd: monthView ? dropdownUpIcon.DropdownUpIcon : dropdownDownIcon.DropdownDownIcon,
      onClick: () => {
        setMonthView(!monthView);
      },
      size: size,
      styles: {
        iconEnd: captionIconEndStyles.style,
        label: captionLabelStyles.style
      },
      variant: "neutralTertiary",
      ...styleProps('captionWrapper'),
      ...triggerProps,
      children: [formattedDisplayMonthName, ' ', /*#__PURE__*/jsxRuntime.jsx(text.Text, {
        ...styleProps('captionYear'),
        children: dateFns.format(displayMonth, 'yyyy', {
          locale
        })
      })]
    }) : /*#__PURE__*/jsxRuntime.jsxs(heading.Heading, {
      "aria-hidden": "true",
      as: "h3",
      id: id,
      ...styleProps('captionWrapper'),
      children: [/*#__PURE__*/jsxRuntime.jsxs(text.Text, {
        ...styleProps('captionLabel'),
        children: [formattedDisplayMonthName, ' ']
      }), /*#__PURE__*/jsxRuntime.jsx(text.Text, {
        ...styleProps('captionYear'),
        children: dateFns.format(displayMonth, 'yyyy', {
          locale
        })
      })]
    })
  });
}
function Nav({
  displayMonth
}) {
  const {
    styleProps,
    labels,
    size,
    monthView,
    setMonthView,
    onVisibleYearChange
  } = useCalendarContext();
  const {
    goToMonth,
    nextMonth,
    previousMonth,
    currentMonth
  } = reactDayPicker.useNavigation();
  const [shouldAnnounceCaption, setShouldAnnounceCaption] = React.useState(false);
  const {
    dir,
    locale,
    today
  } = reactDayPicker.useDayPicker();
  const formattedDate = dateFns.format(displayMonth, 'MMMM yyyy', {
    locale
  });
  const isCurrentMonth = dateFns.isSameMonth(today, currentMonth);
  const onGoTodayClick = () => {
    setShouldAnnounceCaption(true);
    goToMonth(today);
    setMonthView(false);
  };
  const onNextButtonClick = () => {
    if (nextMonth) {
      setShouldAnnounceCaption(true);
      if (monthView) {
        goToMonth(dateFns.addYears(currentMonth, 1)); // go to next year
        onVisibleYearChange === null || onVisibleYearChange === void 0 || onVisibleYearChange(dateFns.addYears(currentMonth, 1));
      } else {
        goToMonth(nextMonth);
      }
    }
  };
  const onPreviousButtonClick = () => {
    if (previousMonth) {
      setShouldAnnounceCaption(true);
      if (monthView) {
        goToMonth(dateFns.addYears(currentMonth, -1)); // go to previous year
        onVisibleYearChange === null || onVisibleYearChange === void 0 || onVisibleYearChange(dateFns.addYears(currentMonth, -1));
      } else {
        goToMonth(previousMonth);
      }
    }
  };
  return /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
    ...styleProps('nav'),
    children: [isCurrentMonth ? null : /*#__PURE__*/jsxRuntime.jsx(tooltip.Tooltip, {
      content: labels.goTodayLabel,
      placement: "bottom",
      children: ({
        triggerProps
      }) => /*#__PURE__*/jsxRuntime.jsx(iconButton.IconButton, {
        "aria-label": labels.goTodayLabel,
        icon: goTodayIcon.GoTodayIcon,
        onClick: onGoTodayClick,
        size: size,
        variant: "neutralTertiary",
        ...styleProps('navButtonPrevious'),
        ...triggerProps
      })
    }), /*#__PURE__*/jsxRuntime.jsx(tooltip.Tooltip, {
      content: monthView ? labels.previousYearLabel : labels.previousMonthLabel,
      placement: "bottom",
      children: ({
        triggerProps
      }) => /*#__PURE__*/jsxRuntime.jsx(iconButton.IconButton, {
        "aria-label": monthView ? labels.previousYearLabel : labels.previousMonthLabel,
        icon: dir === 'rtl' ? chevronRightIcon.ChevronRightIcon : chevronRightIcon.ChevronLeftIcon,
        isDisabled: !previousMonth,
        onClick: onPreviousButtonClick,
        size: size,
        variant: "neutralTertiary",
        ...styleProps('navButtonPrevious'),
        ...triggerProps
      })
    }), /*#__PURE__*/jsxRuntime.jsx(tooltip.Tooltip, {
      content: monthView ? labels.nextYearLabel : labels.nextMonthLabel,
      placement: "bottom",
      children: ({
        triggerProps
      }) => /*#__PURE__*/jsxRuntime.jsx(iconButton.IconButton, {
        "aria-label": monthView ? labels.nextYearLabel : labels.nextMonthLabel,
        icon: dir === 'rtl' ? chevronRightIcon.ChevronLeftIcon : chevronRightIcon.ChevronRightIcon,
        isDisabled: !nextMonth,
        onClick: onNextButtonClick,
        size: size,
        variant: "neutralTertiary",
        ...styleProps('navButtonNext'),
        ...triggerProps
      })
    }), /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      "aria-live": "assertive",
      className: "sr-only",
      children: shouldAnnounceCaption ? formattedDate : null
    })]
  });
}

/**
 * Generate a series of 7 days, starting from the week, to use for formatting
 * the weekday names (Monday, Tuesday, etc.).
 *
 * Not exported from react-day-picker, see original
 * @see https://github.com/gpbl/react-day-picker/blob/main/src/components/HeadRow/utils/getWeekdays.ts
 */
function getWeekdays( /** The date-fns locale object used to localize dates. */
locale, /** The index of the first day of the week (0 - Sunday). */
weekStartsOn, /** Use ISOWeek instead of locale/ */
ISOWeek) {
  const start = ISOWeek ? dateFns.startOfISOWeek(new Date()) : dateFns.startOfWeek(new Date(), {
    locale,
    weekStartsOn
  });
  const days = [];
  for (let i = 0; i < 7; i++) {
    const day = dateFns.addDays(start, i);
    days.push(day);
  }
  return days;
}

/** Render the table head. */
function Head() {
  const {
    formatters,
    ISOWeek,
    locale,
    weekStartsOn
  } = reactDayPicker.useDayPicker();
  const weekdays = getWeekdays(locale, weekStartsOn, ISOWeek);
  const {
    styleProps,
    hasMonthPicker
  } = useCalendarContext();
  return /*#__PURE__*/jsxRuntime.jsxs(React.Fragment, {
    children: [/*#__PURE__*/jsxRuntime.jsx(box.Box, {
      "aria-hidden": true,
      as: "thead",
      ...styleProps('head'),
      children: /*#__PURE__*/jsxRuntime.jsx(box.Box, {
        as: "tr",
        ...styleProps('headRow'),
        children: weekdays.map((day, index) => {
          var _formatters$formatWee;
          return /*#__PURE__*/jsxRuntime.jsx(box.Box, {
            as: "th",
            scope: "col",
            ...styleProps('headCell'),
            children: /*#__PURE__*/jsxRuntime.jsx(text.Text, {
              ...styleProps('headCellContent'),
              children: (_formatters$formatWee = formatters.formatWeekdayName(day, {
                locale
              })) === null || _formatters$formatWee === void 0 ? void 0 : _formatters$formatWee.toLocaleString().charAt(0).toUpperCase()
            })
          }, index);
        })
      })
    }), hasMonthPicker ? /*#__PURE__*/jsxRuntime.jsx(MonthsGrid, {}) : null]
  });
}

/** Render a row in the calendar. */
function Row({
  ...props
}) {
  const {
    styles,
    classNames
  } = reactDayPicker.useDayPicker();
  return /*#__PURE__*/jsxRuntime.jsx("tr", {
    className: classNames.row,
    style: styles.row,
    children: props.dates.map(date => /*#__PURE__*/jsxRuntime.jsx(Day, {
      date: date,
      displayMonth: props.displayMonth
    }, dateFns.getUnixTime(date)))
  });
}

/**
 * The content of a day cell – as a button or span element according to its
 * modifiers.
 */
function Day({
  date,
  displayMonth
}) {
  const buttonRef = React.useRef(null);
  const {
    today,
    locale
  } = reactDayPicker.useDayPicker();
  const {
    activeModifiers,
    buttonProps: {
      'aria-selected': ariaSelected,
      // @ts-expect-error: role is not in the types for buttonProps
      role: _unusedRoleProp,
      ...buttonProps
    },
    isHidden
  } = reactDayPicker.useDayRender(date, displayMonth, buttonRef);
  const {
    size,
    styleProps: consumerStyleProps
  } = useCalendarContext();
  const isCurrentDay = dateFns.isSameDay(today, date);
  const isDisabled = !!activeModifiers.disabled;
  const styleProps = useCalendarDayStyles({
    isDisabled: isDisabled,
    isMonthView: false,
    isOutside: !!activeModifiers.outside,
    isSelected: !!activeModifiers.selected,
    isToday: isCurrentDay,
    size
  });
  const {
    labels
  } = useCalendarContext();
  const isSelected = ariaSelected === true || ariaSelected === 'true';
  const humanReadableDate = getHumanReadableDate(date, locale);
  return /*#__PURE__*/jsxRuntime.jsx(CalendarDayStylesProvider, {
    value: styleProps,
    children: /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      "aria-selected": ariaSelected,
      as: "td",
      ...getMergedStyleProps(styleProps('cell'), consumerStyleProps('cell')),
      children: isHidden ? null : /*#__PURE__*/jsxRuntime.jsx(box.Box, {
        as: "span",
        children: isDisabled ? /*#__PURE__*/jsxRuntime.jsx(tooltip.Tooltip, {
          content: labels.disabledDayLabel,
          placement: "bottom",
          children: ({
            triggerProps
          }) => /*#__PURE__*/jsxRuntime.jsx(box.Box, {
            ...triggerProps,
            children: /*#__PURE__*/jsxRuntime.jsx(DayCell, {
              ...buttonProps,
              "aria-disabled": "true",
              buttonRef: buttonRef,
              humanReadableDate: humanReadableDate,
              isSelected: isSelected,
              children: buttonProps.children
            })
          })
        }) : /*#__PURE__*/jsxRuntime.jsx(DayCell, {
          ...buttonProps,
          buttonRef: buttonRef,
          humanReadableDate: humanReadableDate,
          isSelected: isSelected,
          children: buttonProps.children
        })
      })
    })
  });
}
function DayCell({
  buttonRef,
  children,
  humanReadableDate,
  isSelected,
  ...restProps
}) {
  const {
    styleProps: consumerStyleProps
  } = useCalendarContext();
  const styleProps = useCalendarDayStylesContext();
  const [announcement, setAnnouncement] = React.useState('');
  React.useEffect(() => {
    if (isSelected) {
      setAnnouncement(`Selected date: ${humanReadableDate}`);
    }
  }, [isSelected, humanReadableDate]);
  return /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
    ...restProps,
    "aria-label": `${humanReadableDate}${isSelected ? ' selected' : ''}`,
    as: "button",
    name: "day",
    ref: buttonRef,
    ...getMergedStyleProps(styleProps('cellButton'), consumerStyleProps('cellButton')),
    children: [/*#__PURE__*/jsxRuntime.jsx(text.Text, {
      ...getMergedStyleProps(styleProps('cellText'), consumerStyleProps('cellText')),
      children: children
    }), /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      "aria-live": "polite",
      as: "span",
      className: "sr-only",
      children: announcement
    })]
  });
}

/**
 *  Renders a grid of months using the styles from the calendar context
 */
function MonthsGrid() {
  const {
    styleProps,
    monthPickerProps: {
      menuProps
    },
    monthView
  } = useCalendarContext();
  const {
    monthLabels
  } = useMonthPicker();
  return /*#__PURE__*/jsxRuntime.jsx(box.Box, {
    as: "tbody",
    ...menuProps,
    children: monthView ? /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      as: "tr",
      children: /*#__PURE__*/jsxRuntime.jsx(box.Box, {
        as: "td",
        ...styleProps('monthGrid'),
        children: /*#__PURE__*/jsxRuntime.jsx(focus.FocusScope, {
          children: monthLabels.map((monthLabel, monthIndex) => {
            return /*#__PURE__*/jsxRuntime.jsx(MonthCell, {
              monthIndex: monthIndex,
              monthLabel: monthLabel
            }, monthIndex);
          })
        })
      })
    }) : null
  });
}

/**
 * Renders individual month cells within a grid, allowing navigation and selection of months.
 * Handles focus management and keyboard navigation for accessibility.
 */
function MonthCell({
  monthIndex,
  monthLabel
}) {
  const {
    size,
    labels,
    setMonthView,
    setSelectedMonth,
    styleProps: consumerStyleProps
  } = useCalendarContext();
  const {
    goToMonth,
    currentMonth
  } = reactDayPicker.useNavigation();
  const {
    onKeyDown
  } = useMonthPicker();
  const currentYear = dateFns.getYear(currentMonth);
  const {
    selectedMonthAnnouncement,
    isCurrentMonth,
    isSelected,
    isDisabled
  } = useMonthRender({
    monthLabel,
    monthIndex
  });
  const styleProps = useCalendarDayStyles({
    isDisabled,
    isMonthView: true,
    isOutside: false,
    // Months have no outside state.
    isSelected,
    isToday: isCurrentMonth,
    size
  });
  return isDisabled ? /*#__PURE__*/jsxRuntime.jsx(tooltip.Tooltip, {
    content: labels.disableMonthLabel,
    placement: "bottom",
    children: ({
      triggerProps
    }) => /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      ...getMergedStyleProps(styleProps('cell'), consumerStyleProps('cell')),
      ...triggerProps,
      children: /*#__PURE__*/jsxRuntime.jsx(box.Box, {
        "aria-disabled": "true",
        as: "button",
        disabled: true,
        name: "month",
        ...getMergedStyleProps(styleProps('cellButton'), consumerStyleProps('cellButton')),
        tabIndex: -1,
        children: /*#__PURE__*/jsxRuntime.jsx("span", {
          ...getMergedStyleProps(styleProps('cellText'), consumerStyleProps('cellText')),
          children: monthLabel.short
        })
      })
    })
  }) : /*#__PURE__*/jsxRuntime.jsx(box.Box, {
    ...getMergedStyleProps(styleProps('cell'), consumerStyleProps('cell')),
    children: /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
      "aria-label": `${monthLabel.full} ${currentYear}`,
      as: "button",
      ...getMergedStyleProps(styleProps('cellButton'), consumerStyleProps('cellButton')),
      name: "month",
      onClick: () => {
        goToMonth(new Date(currentYear, monthIndex));
        setMonthView(false);
        setSelectedMonth(new Date(currentYear, monthIndex));
      },
      onKeyDown: onKeyDown,
      tabIndex: 0,
      children: [/*#__PURE__*/jsxRuntime.jsx(box.Box, {
        "aria-hidden": "true",
        as: "span",
        ...getMergedStyleProps(styleProps('cellText'), consumerStyleProps('cellText')),
        children: monthLabel.short
      }), selectedMonthAnnouncement ? /*#__PURE__*/jsxRuntime.jsx(box.Box, {
        "aria-live": "polite",
        as: "span",
        className: "sr-only",
        children: selectedMonthAnnouncement
      }) : /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
        as: "span",
        children: [monthLabel.full, " ", currentYear]
      })]
    })
  });
}

exports.Calendar = Calendar;
exports.Day = Day;
exports.MonthCell = MonthCell;
exports.Row = Row;
exports.getWeekdays = getWeekdays;
