// @adaptavant/eds-core/dist/calendar-50f2986d.esm.js
'use client';
import { format, startOfISOWeek, startOfWeek, addDays, getUnixTime } from 'date-fns';
import { useContext, createContext, useState, useRef, useEffect } from 'react';
import { DayPicker, useNavigation, useDayPicker, useDayRender } from 'react-day-picker';
import { a as ChevronRightIcon, C as ChevronLeftIcon } from './chevron-right-icon-ae803e37.esm.js';
import { g as getStyleProps } from './index-dd10a77e.esm.js';
import { b as buttonBaseStyles } from './styles-56e113a0.esm.js';
import { t as tw } from './tw-c6a9a210.esm.js';
import { jsx, jsxs } from 'react/jsx-runtime';
import { B as Box } from './box-a5dda214.esm.js';
import { I as IconButton } from './icon-button-c905d5af.esm.js';
import { H as Heading } from './heading-8fb66234.esm.js';
import { T as Text } from './text-3fc88ad6.esm.js';
import { useTranslations } from './translations-1012abc4.esm.js';
import { Track } from './track-26dc4fe4.esm.js';
import { useBrandStyles } from './config-e49cd2d8.esm.js';
import './styles-53001919.esm.js';
import './tw-merge-91a8b6ee.esm.js';
import 'tailwind-merge';
import '../tokens/dist/adaptavant-eds-core-tokens.esm.js';
import './button-22306f71.esm.js';
import './use-button-36a8da62.esm.js';
import './styles-6927e7c9.esm.js';
import './loading-7c0b35b6.esm.js';
import './grow-your-brand-icon-b4a6e8da.esm.js';
import './setmore-icon-0a46ffb4.esm.js';

const getCalendarStyles = () => ({
  root: tw``,
  month: tw`
		flex
		flex-col
		items-stretch
		-mb-1 #1
	`,
  // #1 account for the padding on the last row of table cells
  caption: tw`
		w-full
	`,
  captionLabel: tw`
		px-2
		text-heading-16
	`,
  captionYear: tw`
		inline
		text-secondary
	`,
  nav: tw`
		flex
		gap-1
		items-center
		ps-0.5
		pe-0.5
	`,
  navButtonPrevious: tw``,
  navButtonNext: tw``,
  table: tw`
		border-collapse
	`,
  head: tw``,
  headRow: tw``,
  headCell: tw`
		border-0
		text-center
	`,
  headCellContent: tw`
		flex
		h-8
		items-center
		justify-center
		text-body-12
		text-secondary
		w-full
	`,
  row: tw``
});
const getCalendarDayStyles = ({
  isDisabled,
  isOutside,
  isSelected,
  isToday
}) => ({
  root: tw`
		border-none
	`,
  cellButtonWrapper: tw`
		flex
		items-center
		justify-center
		px-1.5
		py-2
	`,
  cellButton: tw`
		${buttonBaseStyles}
		${function () {
    if (isSelected) return tw`
				active:bg-accent-pressed
				bg-accent
				hover:bg-accent-hover
				text-onPrimary
			`;
    if (isDisabled) return tw`
				${isToday ? 'bg-neutral-disabled' : 'bg-neutral'}
				cursor-default
				high-contrast:border-none
				text-disabled
		`;
    if (isOutside) return tw`
				active:bg-neutral-pressed
				bg-neutral
				high-contrast:border-dashed
				hover:bg-neutral-hover
				hover:border-accent
				text-secondary
		`;
    return tw`
				${isToday ? 'text-accent' : 'text-primary'}
				active:bg-neutral-pressed
				bg-neutral
				high-contrast:border-dashed
				hover:bg-neutral-hover
				hover:border-accent
				hover:text-accent-hover
			`;
  }()}
		focus-visible:focus-ring
		font-regular
		h-6
		relative
		rounded-6px
		w-6
	`,
  cellText: tw`
			${isDisabled ? 'line-through' : ''}
			${isToday ? 'font-strong' : ''}
			-inset-1
			absolute
			flex
			items-center
			justify-center
			text-body-14
	`
});

function useCalendarStyles({
  className = '',
  classNames = {},
  style = {},
  styles = {}
}) {
  return getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getCalendarStyles()
  });
}
const CalendarStylesContext = /*#__PURE__*/createContext(undefined);
const CalendarStylesProvider = CalendarStylesContext.Provider;
function useCalendarStylesContext() {
  const context = useContext(CalendarStylesContext);
  if (context === undefined) {
    throw new Error('useCalendarStylesContext must be used within a CalendarStylesContext');
  }
  return context;
}

/**
 * @deprecated use `Calendar` from `@adaptavant/eds-core/experimental` instead.
 *
 * @description
 * A calendar component that allows users to select a single date.
 */

function Calendar({
  className = '',
  classNames = {},
  isDateUnavailable,
  isToday,
  locale,
  maxValue,
  minValue,
  onValueChange,
  onVisibleMonthChange,
  style = {},
  styles = {},
  value,
  visibleMonth,
  ...consumerProps
}) {
  const styleProps = useCalendarStyles({
    className,
    classNames,
    style,
    styles
  });
  const monthStyles = styleProps('month');
  const tableStyles = styleProps('table');
  const headRowStyles = styleProps('headRow');
  const headCellStyles = styleProps('headCell');
  const rowStyles = styleProps('row');
  return /*#__PURE__*/jsx(CalendarStylesProvider, {
    value: styleProps,
    children: /*#__PURE__*/jsx(DayPicker, {
      ...consumerProps,
      classNames: {
        month: monthStyles.className,
        table: tableStyles.className,
        head_row: headRowStyles.className,
        head_cell: headCellStyles.className,
        row: rowStyles.className,
        ...classNames
      },
      components: {
        Caption,
        Head,
        Row: props => Row({
          ...props,
          isToday
        })
      },
      disabled: isDateUnavailable,
      fixedWeeks: true,
      fromDate: minValue,
      locale: locale,
      mode: "single",
      month: visibleMonth,
      onMonthChange: onVisibleMonthChange,
      onSelect: onValueChange,
      selected: value,
      showOutsideDays: true,
      styles: {
        month: monthStyles.style,
        table: tableStyles.style,
        head_row: headRowStyles.style,
        head_cell: headCellStyles.style,
        row: rowStyles.style
      },
      toDate: maxValue,
      ...styleProps('root')
    })
  });
}
Calendar.displayName = 'Calendar';
function Caption({
  displayMonth,
  id
}) {
  const {
    goToMonth,
    nextMonth,
    previousMonth
  } = useNavigation();
  const {
    dir,
    locale
  } = useDayPicker();
  const [shouldAnnounceCaption, setShouldAnnounceCaption] = useState(false);
  const formattedDate = format(displayMonth, 'MMMM yyyy', {
    locale
  });
  const {
    experimental_calendar: translations
  } = useTranslations();
  const styleProps = useCalendarStylesContext();
  const nav = /*#__PURE__*/jsxs(Box, {
    ...styleProps('nav'),
    children: [/*#__PURE__*/jsx(IconButton, {
      "aria-label": translations.previousMonthLabel,
      icon: dir === 'rtl' ? ChevronRightIcon : ChevronLeftIcon,
      isDisabled: !previousMonth,
      onClick: () => {
        if (previousMonth) {
          setShouldAnnounceCaption(true);
          goToMonth(previousMonth);
        }
      },
      size: "standard",
      variant: "neutralTertiary",
      ...styleProps('navButtonPrevious')
    }), /*#__PURE__*/jsx(IconButton, {
      "aria-label": translations.nextMonthLabel,
      icon: dir === 'rtl' ? ChevronLeftIcon : ChevronRightIcon,
      isDisabled: !nextMonth,
      onClick: () => {
        if (nextMonth) {
          setShouldAnnounceCaption(true);
          goToMonth(nextMonth);
        }
      },
      size: "standard",
      variant: "neutralTertiary",
      ...styleProps('navButtonNext')
    }), /*#__PURE__*/jsx(Box, {
      "aria-live": "assertive",
      className: "sr-only",
      children: shouldAnnounceCaption ? formattedDate : null
    })]
  });
  return /*#__PURE__*/jsx(Track, {
    railEnd: nav,
    ...styleProps('caption'),
    children: /*#__PURE__*/jsxs(Heading, {
      "aria-hidden": "true",
      as: "h3",
      id: id,
      ...styleProps('captionLabel'),
      children: [format(displayMonth, 'MMMM', {
        locale
      }), ' ', /*#__PURE__*/jsx(Text, {
        ...styleProps('captionYear'),
        children: format(displayMonth, 'yyyy', {
          locale
        })
      })]
    })
  });
}

/**
 * Generate a series of 7 days, starting from the week, to use for formatting
 * the weekday names (Monday, Tuesday, etc.).
 *
 * Not exported from react-day-picker, see original
 * @see https://github.com/gpbl/react-day-picker/blob/main/src/components/HeadRow/utils/getWeekdays.ts
 */
function getWeekdays( /** The date-fns locale object used to localize dates. */
locale, /** The index of the first day of the week (0 - Sunday). */
weekStartsOn, /** Use ISOWeek instead of locale/ */
ISOWeek) {
  const start = ISOWeek ? startOfISOWeek(new Date()) : startOfWeek(new Date(), {
    locale,
    weekStartsOn
  });
  const days = [];
  for (let i = 0; i < 7; i++) {
    const day = addDays(start, i);
    days.push(day);
  }
  return days;
}

/** Render the table head. */
function Head() {
  const {
    formatters,
    ISOWeek,
    locale,
    weekStartsOn
  } = useDayPicker();
  const weekdays = getWeekdays(locale, weekStartsOn, ISOWeek);
  const styleProps = useCalendarStylesContext();
  return /*#__PURE__*/jsx(Box, {
    "aria-hidden": true,
    as: "thead",
    ...styleProps('head'),
    children: /*#__PURE__*/jsx(Box, {
      as: "tr",
      ...styleProps('headRow'),
      children: weekdays.map((day, index) => {
        var _formatters$formatWee;
        return /*#__PURE__*/jsx(Box, {
          as: "th",
          scope: "col",
          ...styleProps('headCell'),
          children: /*#__PURE__*/jsx(Text, {
            ...styleProps('headCellContent'),
            children: (_formatters$formatWee = formatters.formatWeekdayName(day, {
              locale
            })) === null || _formatters$formatWee === void 0 ? void 0 : _formatters$formatWee.toLocaleString().charAt(0)
          })
        }, index);
      })
    })
  });
}

/** Render a row in the calendar. */
function Row(props) {
  const {
    styles,
    classNames
  } = useDayPicker();
  return /*#__PURE__*/jsx("tr", {
    className: classNames.row,
    style: styles.row,
    children: props.dates.map(date => /*#__PURE__*/jsx(Day, {
      date: date,
      displayMonth: props.displayMonth,
      isToday: props.isToday
    }, getUnixTime(date)))
  });
}

/**
 * The content of a day cell – as a button or span element according to its
 * modifiers.
 */
function Day(props) {
  var _props$isToday;
  const buttonRef = useRef(null);
  const {
    activeModifiers,
    buttonProps: {
      'aria-selected': ariaSelected,
      // @ts-expect-error: role is not in the types for buttonProps
      role: _unusedRoleProp,
      ...buttonProps
    },
    isHidden
  } = useDayRender(props.date, props.displayMonth, buttonRef);
  const config = useBrandStyles();
  const styleProps = getStyleProps({
    className: '',
    classNames: {},
    style: {},
    styles: {},
    tw: getCalendarDayStyles({
      ...config,
      isDisabled: Boolean(activeModifiers.disabled),
      isOutside: Boolean(activeModifiers.outside),
      isSelected: Boolean(activeModifiers.selected),
      isToday: ((_props$isToday = props.isToday) === null || _props$isToday === void 0 ? void 0 : _props$isToday.call(props, props.date)) ?? Boolean(activeModifiers.today)
    })
  });
  const humanReadableDate = format(props.date, 'EEEE, d LLLL yyyy');
  const isSelected = ariaSelected === true || ariaSelected === 'true';
  const [announcement, setAnnouncement] = useState('');
  useEffect(() => {
    if (isSelected) {
      setAnnouncement(`Selected date: ${humanReadableDate}`);
    }
  }, [isSelected, humanReadableDate]);
  return /*#__PURE__*/jsx(Box, {
    "aria-selected": ariaSelected,
    as: "td",
    ...styleProps('root'),
    children: isHidden ? null : /*#__PURE__*/jsx(Box, {
      as: "span",
      ...styleProps('cellButtonWrapper'),
      children: /*#__PURE__*/jsxs(Box, {
        ...buttonProps,
        "aria-label": `${humanReadableDate}${isSelected ? ' selected' : ''}`,
        as: "button",
        name: "day",
        ref: buttonRef,
        ...styleProps('cellButton'),
        children: [/*#__PURE__*/jsx(Text, {
          fontWeight: activeModifiers.today ? 'strong' : undefined,
          ...styleProps('cellText'),
          children: buttonProps.children
        }), /*#__PURE__*/jsx(Box, {
          "aria-live": "polite",
          as: "span",
          className: "sr-only",
          children: announcement
        })]
      })
    })
  });
}

export { Calendar, Day, Row, getWeekdays };
