// @adaptavant/eds-core/dist/combobox-popover-1dbf686e.cjs.dev.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var utils = require('@react-aria/utils');
var React = require('react');
var context = require('./context-1e651be3.cjs.dev.js');
var index = require('./index-3bdbdb42.cjs.dev.js');
var visuallyHidden = require('./visually-hidden-9e9bf258.cjs.dev.js');
var use_click_away = require('./use-click-away-d103f9ae.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var popover = require('./popover-8c8f3261.cjs.dev.js');
var context$1 = require('./context-fb7f9049.cjs.dev.js');
require('./tw-merge-70bc959c.cjs.dev.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js');

const ComboboxPopover = /*#__PURE__*/React.forwardRef(function ComboboxPopover(props, forwardedRef) {
  const {
    popoverProps
  } = useComboboxPopover(props, forwardedRef);
  return /*#__PURE__*/jsxRuntime.jsx(popover.Popover, {
    ...popoverProps
  });
});
ComboboxPopover.displayName = 'ComboboxPopover';
function useComboboxPopover({
  children,
  className = '',
  classNames = {},
  shouldUsePortal = true,
  style = {},
  styles = {},
  ...consumerProps
}, forwardedRef) {
  const {
    closeMenu,
    isMenuOpen,
    popover,
    popoverId
  } = context.useDropdownContext();
  const {
    labelRef
  } = context$1.useFieldContext();

  // When the dropdown is opened, clicking outside should close it
  const handleClickOutside = React.useCallback(() => {
    if (isMenuOpen) closeMenu();
  }, [closeMenu, isMenuOpen]);
  const {
    popoverRef,
    referenceRef,
    getPopoverProps
  } = popover;
  const popoverProps = React.useMemo(() => getPopoverProps(), [getPopoverProps]);
  const memoizedRefs = React.useMemo(() => [labelRef, popoverRef, referenceRef], [labelRef, popoverRef, referenceRef]);
  use_click_away.useClickAway({
    refs: memoizedRefs,
    onClickAway: handleClickOutside
  });
  const styleProps = React.useMemo(() => index.getStyleProps({
    className,
    classNames,
    style,
    styles
  }), [className, classNames, style, styles]);
  return {
    popoverProps: React.useMemo(() => {
      return {
        ...consumerProps,
        children: isMenuOpen ? children : null,
        id: popoverId,
        ref: utils.mergeRefs(forwardedRef, popoverProps.ref),
        role: 'presentation',
        shouldUsePortal,
        ...styleProps('root',
        // Only apply styles when the menu is open, otherwise hide it
        isMenuOpen ? popoverProps.style : visuallyHidden.visuallyHiddenStyles)
      };
    }, [children, consumerProps, forwardedRef, isMenuOpen, popoverId, popoverProps.ref, popoverProps.style, shouldUsePortal, styleProps])
  };
}

exports.ComboboxPopover = ComboboxPopover;
exports.useComboboxPopover = useComboboxPopover;
