// @adaptavant/eds-core/dist/dialog-dd17b89f.cjs.dev.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var focus = require('@react-aria/focus');
var React = require('react');
var index = require('./index-3bdbdb42.cjs.dev.js');
var use_animate_visibility = require('./use-animate-visibility-6b555a5b.cjs.dev.js');
var styles = require('./styles-3403aaec.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-88b49271.cjs.dev.js');
var create_portal = require('./create-portal-ca12b5da.cjs.dev.js');
require('./tw-merge-70bc959c.cjs.dev.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js');
require('./tw-cabb7e4c.cjs.dev.js');

/**
 * Dialog
 *
 * @description
 * A Dialog component that prevents interaction with the rest of the application.
 * Dialog component renders its children (DialogTitle, DialogContent, DialogFooter) nodes in front of a backdrop component.
 * Offers important features:
 * - It properly manages focus; moves inside Dialog, and keeping it there until the Dialog is closed.
 * - It disables scrolling of the page content while open.
 * - It render with Portal by default to avoid Stacking issues.
 * - Manages the appropriate ARIA roles.
 * - Creates a backdrop, for disabling interaction below the Dialog.
 * - It can be closed by pressing the `Esc` key or clicking on the overlay.
 * - It can be controlled by the `open` prop.
 * - It can be delayed to animate in and out.
 *
 * Initially this is only for internal use,
 * as base for such components like Modals, Drawers or Sheets,
 * as consumer you should consider using one of the above first.
 */

function Dialog({
  children,
  className = '',
  classNames = {},
  closeOnEsc = true,
  closeOnOverlayClick = true,
  delay,
  descriptionId,
  onClose,
  onEscPress,
  onOverlayClick,
  open,
  role = 'dialog',
  style = {},
  styles: styles$1 = {},
  titleId,
  container,
  ...consumerProps
}) {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getDialogStyles()
  });
  const createPortal = create_portal.useCreatePortal();
  const dialogRef = React.useRef(null);
  const dialogWrapperRef = React.useRef(null);
  const {
    shouldRender,
    toggleVisibility
  } = use_animate_visibility.useAnimateVisibility(open, delay);
  React.useEffect(() => {
    const dialogElement = dialogRef.current;
    const dialogWrapperElement = dialogWrapperRef.current;
    const handleKeyDown = event => {
      if (!dialogElement) return;
      const expandedDropdownTriggerList = dialogElement.querySelectorAll('[aria-expanded="true"]');
      if (closeOnEsc && event.key === 'Escape' && !expandedDropdownTriggerList.length // Call esc functionality, only if there are no expanded dropdown.
      ) {
        toggleVisibility();
        if (onEscPress) onEscPress();
        onClose();
      }
    };
    const handleOverlayClick = event => {
      if (event.target === dialogWrapperElement) {
        event.preventDefault();
        toggleVisibility();
        if (onOverlayClick) onOverlayClick();
        onClose();
      }
    };
    if (closeOnEsc && dialogElement) {
      dialogElement.addEventListener('keydown', handleKeyDown);
    }
    if (closeOnOverlayClick && dialogWrapperElement) {
      dialogWrapperElement.addEventListener('mousedown', handleOverlayClick);
      dialogWrapperElement.addEventListener('touchstart', handleOverlayClick);
    }
    return () => {
      if (closeOnEsc && dialogElement) {
        dialogElement.removeEventListener('keydown', handleKeyDown);
      }
      if (closeOnOverlayClick && dialogWrapperElement) {
        dialogWrapperElement.removeEventListener('mousedown', handleOverlayClick);
        dialogWrapperElement.removeEventListener('touchstart', handleOverlayClick);
      }
    };
  }, [closeOnEsc, closeOnOverlayClick, onClose, onEscPress, onOverlayClick, toggleVisibility]);
  return shouldRender ? createPortal( /*#__PURE__*/jsxRuntime.jsx(focus.FocusScope, {
    autoFocus: true,
    contain: true,
    restoreFocus: true,
    children: /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      ...styleProps('dialogWrapper'),
      ref: dialogWrapperRef,
      children: /*#__PURE__*/jsxRuntime.jsx(box.Box, {
        ...consumerProps,
        "aria-describedby": descriptionId,
        "aria-labelledby": titleId,
        "aria-modal": "true",
        ref: dialogRef,
        role: role,
        ...styleProps('root'),
        tabIndex: 0,
        children: children
      })
    })
  }), container ?? undefined) : null;
}
Dialog.displayName = 'Dialog';

exports.Dialog = Dialog;
