// @adaptavant/eds-core/dist/dialog-header-6abacd43.cjs.dev.js
'use strict';

var styles = require('./styles-3403aaec.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-88b49271.cjs.dev.js');
var index = require('./index-3bdbdb42.cjs.dev.js');
var iconButton = require('./icon-button-705aec14.cjs.dev.js');
var removeIcon = require('./remove-icon-ad27b88f.cjs.dev.js');

/**
 * Any main content of Dialog should be wrapped with DialogContent
 * - Compound component can accommodate forms, text, images, dropdowns, much more.
 * - Comes with default overflow behaviour, only content inside ModalContent are scrollable
 * - When large content added ensures header and footer are in viewport.
 */
const DialogContent = ({
  children,
  className = '',
  classNames = {},
  style = {},
  styles: styles$1 = {},
  ...consumerProps
}) => {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getDialogContentStyles()
  });
  return /*#__PURE__*/jsxRuntime.jsx(box.Box, {
    as: "section",
    ...consumerProps,
    ...styleProps('root'),
    children: children
  });
};
DialogContent.displayName = 'DialogContent';

/**
 * Should be added as last child of a Dialog.
 * - Compound component can accommodate action buttons and other footer elements like checkbox or links.
 * - Will render as default flex container with justify-end and gap of 8px
 */
const DialogFooter = ({
  children,
  className = '',
  classNames = {},
  style = {},
  styles: styles$1 = {},
  ...consumerProps
}) => {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getDialogFooterStyles()
  });
  return /*#__PURE__*/jsxRuntime.jsx(box.Box, {
    as: "footer",
    ...consumerProps,
    ...styleProps('root'),
    children: children
  });
};
DialogFooter.displayName = 'DialogFooter';

/**
 * Should be added as first child of a Dialog.
 * - Compound component can accommodate title, subheading, Logo(if any) of Dialog
 * - Will render close button at right end when `closeButtonProps` is passed
 */
const DialogHeader = ({
  children,
  className = '',
  classNames = {},
  closeButtonProps,
  style = {},
  styles: styles$1 = {},
  ...consumerProps
}) => {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getDialogHeaderStyles()
  });
  return /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
    as: "header",
    ...consumerProps,
    ...styleProps('root'),
    children: [children, closeButtonProps ? /*#__PURE__*/jsxRuntime.jsx(iconButton.IconButton, {
      "aria-label": closeButtonProps.label,
      icon: removeIcon.RemoveIcon,
      onClick: closeButtonProps.onClick,
      size: closeButtonProps.size,
      variant: "neutralTertiary",
      ...styleProps('closeButton')
    }) : null]
  });
};
DialogHeader.displayName = 'DialogHeader';

exports.DialogContent = DialogContent;
exports.DialogFooter = DialogFooter;
exports.DialogHeader = DialogHeader;
