// @adaptavant/eds-core/dist/dropdown-menu-trigger-73b2448c.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var utils = require('@react-aria/utils');
var React = require('react');
var dropdownDownIcon = require('./dropdown-down-icon-677561cc.cjs.prod.js');
var dropdownUpIcon = require('./dropdown-up-icon-ef0561ac.cjs.prod.js');
var context = require('./context-38cb595b.cjs.prod.js');
var index = require('./index-7955fb8e.cjs.prod.js');
var mergeProps = require('./merge-props-60701ce8.cjs.prod.js');
var context$1 = require('./context-287e15a8.cjs.prod.js');
var styles = require('./styles-62d461f5.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var button = require('./button-c6c568ca.cjs.prod.js');
require('./styles-30814c9d.cjs.prod.js');
require('./tw-9929267e.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');
require('./cx-86353291.cjs.prod.js');
require('clsx');
require('./merge-ids-2d47bd24.cjs.prod.js');
require('./styles-4d7be1cb.cjs.prod.js');

const DropdownMenuTrigger = /*#__PURE__*/React.forwardRef(function DropdownMenuTrigger({
  className = '',
  classNames = {},
  iconEnd,
  size = 'standard',
  style = {},
  styles: styles$1 = {},
  variant = 'neutralSecondary',
  ...consumerProps
}, forwardedRef) {
  const context$2 = context.useDropdownContext();
  const {
    isDisabled
  } = context$1.useDropdownMenu();
  const {
    triggerProps
  } = useDropdownMenuTriggerProps(context$2, isDisabled);
  const defaultIconEnd = context$2.isMenuOpen ? dropdownUpIcon.DropdownUpIcon : dropdownDownIcon.DropdownDownIcon;
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getDropdownMenuTriggerStyles({
      size,
      variant
    })
  });
  return /*#__PURE__*/jsxRuntime.jsx(button.Button, {
    ...mergeProps.mergeProps(consumerProps, triggerProps),
    iconEnd: iconEnd ? iconEnd : defaultIconEnd,
    isDisabled: isDisabled,
    ref: utils.mergeRefs(triggerProps.ref, forwardedRef),
    size: size,
    variant: variant,
    ...styleProps('root')
  });
});
DropdownMenuTrigger.displayName = 'DropdownMenuTrigger';
function useDropdownMenuTriggerProps({
  descendantNodes,
  goToFirstMenuItem,
  goToLastMenuItem,
  isMenuOpen,
  listId,
  openMenu,
  popover,
  toggleMenu,
  triggerId
}, isDisabled) {
  const [lastKeyPressed, setLastKeyPressed] = React.useState();
  React.useEffect(() => {
    if (!isMenuOpen || !descendantNodes || !lastKeyPressed) return;
    if (lastKeyPressed === 'ArrowUp') {
      goToLastMenuItem();
    } else {
      goToFirstMenuItem();
    }
    setLastKeyPressed(undefined);
  }, [descendantNodes, goToFirstMenuItem, goToLastMenuItem, isMenuOpen, lastKeyPressed]);
  const onKeyDown = React.useCallback(event => {
    if (!supportedKeys.includes(event.code)) return;
    event.preventDefault();
    setLastKeyPressed(event.code);
    openMenu();
  }, [openMenu]);
  return {
    triggerProps: {
      ...popover.getReferenceProps(),
      'aria-controls': listId,
      'aria-expanded': isMenuOpen,
      'aria-haspopup': true,
      'aria-disabled': isDisabled || undefined,
      id: triggerId,
      onClick: isDisabled ? undefined : toggleMenu,
      onKeyDown: isDisabled ? undefined : onKeyDown
    }
  };
}
const supportedKeys = ['ArrowDown', 'ArrowUp', 'Enter', 'Space'];

exports.DropdownMenuTrigger = DropdownMenuTrigger;
exports.useDropdownMenuTriggerProps = useDropdownMenuTriggerProps;
