// @adaptavant/eds-core/dist/field-error-message-deff68ec.cjs.prod.js
'use strict';

var errorIcon = require('./error-icon-892410c8.cjs.prod.js');
var index = require('./index-7955fb8e.cjs.prod.js');
var tw = require('./tw-9929267e.cjs.prod.js');
var styles = require('./styles-3786e407.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var text = require('./text-a5ab226e.cjs.prod.js');
var track = require('./track-f65e0bbd.cjs.prod.js');
var align_child_to_text = require('./align-child-to-text-c75908d4.cjs.prod.js');

const getFieldErrorMessageStyles = ({
  isDisabled,
  size
}) => ({
  root: tw.tw`
		${styles.typography[size]}
		gap-1
	`,
  errorIcon: tw.tw`
		${isDisabled ? 'text-disabled' : 'text-critical'}
	`,
  errorMessage: tw.tw`
		${styles.typography[size]}
		${isDisabled ? 'text-disabled' : 'text-critical'}
	`
});

/**
 * ErrorMessage
 *
 * @description
 * A component that displays an error message.
 */

function FieldErrorMessage({
  children,
  className = '',
  classNames = {},
  id,
  isDisabled = false,
  size = 'standard',
  style = {},
  styles = {},
  shouldAnnounceOnMount = true,
  ...consumerProps
}) {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getFieldErrorMessageStyles({
      isDisabled,
      size
    })
  });
  return /*#__PURE__*/jsxRuntime.jsx(track.Track, {
    ...consumerProps,
    railStart: /*#__PURE__*/jsxRuntime.jsx(align_child_to_text.AlignChildToText, {
      children: /*#__PURE__*/jsxRuntime.jsx(errorIcon.ErrorIcon, {
        size: "16",
        ...styleProps('errorIcon')
      })
    }),
    role: shouldAnnounceOnMount ? 'alert' : undefined,
    verticalAlign: "top",
    ...styleProps('root'),
    children: /*#__PURE__*/jsxRuntime.jsx(text.Text, {
      id: id,
      ...styleProps('errorMessage'),
      children: children
    })
  });
}
FieldErrorMessage.displayName = 'FieldErrorMessage';

exports.FieldErrorMessage = FieldErrorMessage;
