// @adaptavant/eds-core/dist/filter-menu-search-field-9700fe1a.esm.js
'use client';
import { useDropdownContext } from './context-6e374e37.esm.js';
import { g as getFilterMenuSearchFieldStyles } from './styles-56119827.esm.js';
import { jsx } from 'react/jsx-runtime';
import { useFieldContext } from './context-2a1498a6.esm.js';
import { g as getStyleProps } from './index-dd10a77e.esm.js';
import { Field } from './field-d7a4c6ed.esm.js';
import './styles-267c9284.esm.js';
import './tw-c6a9a210.esm.js';
import './tw-merge-91a8b6ee.esm.js';
import 'tailwind-merge';
import '../tokens/dist/adaptavant-eds-core-tokens.esm.js';

function FilterMenuSearchField({
  className = '',
  classNames = {},
  labelVisibility = 'hidden',
  style = {},
  styles = {},
  ...consumerProps
}) {
  const {
    comboboxId
  } = useDropdownContext();
  const {
    size
  } = useFieldContext();
  const styleProps = getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getFilterMenuSearchFieldStyles()
  });
  return /*#__PURE__*/jsx(Field, {
    ...consumerProps,
    classNames: classNames,
    controlId: comboboxId,
    labelVisibility: labelVisibility,
    size: size,
    styles: styles,
    ...styleProps('root')
  });
}
FilterMenuSearchField.displayName = 'FilterMenuSearchField';

export { FilterMenuSearchField };
