// @adaptavant/eds-core/dist/filter-menu-search-field-33699d6b.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var context = require('./context-38cb595b.cjs.prod.js');
var styles = require('./styles-d86d95e2.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var context$1 = require('./context-008fcad0.cjs.prod.js');
var index = require('./index-7955fb8e.cjs.prod.js');
var field = require('./field-8fcfe704.cjs.prod.js');
require('./styles-3786e407.cjs.prod.js');
require('./tw-9929267e.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');

function FilterMenuSearchField({
  className = '',
  classNames = {},
  labelVisibility = 'hidden',
  style = {},
  styles: styles$1 = {},
  ...consumerProps
}) {
  const {
    comboboxId
  } = context.useDropdownContext();
  const {
    size
  } = context$1.useFieldContext();
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getFilterMenuSearchFieldStyles()
  });
  return /*#__PURE__*/jsxRuntime.jsx(field.Field, {
    ...consumerProps,
    classNames: classNames,
    controlId: comboboxId,
    labelVisibility: labelVisibility,
    size: size,
    styles: styles$1,
    ...styleProps('root')
  });
}
FilterMenuSearchField.displayName = 'FilterMenuSearchField';

exports.FilterMenuSearchField = FilterMenuSearchField;
