// @adaptavant/eds-core/dist/filter-menu-search-field-357f3d67.cjs.dev.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var context = require('./context-1e651be3.cjs.dev.js');
var styles = require('./styles-590374a3.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var context$1 = require('./context-fb7f9049.cjs.dev.js');
var index = require('./index-3bdbdb42.cjs.dev.js');
var field = require('./field-56c833ca.cjs.dev.js');
require('./styles-3e350289.cjs.dev.js');
require('./tw-cabb7e4c.cjs.dev.js');
require('./tw-merge-70bc959c.cjs.dev.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js');

function FilterMenuSearchField({
  className = '',
  classNames = {},
  labelVisibility = 'hidden',
  style = {},
  styles: styles$1 = {},
  ...consumerProps
}) {
  const {
    comboboxId
  } = context.useDropdownContext();
  const {
    size
  } = context$1.useFieldContext();
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getFilterMenuSearchFieldStyles()
  });
  return /*#__PURE__*/jsxRuntime.jsx(field.Field, {
    ...consumerProps,
    classNames: classNames,
    controlId: comboboxId,
    labelVisibility: labelVisibility,
    size: size,
    styles: styles$1,
    ...styleProps('root')
  });
}
FilterMenuSearchField.displayName = 'FilterMenuSearchField';

exports.FilterMenuSearchField = FilterMenuSearchField;
