// @adaptavant/eds-core/dist/heading-2f0d515d.cjs.dev.js
'use strict';

var React = require('react');
var index = require('./index-3bdbdb42.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var text = require('./text-b4d8b4ef.cjs.dev.js');

/**
 * Heading
 *
 * @description
 * Headings are a typographic component that support SEO and help structure page
 * content using relative visual prominence.
 */

const Heading = /*#__PURE__*/React.forwardRef(function Heading({
  as,
  children,
  className = '',
  classNames = {},
  fontWeight,
  style = {},
  styles = {},
  textAlign,
  textTransform,
  tone,
  typography,
  ...consumerProps
}, forwardedRef) {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles
  });
  return /*#__PURE__*/jsxRuntime.jsx(text.Text, {
    ...consumerProps,
    as: as,
    fontWeight: fontWeight,
    ref: forwardedRef,
    textAlign: textAlign,
    textTransform: textTransform,
    tone: tone,
    typography: typography || defaultElementToTypography[as],
    ...styleProps('root'),
    children: children
  });
});
Heading.displayName = 'Heading';
const defaultElementToTypography = {
  h1: 'heading.1',
  h2: 'heading.2',
  h3: 'heading.3',
  h4: 'heading.4',
  h5: 'heading.5',
  h6: 'heading.6'
};

exports.Heading = Heading;
