// @adaptavant/eds-core/dist/heading-8fb66234.esm.js
import { forwardRef } from 'react';
import { g as getStyleProps } from './index-dd10a77e.esm.js';
import { jsx } from 'react/jsx-runtime';
import { T as Text } from './text-3fc88ad6.esm.js';

/**
 * Heading
 *
 * @description
 * Headings are a typographic component that support SEO and help structure page
 * content using relative visual prominence.
 */

const Heading = /*#__PURE__*/forwardRef(function Heading({
  as,
  children,
  className = '',
  classNames = {},
  fontWeight,
  style = {},
  styles = {},
  textAlign,
  textTransform,
  tone,
  typography,
  ...consumerProps
}, forwardedRef) {
  const styleProps = getStyleProps({
    className,
    classNames,
    style,
    styles
  });
  return /*#__PURE__*/jsx(Text, {
    ...consumerProps,
    as: as,
    fontWeight: fontWeight,
    ref: forwardedRef,
    textAlign: textAlign,
    textTransform: textTransform,
    tone: tone,
    typography: typography || defaultElementToTypography[as],
    ...styleProps('root'),
    children: children
  });
});
Heading.displayName = 'Heading';
const defaultElementToTypography = {
  h1: 'heading.1',
  h2: 'heading.2',
  h3: 'heading.3',
  h4: 'heading.4',
  h5: 'heading.5',
  h6: 'heading.6'
};

export { Heading as H };
