// @adaptavant/eds-core/dist/inline-field-211ccfc8.cjs.dev.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var context = require('./context-fb7f9049.cjs.dev.js');
var field = require('./field-56c833ca.cjs.dev.js');
var index = require('./index-3bdbdb42.cjs.dev.js');
var tw = require('./tw-cabb7e4c.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var stack = require('./stack-b8f38fa7.cjs.dev.js');
var label = require('./label-800ebb2b.cjs.dev.js');
var counter = require('./counter-adac6bc5.cjs.dev.js');
var fieldErrorMessage = require('./field-error-message-009c1fca.cjs.dev.js');
var track = require('./track-1da464e0.cjs.dev.js');
require('./tw-merge-70bc959c.cjs.dev.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js');
require('./styles-3e350289.cjs.dev.js');
require('./text-b4d8b4ef.cjs.dev.js');
require('./error-icon-e17bfc88.cjs.dev.js');
require('./styles-32c512c4.cjs.dev.js');
require('./align-child-to-text-b4992541.cjs.dev.js');

const getInlineFieldStyles = () => ({
  root: tw.tw`
		gap-0.5
		items-stretch
		[&:has(textarea)_label]:[--padding:theme(spacing.2)] #1
	`,
  // #1 when there is a textarea inside the inline field, we need to add more padding to the label so it aligns with the text in textarea.
  errorIcon: tw.tw`
	`,
  errorMessage: tw.tw`
	`,
  errorTrack: tw.tw`
	`,
  inputWrapper: tw.tw`
		items-stretch
		gap-1 #2
	`,
  // #2 to create spacing between (input or textarea) and (error message or counter) elements
  label: tw.tw`
	    font-strong
		py-[var(--padding,theme(spacing[1.5]))]
		text-secondary		
	`,
  labelWrapper: tw.tw`
	`,
  secondaryLabel: tw.tw`
	`,
  counter: tw.tw``
});

/**
 * InlineField
 *
 * @description
 * The InlineField component exposes the elements around form inputs, and an API
 * to compose them.
 */
function InlineField({
  children,
  className = '',
  classNames = {},
  errorMessage,
  id: idProp,
  isDisabled = false,
  isRequired = false,
  label: label$1,
  labelVisibility = 'visible',
  secondaryLabel,
  counter: counter$1,
  style = {},
  styles = {},
  ...consumerProps
}) {
  const {
    controlId,
    descriptionId,
    errorMessageId,
    labelId
  } = field.useFieldIds(idProp);
  const parts = index.useComponentStyleParts({
    className,
    classNames,
    style,
    styles,
    tw: getInlineFieldStyles()
  }, ['root', 'description', 'errorIcon', 'errorMessage', 'errorTrack', 'inputWrapper', 'labelWrapper', 'label', 'secondaryLabel', 'counter']);
  const labelStyleProps = {
    ...parts.label,
    classNames: {
      secondaryLabel: parts.secondaryLabel.className
    },
    styles: {
      secondaryLabel: parts.secondaryLabel.style
    }
  };
  const fieldErrorMessageStyleProps = {
    ...parts.errorTrack,
    classNames: {
      errorIcon: parts.errorIcon.className,
      errorMessage: parts.errorMessage.className
    },
    styles: {
      errorIcon: parts.errorIcon.style,
      errorMessage: parts.errorMessage.style
    }
  };
  const showCounterOrErrorMessage = Boolean(errorMessage || counter$1);
  const labelRef = React.useRef(null);
  return /*#__PURE__*/jsxRuntime.jsx(context.FieldContextProvider, {
    value: {
      controlId,
      descriptionId,
      errorMessage,
      errorMessageId,
      isDisabled,
      isInvalid: Boolean(errorMessage),
      isRequired,
      labelId,
      labelRef,
      label: label$1,
      size: 'standard',
      variant: 'subtle'
    },
    children: /*#__PURE__*/jsxRuntime.jsx(track.Track, {
      ...consumerProps,
      ...parts.root,
      railStart: /*#__PURE__*/jsxRuntime.jsx(stack.Stack, {
        ...parts.labelWrapper,
        children: /*#__PURE__*/jsxRuntime.jsx(label.Label, {
          htmlFor: controlId,
          id: labelId,
          labelVisibility: labelVisibility,
          ref: labelRef,
          secondaryLabel: secondaryLabel,
          ...labelStyleProps,
          children: label$1
        })
      }),
      children: /*#__PURE__*/jsxRuntime.jsxs(stack.Stack, {
        ...parts.inputWrapper,
        children: [children, showCounterOrErrorMessage ? /*#__PURE__*/jsxRuntime.jsx(track.Track, {
          className: "gap-2",
          railEnd: counter$1 ? /*#__PURE__*/jsxRuntime.jsx(counter.Counter, {
            ...counter$1,
            ...parts.counter
          }) : null,
          verticalAlign: "top",
          children: errorMessage ? /*#__PURE__*/jsxRuntime.jsx(fieldErrorMessage.FieldErrorMessage, {
            id: errorMessageId,
            isDisabled: isDisabled,
            ...fieldErrorMessageStyleProps,
            children: errorMessage
          }) : null
        }) : null]
      })
    })
  });
}
InlineField.displayName = 'InlineField';

exports.InlineField = InlineField;
