// @adaptavant/eds-core/dist/label-800ebb2b.cjs.dev.js
'use strict';

var React = require('react');
var index = require('./index-3bdbdb42.cjs.dev.js');
var tw = require('./tw-cabb7e4c.cjs.dev.js');
var styles = require('./styles-3e350289.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var text = require('./text-b4d8b4ef.cjs.dev.js');

const getLabelStyles = ({
  labelVisibility,
  size
}) => ({
  root: tw.tw`
		${labelVisibility === 'hidden' ? 'sr-only' : ''}
		${styles.typography[size]}
		flex
		gap-1
	`,
  secondaryLabel: tw.tw`
		${styles.typography[size]}
		break-normal
		text-secondary
	`
});

/**
 * Label
 *
 * @description
 * Used to label a form field.
 */

const Label = /*#__PURE__*/React.forwardRef(function Label({
  as = 'label',
  children,
  className = '',
  classNames = {},
  labelVisibility = 'visible',
  secondaryLabel,
  size = 'standard',
  style = {},
  styles = {},
  ...consumerProps
}, forwardedRef) {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getLabelStyles({
      labelVisibility,
      size
    })
  });
  return /*#__PURE__*/jsxRuntime.jsxs(text.Text, {
    ...consumerProps,
    as: as,
    ref: forwardedRef,
    ...styleProps('root'),
    children: [children, secondaryLabel ? /*#__PURE__*/jsxRuntime.jsx(text.Text, {
      ...styleProps('secondaryLabel'),
      children: secondaryLabel
    }) : null]
  });
});
Label.displayName = 'Label';

exports.Label = Label;
