// @adaptavant/eds-core/dist/listbox-98182f67.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var use_cached_children = require('./use-cached-children-8f8d3c8d.cjs.prod.js');
var forwardRef = require('./forward-ref-0e64c63e.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-e39057bd.cjs.prod.js');
require('react');
require('./index-7955fb8e.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');

/**
 * @private
 * Listbox
 *
 * @description
 * Used for lists from which a user can select one or more items.
 *
 * Assigns a role of 'listbox' for accessibility purposes.
 * If `options` are provided, children can be a function that we can call to
 * render the options.
 *
 * @example
 * <Listbox options={options}>
 * 	{(option) => (
 * 		<div aria-selected={isSelected} role="option">
 * 			{option.label}
 * 		</div>
 * 	)}
 * </Listbox>
 */
const Listbox = forwardRef.forwardRefWithGenericComponent(function Listbox({
  children,
  options,
  ...consumerProps
}, forwardedRef) {
  const cachedChildren = use_cached_children.useCachedChildren({
    children,
    options
  });
  return /*#__PURE__*/jsxRuntime.jsx(box.Box, {
    ...consumerProps,
    ref: forwardedRef,
    role: "listbox",
    tabIndex: -1,
    children: cachedChildren
  });
});

exports.Listbox = Listbox;
