// @adaptavant/eds-core/dist/parse-image-processing-options-6f68c68d.esm.js
const DEFAULT_ERROR_TRANSLATIONS = {
  fileSizeExceedsLimitError: 'This image file is too big. Please upload another.',
  fileTypeMismatchError: 'The file type is not supported.',
  fileUploadError: 'An error occurred while uploading the file.',
  androidFileReadError: 'Image couldn’t be uploaded. Please try again.',
  fetchIncomingImageError: 'An error occurred while fetching the image.',
  localFileProcessingError: 'An error occurred while uploading the file.'
};
const parseErrorTranslations = (errorTranslations = DEFAULT_ERROR_TRANSLATIONS) => ({
  ...DEFAULT_ERROR_TRANSLATIONS,
  ...errorTranslations
});

const DEFAULT_IMAGE_PROCESSING_OPTIONS = {
  maxWidth: 2560,
  maxHeight: 2560,
  maxFileSizeMB: 5,
  enableResize: true,
  compressionQuality: 0.6,
  enableCompression: true,
  allowedMimeTypes: ['image/png', 'image/jpeg', 'image/jpg']
};
const parseImageProcessingOptions = imageProcessingOptions => {
  const parsedOptions = imageProcessingOptions ? {
    ...DEFAULT_IMAGE_PROCESSING_OPTIONS,
    ...imageProcessingOptions
  } : DEFAULT_IMAGE_PROCESSING_OPTIONS;
  return {
    ...parsedOptions,
    allowedMimeType: parsedOptions.allowedMimeTypes.join(', ')
  };
};

export { DEFAULT_IMAGE_PROCESSING_OPTIONS as D, parseErrorTranslations as a, parseImageProcessingOptions as p };
