// @adaptavant/eds-core/dist/radio-primitive-fba710d2.esm.js
import { forwardRef } from 'react';
import { R as RadioControlIcon } from './radio-control-icon-815a20cd.esm.js';
import { g as getStyleProps } from './index-dd10a77e.esm.js';
import { useCancellableHandler } from './checkbox-primitive-7f83b97f.esm.js';
import { t as tw } from './tw-c6a9a210.esm.js';
import { t as typography } from './styles-267c9284.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import { B as Box } from './box-a5dda214.esm.js';
import { AlignChildToText } from './align-child-to-text-c6521616.esm.js';

// ------------------------------ Radio ----------------------------------------

const getRadioStyles = ({
  isDisabled,
  size
}) => ({
  root: tw`
	`,
  radioPrimitiveRoot: tw`
	`,
  radioPrimitiveInput: tw`
	`,
  radioPrimitiveControl: tw`
	`,
  radioPrimitiveIcon: tw`
	`,
  label: tw`
		${isDisabled ? 'text-disabled' : ''}
		${typography[size]}
		gap-2
		group
	`,
  descriptionTrack: tw`
		gap-2
	`,
  description: tw`
		${typography[size]}
		${isDisabled ? 'text-disabled' : 'text-secondary'}
	`,
  spacer: tw`
		${iconSizeMap[size]}
	`
});
const iconSizeMap = {
  large: tw`
		h-4
		w-4
	`,
  standard: tw`
		h-4
		w-4
	`
};

// -------------------------- RadioPrimitive -----------------------------------

const getRadioPrimitiveStyles = ({
  size
}) => ({
  root: tw`
	`,
  input: tw`
		peer
		sr-only
	`,
  control: tw`
		${disabledStyles}
		${iconSizeMap[size]}
		${typography[size]}
		[--visibility:hidden]
		[transition-timing-function:ease]
		bg-canvas
		border
		border-input
		duration-300
		group-hover:border-input-hover
		hover:border-input-hover
		inline-flex
		items-center
		justify-center
		p-0.5
		peer-checked:[--visibility:visible]
		peer-checked:bg-inverse
		peer-checked:border-input-active
		peer-checked:hover:border-input-active-hover
		peer-checked:text-inverse
		peer-focus-visible:focus-ring
		rounded-full
		text-transparent
		transition-colors
	`,
  icon: tw`
		[visibility:--visibility]
		fill-current
		h-full
		w-full
	`
});
const disabledStyles = tw`
	peer-aria-disabled:bg-neutral-disabled
	peer-aria-disabled:border-input-disabled
	peer-aria-disabled:high-contrast:border-dotted
	peer-aria-disabled:hover:border-input-disabled
	peer-aria-disabled:peer-checked:text-disabled
	peer-aria-disabled:text-transparent
`;

/**
 * Radio Primitive
 *
 * @description
 * Radio Primitive can be used in places where the entire layout (like card) is to be selected
 * If multiple choices are valid, consider using a `Checkbox` instead.
 */
const RadioPrimitive = /*#__PURE__*/forwardRef(function RadioPrimitive({
  checked,
  className = '',
  classNames = {},
  id,
  isDisabled = false,
  name,
  onBlur,
  onChange,
  onClick,
  size = 'standard',
  style = {},
  styles = {},
  ...consumerProps
}, forwardedRef) {
  const styleProps = getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getRadioPrimitiveStyles({
      isDisabled,
      size
    })
  });
  const handleOnChange = useCancellableHandler(onChange, isDisabled);
  const handleOnClick = useCancellableHandler(onClick, isDisabled);
  return /*#__PURE__*/jsxs(AlignChildToText, {
    ...styleProps('root'),
    children: [/*#__PURE__*/jsx(Box, {
      ...consumerProps,
      "aria-disabled": isDisabled || undefined,
      as: "input",
      checked: checked,
      id: id,
      name: name,
      onBlur: onBlur,
      onChange: handleOnChange,
      onClick: handleOnClick,
      ref: forwardedRef,
      type: "radio",
      ...styleProps('input')
    }), /*#__PURE__*/jsx(Box, {
      as: "span",
      ...styleProps('control'),
      children: /*#__PURE__*/jsx(RadioControlIcon, {
        ...styleProps('icon')
      })
    })]
  });
});

export { RadioPrimitive as R, getRadioStyles as g };
