// @adaptavant/eds-core/dist/select-menu-1b85377e.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var context = require('./context-38cb595b.cjs.prod.js');
var use_dropdown_context_values = require('./use-dropdown-context-values-768aae43.cjs.prod.js');
var use_populate_descendant_nodes = require('./use-populate-descendant-nodes-a360df43.cjs.prod.js');
var mobile_context = require('./mobile-context-2364ec70.cjs.prod.js');
var utils = require('./utils-72a515c0.cjs.prod.js');
var context$1 = require('./context-621bbefc.cjs.prod.js');
var select_menu_trigger = require('./select-menu-trigger-2b983932.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var use_viewport_below_or_equal = require('./use-viewport-below-or-equal-8326f73f.cjs.prod.js');

/**
 * SelectMenu
 *
 * @description
 * A dropdown select component that allows the user to select a single option
 * from a list.
 */
function SelectMenu({
  children,
  closeButtonPropsForMobile,
  mobileFriendly = true,
  popoverMatchReferenceWidth = false,
  popoverMaxHeight = 356,
  popoverMaxWidth = 400,
  popoverOffset = 4,
  popoverPlacement = 'bottom-start',
  selectedOption,
  strategy = 'absolute',
  titleForMobile,
  togglePoint = utils.MOBILE_TOGGLE_POINT
}) {
  const dropdownContext = use_dropdown_context_values.useDropdownContextValues({
    popoverMatchReferenceWidth,
    popoverMaxHeight,
    popoverMaxWidth,
    popoverOffset,
    popoverPlacement,
    strategy
  });
  const selectMenuContext = {
    selectedOption
  };
  const {
    isMenuOpen,
    dispatch,
    popover
  } = dropdownContext;
  const {
    triggerProps
  } = select_menu_trigger.useSelectMenuTriggerProps(dropdownContext);
  const isResponsiveViewport = use_viewport_below_or_equal.useIsViewportBelowOrEqual(togglePoint);
  use_populate_descendant_nodes.usePopulateDescendantNodes({
    isMenuOpen,
    children,
    dispatch,
    popoverRef: popover.popoverRef,
    selectors: '[role="option"]',
    isMobile: mobileFriendly && isResponsiveViewport
  });
  return /*#__PURE__*/jsxRuntime.jsx(context.DropdownProvider, {
    value: dropdownContext,
    children: /*#__PURE__*/jsxRuntime.jsx(mobile_context.MobileDropdownProvider, {
      value: {
        closeButtonProps: closeButtonPropsForMobile,
        togglePoint: togglePoint,
        mobileFriendly,
        title: titleForMobile
      },
      children: /*#__PURE__*/jsxRuntime.jsx(context$1.SelectMenuProvider, {
        value: selectMenuContext,
        children: typeof children === 'function' ? children({
          isMenuOpen,
          triggerProps
        }) : children
      })
    })
  });
}

exports.SelectMenu = SelectMenu;
