// @adaptavant/eds-core/dist/select-menu-popover-5397b996.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var utils = require('@react-aria/utils');
var React = require('react');
var context = require('./context-38cb595b.cjs.prod.js');
var mobile_context = require('./mobile-context-2364ec70.cjs.prod.js');
var index = require('./index-7955fb8e.cjs.prod.js');
var visuallyHidden = require('./visually-hidden-dbb08e35.cjs.prod.js');
var use_click_away = require('./use-click-away-3152695d.cjs.prod.js');
var use_viewport_below_or_equal = require('./use-viewport-below-or-equal-8326f73f.cjs.prod.js');
var styles = require('./styles-afa010e9.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-e39057bd.cjs.prod.js');
var heading = require('./heading-6ba7ccd6.cjs.prod.js');
var context$1 = require('./context-008fcad0.cjs.prod.js');
var modal = require('./modal-861893ff.cjs.prod.js');
var popover = require('./popover-f1e8b1ac.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');
require('./styles-3786e407.cjs.prod.js');
require('./tw-9929267e.cjs.prod.js');
require('./styles-d86d95e2.cjs.prod.js');
require('./text-a5ab226e.cjs.prod.js');

const SelectMenuPopover = /*#__PURE__*/React.forwardRef(function SelectMenuPopover({
  children,
  className = '',
  classNames = {},
  shouldUsePortal = true,
  style = {},
  styles: styles$1 = {},
  ...consumerProps
}, forwardedRef) {
  const {
    closeMenu,
    isMenuOpen,
    popover: popover$1,
    popoverId
  } = context.useDropdownContext();
  const {
    closeButtonProps,
    mobileFriendly = true,
    title,
    togglePoint
  } = mobile_context.useMobileDropdownContext();
  const {
    labelRef,
    label
  } = context$1.useFieldContext();
  const isResponsiveViewport = use_viewport_below_or_equal.useIsViewportBelowOrEqual(togglePoint);

  // When the dropdown is opened, clicking outside should close it
  const handleClickOutside = React.useCallback(() => {
    if (isMenuOpen) closeMenu();
  }, [isMenuOpen, closeMenu]);
  const {
    popoverRef,
    referenceRef,
    getPopoverProps
  } = popover$1;
  const popoverProps = React.useMemo(() => getPopoverProps(), [getPopoverProps]);
  const memoizedRefs = React.useMemo(() => [labelRef, popoverRef, referenceRef], [labelRef, popoverRef, referenceRef]);
  use_click_away.useClickAway({
    refs: memoizedRefs,
    onClickAway: handleClickOutside,
    isDisabled: mobileFriendly && isResponsiveViewport
  });
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getSelectMenuPopoverStyles()
  });
  const localRef = React.useRef(null);
  if (mobileFriendly && isResponsiveViewport) {
    return /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      ref: utils.mergeRefs(forwardedRef, popoverRef, localRef),
      children: /*#__PURE__*/jsxRuntime.jsxs(modal.Modal, {
        container: localRef.current,
        onClose: closeMenu,
        open: isMenuOpen,
        titleId: "select-header",
        ...styleProps('root'),
        classNames: {
          modalWrapper: styleProps('modalWrapper').className
        },
        styles: {
          modalWrapper: styleProps('modalWrapper').style
        },
        children: [/*#__PURE__*/jsxRuntime.jsx(modal.ModalHeader, {
          closeButtonProps: closeButtonProps ? {
            label: closeButtonProps.label ?? '',
            onClick: () => {
              var _closeButtonProps$onC;
              (_closeButtonProps$onC = closeButtonProps.onClick) === null || _closeButtonProps$onC === void 0 || _closeButtonProps$onC.call(closeButtonProps);
              closeMenu();
            }
          } : undefined,
          children: /*#__PURE__*/jsxRuntime.jsx(box.Box, {
            className: "text-center",
            id: "select-header",
            children: /*#__PURE__*/jsxRuntime.jsx(heading.Heading, {
              as: "h5",
              children: title ?? label
            })
          })
        }), /*#__PURE__*/jsxRuntime.jsx(modal.ModalContent, {
          className: "overflow-hidden flex flex-col",
          children: children
        })]
      })
    });
  }
  return /*#__PURE__*/jsxRuntime.jsx(popover.Popover, {
    ...consumerProps,
    id: popoverId,
    ref: utils.mergeRefs(forwardedRef, popoverProps.ref),
    role: "presentation",
    shouldUsePortal: shouldUsePortal,
    ...styleProps('root', isMenuOpen ? popoverProps.style : visuallyHidden.visuallyHiddenStyles),
    children: isMenuOpen ? children : null
  });
});

exports.SelectMenuPopover = SelectMenuPopover;
