// @adaptavant/eds-core/dist/share-modal-d8442e50.esm.js
import { Modal, ModalHeader, Box, Avatar, AvatarImage, CompanyIcon, Heading, ModalContent, Text, Stack, Button, ClipboardCopyIcon, MailIcon, FacebookFilledIcon, MessengerFilledIcon, WhatsappFilledIcon } from '@adaptavant/eds-core';
import { useCopyToClipboard } from '@adaptavant/eds-core/utils';
import { t as tw } from './tw-c6a9a210.esm.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import { g as getStyleProps } from './index-dd10a77e.esm.js';

/**
 * URIs for building sharing links
 * */
const WHATSAPP_SHARE_BASE_URI = 'https://web.whatsapp.com/send';
const FACEBOOK_SHARE_BASE_URI = 'https://www.facebook.com/share.php';
const MESSENGER_SHARE_BASE_URI = 'https://www.facebook.com/dialog/send';
const MESSENGER_REDIRECT_URI = 'https://www.messenger.com';

const getShareModalStyles = () => ({
  root: tw``,
  modalWrapper: tw``,
  modalHeaderContent: tw`  
        flex  
        gap-2  
    `,
  title: tw`
		line-clamp-1
    	text-heading-16 
    `,
  modalHeader: tw`
		flex
		gap-2
	`,
  modalContent: tw`
		flex 
		flex-col 
		gap-2
	`,
  content: tw`
		line-clamp-1
		text-body-14
	`,
  optionStack: tw`
		gap-2 
		grid 
		grid-cols-2
		py-2
	`
});

const openInNewWindow = link => {
  if (typeof window !== 'undefined') {
    window.open(link, '_blank');
  }
};

const ShareModal = ({
  className = '',
  classNames = {},
  closeButtonProps,
  headingImageProps,
  link,
  onClose,
  onShareOptionSelect,
  shareOptions,
  style = {},
  styles = {},
  translations,
  ...modalProps
}) => {
  const [copyToClipboard] = useCopyToClipboard();
  const onOptionSelect = async option => {
    onShareOptionSelect === null || onShareOptionSelect === void 0 || onShareOptionSelect(option);
    option.onClick();
    onClose();
  };
  const styleProps = getStyleProps({
    className,
    classNames,
    styles,
    style,
    tw: getShareModalStyles()
  });
  const generateDefaultOptions = optionLabels => {
    if (!optionLabels) {
      return [];
    }
    return [...(optionLabels.copyLabel ? [{
      id: 'shared-via-copy-to-clipBoard',
      label: optionLabels.copyLabel || '',
      icon: ClipboardCopyIcon,
      onClick: () => {
        copyToClipboard(link);
      }
    }] : []), ...(optionLabels.emailLabel ? [{
      id: 'shared-via-email',
      label: optionLabels.emailLabel,
      icon: MailIcon,
      onClick: () => {
        openInNewWindow(`mailto:?&body=${link}`);
      }
    }] : []), ...(optionLabels.facebookLabel ? [{
      id: 'shared-via-facebook',
      label: optionLabels.facebookLabel,
      icon: FacebookFilledIcon,
      onClick: () => {
        openInNewWindow(`${FACEBOOK_SHARE_BASE_URI}?u=${encodeURIComponent(link)}`);
      }
    }] : []), ...(optionLabels.messengerLabel ? [{
      id: 'shared-via-messenger',
      label: optionLabels.messengerLabel,
      icon: MessengerFilledIcon,
      onClick: () => openInNewWindow(`${MESSENGER_SHARE_BASE_URI}?app_id=1192880291854772&link=${encodeURIComponent(link)}&redirect_uri=${encodeURIComponent(MESSENGER_REDIRECT_URI)}`)
    }] : []), ...(optionLabels.whatsappLabel ? [{
      id: 'shared-via-whatsapp',
      label: optionLabels.whatsappLabel,
      icon: WhatsappFilledIcon,
      onClick: () => openInNewWindow(`${WHATSAPP_SHARE_BASE_URI}?text=${link}`)
    }] : [])];
  };
  const options = [...generateDefaultOptions(translations.optionLabels), ...(shareOptions ?? [])];
  return /*#__PURE__*/jsxs(Modal, {
    classNames: {
      modalWrapper: styleProps('modalWrapper').className
    },
    onClose: onClose,
    size: 480,
    styles: {
      modalWrapper: styleProps('modalWrapper').style
    },
    titleId: "share-modal-title",
    ...modalProps,
    ...styleProps('root'),
    children: [/*#__PURE__*/jsx(ModalHeader, {
      closeButtonProps: closeButtonProps,
      ...styleProps('modalHeader'),
      children: /*#__PURE__*/jsxs(Box, {
        ...styleProps('modalHeaderContent'),
        children: [headingImageProps ? /*#__PURE__*/jsx(Avatar, {
          className: "shrink-0",
          name: headingImageProps.label,
          size: "24",
          children: /*#__PURE__*/jsx(AvatarImage, {
            src: headingImageProps.src
          })
        }) : /*#__PURE__*/jsx(CompanyIcon, {
          size: "24"
        }), /*#__PURE__*/jsx(Heading, {
          as: "h2",
          id: "share-modal-title",
          ...styleProps('title'),
          children: translations.title
        })]
      })
    }), /*#__PURE__*/jsxs(ModalContent, {
      ...styleProps('modalContent'),
      children: [translations.content ? /*#__PURE__*/jsx(Text, {
        ...styleProps('content'),
        children: translations.content
      }) : null, Boolean(options.length) && /*#__PURE__*/jsx(Stack, {
        ...styleProps('optionStack'),
        children: options.map(option => {
          return /*#__PURE__*/jsx(Button, {
            iconStart: option.icon,
            onClick: () => {
              onOptionSelect(option);
            },
            variant: "neutralSecondary",
            ...styleProps('optionButton'),
            children: option.label
          }, option.id);
        })
      })]
    })]
  });
};

export { ShareModal as S };
