// @adaptavant/eds-core/dist/sheet-572b219b.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var dialog = require('./dialog-7dd5a841.cjs.prod.js');
var dialogHeader = require('./dialog-header-f473f372.cjs.prod.js');
var utils = require('./utils-a47fd647.cjs.prod.js');
var tw = require('./tw-9929267e.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var index = require('./index-7955fb8e.cjs.prod.js');
var modal = require('./modal-861893ff.cjs.prod.js');
require('./styles-4e86148b.cjs.prod.js');
require('./box-e39057bd.cjs.prod.js');
require('./icon-button-42ea08f2.cjs.prod.js');
require('./button-c6c568ca.cjs.prod.js');
require('./use-button-c05d68a1.cjs.prod.js');
require('./styles-3c34b522.cjs.prod.js');
require('./styles-4d7be1cb.cjs.prod.js');
require('./loading-8f0b3165.cjs.prod.js');
require('./grow-your-brand-icon-cce9ad51.cjs.prod.js');
require('./styles-30814c9d.cjs.prod.js');
require('./setmore-icon-24e17637.cjs.prod.js');
require('./translations-4a8a2311.cjs.prod.js');
require('./config-7c4ee3f2.cjs.prod.js');
require('./remove-icon-50c42d9b.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');

const getSheetStyles = ({
  placement,
  open
}) => ({
  root: tw.tw`
		${placementStylesLookup({
    placement,
    open
  })}
		${placement}-0
		fixed
	`,
  sheetWrapper: tw.tw`
		${open ? 'animate-fade-in' : 'animate-fade-out'}
	`
});
function placementStylesLookup({
  placement,
  open
}) {
  switch (placement) {
    case 'left':
      return tw.tw`
				${open ? 'animate-sheet-in-left' : 'animate-sheet-out-left'}	
				w-4/5
				rounded-l-none
				h-full
			`;
    case 'right':
      return tw.tw`
				${open ? 'animate-sheet-in-right' : 'animate-sheet-out-right'}
				w-4/5
				rounded-r-none
				h-full
			`;
    case 'top':
      return tw.tw`
				${open ? 'animate-sheet-in-top' : 'animate-sheet-out-top'}
				rounded-t-none
				inset-x-0
				max-h-[80%]
			`;
    default:
    case 'bottom':
      return tw.tw`
				${open ? 'animate-sheet-in-bottom' : 'animate-sheet-out-bottom'}
				rounded-b-none
				inset-x-0
				max-h-[80%]
			`;
  }
}

/**
 * The Sheet component is a container that acts as a supplementary view,
 * providing access to important contextual information. It can be used to
 * display additional content or functionality in mobile views replacing Modal
 * from desktop.
 */
function Sheet({
  children,
  className,
  classNames = {},
  open,
  placement = 'bottom',
  style = {},
  styles = {},
  ...consumerProps
}) {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getSheetStyles({
      placement,
      open
    })
  });
  utils.validateChildren(React.Children.toArray(children), [SheetHeader, SheetContent, SheetFooter,
  /**
   * Sheet can appear as a Modal on mobile
   * so we need to allow Modal components
   */
  modal.ModalHeader, modal.ModalContent, modal.ModalFooter]);
  return /*#__PURE__*/jsxRuntime.jsx(dialog.Dialog, {
    ...consumerProps,
    classNames: {
      dialogWrapper: styleProps('sheetWrapper').className
    },
    delay: 500,
    open: open,
    ...styleProps('root'),
    children: children
  });
}
Sheet.displayName = 'Sheet';
const SheetContent = dialogHeader.DialogContent;
SheetContent.displayName = 'SheetContent';
const SheetFooter = dialogHeader.DialogFooter;
SheetFooter.displayName = 'SheetFooter';
function SheetHeader({
  children,
  closeButtonProps,
  ...consumerProps
}) {
  const dialogCloseButtonProps = closeButtonProps ? {
    ...closeButtonProps,
    size: 'large'
  } : undefined;
  return /*#__PURE__*/jsxRuntime.jsx(dialogHeader.DialogHeader, {
    ...consumerProps,
    closeButtonProps: dialogCloseButtonProps,
    children: children
  });
}
SheetHeader.displayName = 'SheetHeader';

exports.Sheet = Sheet;
exports.SheetContent = SheetContent;
exports.SheetFooter = SheetFooter;
exports.SheetHeader = SheetHeader;
