// @adaptavant/eds-core/dist/sheet-d2ba52fa.esm.js
'use client';
import { Children } from 'react';
import { Dialog } from './dialog-8a157f47.esm.js';
import { D as DialogContent, a as DialogFooter, b as DialogHeader } from './dialog-header-3dfee2e0.esm.js';
import { v as validateChildren } from './utils-c98f0558.esm.js';
import { t as tw } from './tw-c6a9a210.esm.js';
import { jsx } from 'react/jsx-runtime';
import { g as getStyleProps } from './index-dd10a77e.esm.js';
import { ModalHeader, ModalContent, ModalFooter } from './modal-682ec106.esm.js';
import './styles-015cb1bf.esm.js';
import './box-a5dda214.esm.js';
import './icon-button-c905d5af.esm.js';
import './button-22306f71.esm.js';
import './use-button-36a8da62.esm.js';
import './styles-6927e7c9.esm.js';
import './styles-56e113a0.esm.js';
import './loading-7c0b35b6.esm.js';
import './grow-your-brand-icon-b4a6e8da.esm.js';
import './styles-53001919.esm.js';
import './setmore-icon-0a46ffb4.esm.js';
import './translations-1012abc4.esm.js';
import './config-e49cd2d8.esm.js';
import './remove-icon-2073e50f.esm.js';
import './tw-merge-91a8b6ee.esm.js';
import 'tailwind-merge';
import '../tokens/dist/adaptavant-eds-core-tokens.esm.js';

const getSheetStyles = ({
  placement,
  open
}) => ({
  root: tw`
		${placementStylesLookup({
    placement,
    open
  })}
		${placement}-0
		fixed
	`,
  sheetWrapper: tw`
		${open ? 'animate-fade-in' : 'animate-fade-out'}
	`
});
function placementStylesLookup({
  placement,
  open
}) {
  switch (placement) {
    case 'left':
      return tw`
				${open ? 'animate-sheet-in-left' : 'animate-sheet-out-left'}	
				w-4/5
				rounded-l-none
				h-full
			`;
    case 'right':
      return tw`
				${open ? 'animate-sheet-in-right' : 'animate-sheet-out-right'}
				w-4/5
				rounded-r-none
				h-full
			`;
    case 'top':
      return tw`
				${open ? 'animate-sheet-in-top' : 'animate-sheet-out-top'}
				rounded-t-none
				inset-x-0
				max-h-[80%]
			`;
    default:
    case 'bottom':
      return tw`
				${open ? 'animate-sheet-in-bottom' : 'animate-sheet-out-bottom'}
				rounded-b-none
				inset-x-0
				max-h-[80%]
			`;
  }
}

/**
 * The Sheet component is a container that acts as a supplementary view,
 * providing access to important contextual information. It can be used to
 * display additional content or functionality in mobile views replacing Modal
 * from desktop.
 */
function Sheet({
  children,
  className,
  classNames = {},
  open,
  placement = 'bottom',
  style = {},
  styles = {},
  ...consumerProps
}) {
  const styleProps = getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getSheetStyles({
      placement,
      open
    })
  });
  validateChildren(Children.toArray(children), [SheetHeader, SheetContent, SheetFooter,
  /**
   * Sheet can appear as a Modal on mobile
   * so we need to allow Modal components
   */
  ModalHeader, ModalContent, ModalFooter]);
  return /*#__PURE__*/jsx(Dialog, {
    ...consumerProps,
    classNames: {
      dialogWrapper: styleProps('sheetWrapper').className
    },
    delay: 500,
    open: open,
    ...styleProps('root'),
    children: children
  });
}
Sheet.displayName = 'Sheet';
const SheetContent = DialogContent;
SheetContent.displayName = 'SheetContent';
const SheetFooter = DialogFooter;
SheetFooter.displayName = 'SheetFooter';
function SheetHeader({
  children,
  closeButtonProps,
  ...consumerProps
}) {
  const dialogCloseButtonProps = closeButtonProps ? {
    ...closeButtonProps,
    size: 'large'
  } : undefined;
  return /*#__PURE__*/jsx(DialogHeader, {
    ...consumerProps,
    closeButtonProps: dialogCloseButtonProps,
    children: children
  });
}
SheetHeader.displayName = 'SheetHeader';

export { Sheet, SheetContent, SheetFooter, SheetHeader };
