// @adaptavant/eds-core/dist/snackbar-content-8a20d828.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = require('./index-7955fb8e.cjs.prod.js');
var styles = require('./styles-5c8079e1.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-e39057bd.cjs.prod.js');
var mergeProps = require('./merge-props-60701ce8.cjs.prod.js');
var stack = require('./stack-fa75d919.cjs.prod.js');
var text = require('./text-a5ab226e.cjs.prod.js');
var loading = require('./loading-8f0b3165.cjs.prod.js');
var errorIcon = require('./error-icon-892410c8.cjs.prod.js');
var informationIcon = require('./information-icon-8a7c46f6.cjs.prod.js');
var removeIcon = require('./remove-icon-50c42d9b.cjs.prod.js');
var checkIcon = require('./check-icon-19cfc34c.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');
require('./tw-9929267e.cjs.prod.js');
require('react');
require('@react-aria/utils');
require('./cx-86353291.cjs.prod.js');
require('clsx');
require('./merge-ids-2d47bd24.cjs.prod.js');
require('./grow-your-brand-icon-cce9ad51.cjs.prod.js');
require('./styles-30814c9d.cjs.prod.js');
require('./setmore-icon-24e17637.cjs.prod.js');

/**
 * The SnackbarContent component can used as a standalone primitive to build a user feedback system.
 *
 */
const SnackbarContent = ({
  className = '',
  classNames = {},
  style = {},
  styles: styles$1 = {},
  variant,
  onCloseButtonClick,
  title,
  description,
  actionButton,
  ...consumerProps
}) => {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getSnackbarContentStyles()
  });
  return /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
    "aria-atomic": "true",
    "aria-live": "off",
    role: "alert",
    tabIndex: 0,
    ...mergeProps.mergeProps(consumerProps, styleProps('root')),
    children: [/*#__PURE__*/jsxRuntime.jsx(AppearanceIndicator, {
      variant: variant
    }), /*#__PURE__*/jsxRuntime.jsxs(stack.Stack, {
      className: "flex-1",
      children: [/*#__PURE__*/jsxRuntime.jsx(text.Text, {
        ...styleProps('title'),
        children: title
      }), description ? /*#__PURE__*/jsxRuntime.jsx(text.Text, {
        ...styleProps('description'),
        children: description
      }) : null]
    }), actionButton ? /*#__PURE__*/jsxRuntime.jsx("button", {
      type: "button",
      ...styleProps('actionButton'),
      onClick: actionButton.onClick,
      children: actionButton.text
    }) : null, /*#__PURE__*/jsxRuntime.jsx("button", {
      "aria-label": "Close",
      onClick: onCloseButtonClick,
      type: "button",
      ...styleProps('closeButton'),
      children: /*#__PURE__*/jsxRuntime.jsx(removeIcon.RemoveIcon, {
        size: "16"
      })
    })]
  });
};
SnackbarContent.displayName = 'SnackbarContent';

/** Based on the appearance, we render a different icon */
const AppearanceIndicator = ({
  variant
}) => {
  switch (variant) {
    case 'info':
      return /*#__PURE__*/jsxRuntime.jsx(informationIcon.InformationIcon, {
        className: "fill-text-inverse shrink-0",
        size: "20"
      });
    case 'positive':
      return /*#__PURE__*/jsxRuntime.jsx(checkIcon.CheckIcon, {
        className: "fill-text-inverse-positive shrink-0",
        size: "20"
      });
    case 'critical':
      return /*#__PURE__*/jsxRuntime.jsx(errorIcon.ErrorIcon, {
        className: "fill-text-inverse-critical shrink-0",
        size: "20"
      });
    case 'loading':
      return /*#__PURE__*/jsxRuntime.jsx(loading.Loading, {
        className: "shrink-0",
        size: "20",
        tone: "inverse"
      });
    default:
      return null;
  }
};

exports.SnackbarContent = SnackbarContent;
