// @adaptavant/eds-core/dist/snackbar-616fe9a5.cjs.dev.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var index = require('./index-3bdbdb42.cjs.dev.js');
var use_timeout = require('./use-timeout-8e83e1ab.cjs.dev.js');
var snackbar_content = require('./snackbar-content-b68ac44f.cjs.dev.js');
var styles = require('./styles-3379aae1.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-88b49271.cjs.dev.js');
var mergeProps = require('./merge-props-74bd38ff.cjs.dev.js');
var create_portal = require('./create-portal-ca12b5da.cjs.dev.js');
require('./tw-merge-70bc959c.cjs.dev.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js');
require('./tw-cabb7e4c.cjs.dev.js');
require('@react-aria/utils');
require('./cx-26784cbb.cjs.dev.js');
require('clsx');
require('./merge-ids-b19ec20e.cjs.dev.js');

/** Snackbar component helps to display temporary notifications of actions, errors or other events.
 * @description This component extends the `SnackbarContent` component and has the ability to control the visibility, position & auto-hide features of the Snackbar.
 */
const Snackbar = ({
  className = '',
  classNames = {},
  styles: styles$1 = {},
  title,
  description,
  autoCloseDuration = 3000,
  variant,
  placement,
  open,
  offset,
  onClose,
  onCloseButtonClick,
  actionButton,
  ...consumerProps
}) => {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style: styles.offsetStylesLookup({
      placement: placement,
      offset: offset
    }),
    styles: styles$1,
    tw: styles.getSnackbarStyles()
  });
  const createPortal = create_portal.useCreatePortal();
  const [isOpen, setIsOpen] = React.useState(open);

  // Close the snackbar after the autoCloseDuration
  use_timeout.useTimeout(() => {
    setIsOpen(false);
    onClose();
  }, autoCloseDuration);

  // Sets the visibility based on the `open` prop
  React.useEffect(() => {
    setIsOpen(open);
  }, [open]);
  return isOpen && createPortal( /*#__PURE__*/jsxRuntime.jsx(box.Box, {
    ...mergeProps.mergeProps(consumerProps, styleProps('root')),
    children: /*#__PURE__*/jsxRuntime.jsx(snackbar_content.SnackbarContent, {
      actionButton: actionButton,
      description: description,
      onCloseButtonClick: onCloseButtonClick,
      title: title,
      variant: variant
    })
  }));
};
Snackbar.displayName = 'Snackbar';

exports.Snackbar = Snackbar;
