// @adaptavant/eds-core/dist/style-holder-1e1089fe.cjs.dev.js
'use strict';

var edsCore = require('@adaptavant/eds-core');
var React = require('react');
var lodash = require('lodash');
var tw = require('./tw-cabb7e4c.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var index = require('./index-3bdbdb42.cjs.dev.js');
var cx = require('./cx-26784cbb.cjs.dev.js');

const getColorSwatchesStyles = () => ({
  root: tw.tw`
    grid w-fit
  `
});
const swatchRingSizeStyles = {
  20: tw.tw`h-7 w-7`,
  24: tw.tw`h-8 w-8`,
  28: tw.tw`h-9 w-9`,
  32: tw.tw`h-10 w-10`,
  36: tw.tw`h-11 w-11`,
  40: tw.tw`h-12 w-12`
};
const swatchSizeStyles = {
  20: tw.tw`h-5 w-5`,
  24: tw.tw`h-6 w-6`,
  28: tw.tw`h-7 w-7`,
  32: tw.tw`h-8 w-8`,
  36: tw.tw`h-9 w-9`,
  40: tw.tw`h-10 w-10`
};
const getColorSwatchStyles = ({
  size
}) => ({
  root: tw.tw`
    cursor-pointer
  `,
  radio: tw.tw`
    sr-only
    peer/color-swatch
  `,
  swatchRing: tw.tw`
		${swatchRingSizeStyles[size]}
		border-2
		border-solid
		border-transparent
    flex
		items-center
		justify-center
    peer-aria-disabled/color-swatch:border-input-disabled
    peer-checked/color-swatch:border-input-active
    peer-checked/color-swatch:hover:border-input-active-hover
		peer-focus-visible/color-swatch:border-input-active
    peer-focus-visible/color-swatch:hover:border-input-active-hover
    peer-hover/color-swatch:border-input-hover
    rounded-full
  `,
  swatch: tw.tw`
    ${swatchSizeStyles[size]}
    rounded-full
  `
});

const getAriaLabel = (background, ariaLabelPrefix) => {
  if (!lodash.isString(ariaLabelPrefix)) {
    return background;
  }
  return `${ariaLabelPrefix.trim()} ${background}`;
};
const ColorSwatch = ({
  ariaLabelPrefix,
  background,
  className,
  classNames = {},
  onChange,
  size = '32',
  style = {},
  styles = {},
  value,
  inputName,
  ...consumerProps
}) => {
  const handleChange = React.useCallback(event => {
    if (onChange) {
      onChange(event.target.value);
    }
  }, [onChange]);
  const styleProps = index.getStyleProps({
    style,
    styles,
    className,
    classNames,
    tw: getColorSwatchStyles({
      size
    })
  });
  return /*#__PURE__*/jsxRuntime.jsxs(edsCore.Box, {
    as: "label",
    ...styleProps('root'),
    children: [/*#__PURE__*/jsxRuntime.jsx(edsCore.Box, {
      ...consumerProps,
      "aria-checked": value === background,
      "aria-label": getAriaLabel(background, ariaLabelPrefix),
      as: "input",
      checked: value === background,
      name: inputName,
      onChange: handleChange,
      type: "radio",
      value: background,
      ...styleProps('radio')
    }), /*#__PURE__*/jsxRuntime.jsx(edsCore.Box, {
      ...styleProps('swatchRing'),
      children: /*#__PURE__*/jsxRuntime.jsx(edsCore.Box, {
        ...styleProps('swatch', {
          background
        })
      })
    })]
  });
};

const ColorSwatches = ({
  value,
  rows: gridRows = 1,
  colors,
  onChange,
  inputName,
  style = {},
  className,
  styles = {},
  ariaLabelPrefix,
  size = '32',
  classNames = {},
  customLastSwatch,
  customFirstSwatch,
  ...consumerProps
}) => {
  const totalNumberOfColors = React.useMemo(() => {
    const customSwatchesCount = (customFirstSwatch ? 1 : 0) + (customLastSwatch ? 1 : 0);
    return Object.keys(colors).length + customSwatchesCount;
  }, [colors, customFirstSwatch, customLastSwatch]);
  const gridColumns = React.useMemo(() => {
    if (!totalNumberOfColors) return 0;
    const reducedGridRows = Math.max(1, gridRows);
    const numberOfGridColumns = totalNumberOfColors / reducedGridRows;
    return Math.ceil(numberOfGridColumns);
  }, [totalNumberOfColors, gridRows]);
  const styleProps = index.getStyleProps({
    style,
    styles,
    className,
    classNames,
    tw: getColorSwatchesStyles()
  });
  return /*#__PURE__*/jsxRuntime.jsxs(edsCore.Box, {
    role: "radiogroup",
    ...styleProps('root', {
      gridTemplateRows: `repeat(${gridRows}, auto)`,
      gridTemplateColumns: `repeat(${gridColumns}, auto)`
    }),
    ...consumerProps,
    children: [customFirstSwatch ? customFirstSwatch : null, colors.map(color => /*#__PURE__*/jsxRuntime.jsx(ColorSwatch, {
      ariaLabelPrefix: ariaLabelPrefix,
      background: color,
      inputName: inputName,
      onChange: onChange,
      size: size,
      value: value
    }, color)), customLastSwatch ? customLastSwatch : null]
  });
};

const LoadingStyles = tw.tw`animate-spin stroke-[3px]`;
const LoadingStyles2 = tw.tw`stroke-[2px]`;

/**
 * StyleHolder
 *
 * @description This component is only for holding styles that
 * were removed from EDS but are still being used in the consumer codebase.
 * Adding classNames here helps to tailwind to purge those styles.
 */

function StyleHolder() {
  return /*#__PURE__*/jsxRuntime.jsx("div", {
    className: cx.cx(LoadingStyles, LoadingStyles2)
  });
}

exports.ColorSwatch = ColorSwatch;
exports.ColorSwatches = ColorSwatches;
exports.StyleHolder = StyleHolder;
