// @adaptavant/eds-core/dist/styles-c330e9dd.esm.js
import { h as ariaExpandedButtonStylesLookup } from './styles-56e113a0.esm.js';
import { t as tw } from './tw-c6a9a210.esm.js';

const getDropdownMenuTriggerStyles = ({
  size,
  variant
}) => ({
  root: tw`
			${dropdownTriggerTypography[size]}
			${ariaExpandedButtonStylesLookup[variant]}
		`
});
const dropdownTriggerTypography = {
  small: tw`text-body-16 sm:text-body-12`,
  standard: tw`text-body-16 sm:text-body-12`,
  large: tw`text-body-16 sm:text-body-14`
};
const getDropdownMenuPopoverStyles = () => ({
  root: tw`
			
		`,
  modalWrapper: tw`
			z-[1220] #1
		`,
  // #1 This z-index is set to override the legacy layout of the Setmore application. This will be removed,
  // and a better solution will be implemented in the upcoming versions. Follow up ticket: https://anywhereworks.atlassian.net/browse/ADS-1471
  modalHeader: tw`
		`,
  modalContent: tw`
		`,
  modalHeading: tw`
			text-center
		`
});
const getDropdownMenuListStyles = () => ({
  root: tw`
			outline-none
			overflow-y-auto
		`
});

export { getDropdownMenuTriggerStyles as a, getDropdownMenuListStyles as b, getDropdownMenuPopoverStyles as g };
