// @adaptavant/eds-core/dist/styles-6927e7c9.esm.js
import { t as tw } from './tw-c6a9a210.esm.js';
import { b as buttonBaseStyles, a as buttonDisabledStylesLookup, e as buttonRadiusStylesLookup, g as buttonVariantStylesLookup, c as buttonHoverStylesLookup, d as buttonPressedStylesLookup } from './styles-56e113a0.esm.js';

const getIconButtonStyles = ({
  buttonRadius,
  isDisabled,
  isLoading,
  size,
  variant
}) => ({
  root: `
		${buttonBaseStyles}
		${buttonDisabledStylesLookup[variant]}
		${buttonRadiusStylesLookup[buttonRadius]}
		${buttonVariantStylesLookup[variant]}
		${iconButtonSizeStylesLookup[size]}
		${isLoading || isDisabled ? '' : `${buttonHoverStylesLookup[variant]} ${buttonPressedStylesLookup[variant]}`}
		aria-disabled:cursor-default
		relative
	`,
  icon: tw`
		${isLoading ? 'opacity-0' : ''}
		pointer-events-none #1
	`,
  // #1 Prevents Safari from displaying our :focus-visible style on .focus() https://bugs.webkit.org/show_bug.cgi?id=241408
  loadingWrapper: tw`
		absolute
		flex
		inset-0
		items-center
		justify-center
		pointer-events-none
	`,
  loading: tw`
	`
});
const iconButtonSizeStylesLookup = {
  large: tw`
		h-10
		w-10
	`,
  standard: tw`
		h-8
		w-8
	`,
  small: tw`
		h-6
		w-6
	`
};

export { getIconButtonStyles as g, iconButtonSizeStylesLookup as i };
