// @adaptavant/eds-core/dist/styles-3403aaec.cjs.dev.js
'use strict';

var tw = require('./tw-cabb7e4c.cjs.dev.js');

// ------------------------------ Dialog Wrapper ---------------------------------------

const dialogInlinePaddingClass = 'px-6';
const getDialogStyles = () => ({
  dialogWrapper: tw.tw`
		bg-modal-overlay
		fixed
		flex
		h-full w-full #1
		inset-0
		items-center
		justify-center
		p-4
	`,
  // #1 can we replace with min-h-screen, w-screen?
  root: tw.tw`
		[outline:none]
		bg-canvas
		flex
		flex-col
		gap-4
		high-contrast:outline
		max-h-full
		max-w-full
		pb-[calc(env(safe-area-inset-bottom)+24px)]
		pt-5
		rounded-8px
		shadow-30
	`
});

// ------------------------------ Dialog Header ---------------------------------------

const getDialogHeaderStyles = () => ({
  root: tw.tw`
		${dialogInlinePaddingClass}
		flex
		flex-col
		gap-2
		pe-12 #2
		relative
		shrink-0
	`,
  // #2 additional padding end to ensure dialog title always has room for dialog clos
  closeButton: tw.tw`
		absolute 
		right-3
		-top-2
	`
});

// ------------------------------ Dialog Content ---------------------------------------

const getDialogContentStyles = () => ({
  root: tw.tw`
		${dialogInlinePaddingClass}
		overflow-auto
	`
});

// ------------------------------ Dialog Footer ---------------------------------------

const getDialogFooterStyles = () => ({
  root: tw.tw`
		${dialogInlinePaddingClass}
		flex
		gap-2
		items-center
		justify-end
		pt-2
	`
});

exports.getDialogContentStyles = getDialogContentStyles;
exports.getDialogFooterStyles = getDialogFooterStyles;
exports.getDialogHeaderStyles = getDialogHeaderStyles;
exports.getDialogStyles = getDialogStyles;
