// @adaptavant/eds-core/dist/styles-56119827.esm.js
import { t as typography } from './styles-267c9284.esm.js';
import { t as tw } from './tw-c6a9a210.esm.js';

// ----------------------------- FilterMenuTrigger -----------------------------

const getFilterMenuTriggerStyles = ({
  size
}) => ({
  root: tw`
		${typography[size]}
		${inlinePaddingStylesLookup[size]}
		active:bg-neutral-secondary-pressed
		aria-disabled:bg-neutral-disabled
		aria-disabled:text-disabled-secondary
		aria-expanded:bg-canvas
		border-transparent
		gap-2
		group
		high-contrast:outline
		hover:bg-canvas
		peer
		rounded-none
	`,
  chevron: tw`
		fill-secondary
		group-aria-disabled:fill-tertiary
		me-0
	`,
  focusContainer: tw`
	`,
  label: tw`
		grow
		font-regular
		text-start
	`,
  labelTrack: tw`
		w-full
	`
});
const inlinePaddingStylesLookup = {
  standard: 'px-2',
  large: 'px-3'
};

// --------------------------- FilterMenuSearchField ---------------------------

const getFilterMenuSearchFieldStyles = () => ({
  root: tw`
			-mt-1
			-top-1
			bg-neutral
			inset-x-0
			py-1
			sticky
			z-20
		`
});

export { getFilterMenuTriggerStyles as a, getFilterMenuSearchFieldStyles as g, inlinePaddingStylesLookup as i };
