// @adaptavant/eds-core/dist/styles-c9e46eb2.esm.js
import { t as tw } from './tw-c6a9a210.esm.js';
import { t as typography } from './styles-267c9284.esm.js';

// ----------------------------- Checkbox --------------------------------------

const getCheckboxStyles = ({
  isDisabled,
  size
}) => ({
  root: tw`
	`,
  checkboxPrimitiveRoot: tw`
	`,
  checkboxPrimitiveInput: tw`
	`,
  checkboxPrimitiveControl: tw`
	`,
  checkboxPrimitiveIcon: tw`
	`,
  label: tw`
		${isDisabled ? 'text-disabled' : ''}
		${typography[size]}
		gap-2
		group
	`,
  descriptionTrack: tw`
		gap-2
	`,
  description: tw`
		${typography[size]}
		${isDisabled ? 'text-disabled' : 'text-secondary'}
	`,
  spacer: tw`
		${iconSizeMap[size]}
	`
});
const iconSizeMap = {
  large: tw`
		h-4
		w-4
	`,
  standard: tw`
		h-4
		w-4
	`
};

// ------------------------ CheckboxPrimitive ----------------------------------

const getCheckboxPrimitiveStyles = ({
  isInvalid,
  size
}) => ({
  root: tw`
	`,
  input: tw`
		peer
		sr-only
	`,
  control: tw`
		${checkedStyles}
		${indeterminateStyles}
		${disabledStyles}
		${iconSizeMap[size]}
		${isInvalid ? invalidStyles : ''}
		${typography[size]}
		[--visibility:hidden]
		[transition-timing-function:ease]
		bg-canvas
		border
		border-input
		duration-300
		group-hover:border-input-hover
		hover:border-input-hover
		inline-flex
		items-center
		justify-center
		peer-focus-visible:focus-ring
		rounded-4px
		text-transparent
		transition-colors
	`,
  icon: tw`
		[visibility:--visibility]
		fill-current
		h-full
		w-full
	`
});
const checkedStyles = tw`
	peer-checked:[--visibility:visible]
	peer-checked:bg-inverse
	peer-checked:border-input-active
	peer-checked:hover:border-input-active-hover
	peer-checked:text-inverse
`;
const indeterminateStyles = tw`
	peer-indeterminate:[--visibility:visible]
	peer-indeterminate:bg-inverse
	peer-indeterminate:border-input-active
	peer-indeterminate:hover:border-input-active-hover
	peer-indeterminate:text-inverse
`;
const disabledStyles = tw`
	peer-aria-disabled:bg-neutral-disabled
	peer-aria-disabled:border-input-disabled
	peer-aria-disabled:high-contrast:border-dotted
	peer-aria-disabled:hover:border-input-disabled
	peer-aria-disabled:peer-checked:text-disabled
	peer-aria-disabled:peer-indeterminate:text-disabled
	peer-aria-disabled:text-transparent
`;
const invalidStyles = tw`
	peer-[[aria-invalid="true"]]:border-input-critical
	peer-[[aria-invalid="true"]]:group-hover:border-input-critical-hover
	peer-[[aria-invalid="true"]]:hover:border-input-critical-hover
	peer-[[aria-invalid="true"]]:peer-checked:border-input-critical
	peer-[[aria-invalid="true"]]:peer-indeterminate:border-input-critical
	peer-[[aria-invalid="true"]]:peer-checked:hover:border-input-critical-hover
	peer-[[aria-invalid="true"]]:peer-indeterminate:hover:border-input-critical-hover
	peer-[[aria-invalid="true"]]:peer-aria-disabled:border-input-disabled
`;

export { getCheckboxPrimitiveStyles as a, getCheckboxStyles as g };
