// @adaptavant/eds-core/dist/styles-8337486f.cjs.dev.js
'use strict';

var tw = require('./tw-cabb7e4c.cjs.dev.js');
var styles = require('./styles-554c7d0d.cjs.dev.js');

const getIconButtonStyles = ({
  buttonRadius,
  isDisabled,
  isLoading,
  size,
  variant
}) => ({
  root: `
		${styles.buttonBaseStyles}
		${styles.buttonDisabledStylesLookup[variant]}
		${styles.buttonRadiusStylesLookup[buttonRadius]}
		${styles.buttonVariantStylesLookup[variant]}
		${iconButtonSizeStylesLookup[size]}
		${isLoading || isDisabled ? '' : `${styles.buttonHoverStylesLookup[variant]} ${styles.buttonPressedStylesLookup[variant]}`}
		aria-disabled:cursor-default
		relative
	`,
  icon: tw.tw`
		${isLoading ? 'opacity-0' : ''}
		pointer-events-none #1
	`,
  // #1 Prevents Safari from displaying our :focus-visible style on .focus() https://bugs.webkit.org/show_bug.cgi?id=241408
  loadingWrapper: tw.tw`
		absolute
		flex
		inset-0
		items-center
		justify-center
		pointer-events-none
	`,
  loading: tw.tw`
	`
});
const iconButtonSizeStylesLookup = {
  large: tw.tw`
		h-10
		w-10
	`,
  standard: tw.tw`
		h-8
		w-8
	`,
  small: tw.tw`
		h-6
		w-6
	`
};

exports.getIconButtonStyles = getIconButtonStyles;
exports.iconButtonSizeStylesLookup = iconButtonSizeStylesLookup;
