// @adaptavant/eds-core/dist/styles-3dd5cb2b.esm.js
import { t as typography } from './styles-267c9284.esm.js';
import { t as tw } from './tw-c6a9a210.esm.js';
import { i as inlinePaddingStylesLookup } from './styles-56119827.esm.js';

// ----------------------------- SelectMenuTrigger -----------------------------

const getSelectMenuTriggerStyles = ({
  size
}) => ({
  root: tw`
		${typography[size]}
		${inlinePaddingStylesLookup[size]}
		active:bg-neutral-secondary-pressed
		aria-disabled:bg-neutral-disabled
		aria-disabled:text-disabled-secondary
		border-transparent
		gap-2
		group
		high-contrast:outline
		hover:bg-canvas
		peer
		rounded-none
	`,
  chevron: tw`
		fill-secondary
		group-aria-disabled:fill-tertiary
		me-0
	`,
  focusContainer: tw`
	`,
  focusIndicator: tw`
		peer-aria-expanded:border-2
		peer-aria-expanded:border-input-active
	`,
  label: tw`
		grow
		font-regular
		text-start
	`,
  labelTrack: tw`
		w-full
	`
});

// ----------------------------- SelectMenuPopover -----------------------------

const getSelectMenuPopoverStyles = () => ({
  modalWrapper: tw`
	`,
  root: tw`
	`
});

// ----------------------------- SelectMenuListbox -----------------------------

const getSelectMenuListboxStyles = () => ({
  root: tw`
		overflow-y-auto
		outline-none
	`
});

export { getSelectMenuPopoverStyles as a, getSelectMenuTriggerStyles as b, getSelectMenuListboxStyles as g };
