// @adaptavant/eds-core/dist/styles-c7c13eb4.cjs.dev.js
'use strict';

var styles = require('./styles-554c7d0d.cjs.dev.js');
var tw = require('./tw-cabb7e4c.cjs.dev.js');

const getDropdownMenuTriggerStyles = ({
  size,
  variant
}) => ({
  root: tw.tw`
			${dropdownTriggerTypography[size]}
			${styles.ariaExpandedButtonStylesLookup[variant]}
		`
});
const dropdownTriggerTypography = {
  small: tw.tw`text-body-16 sm:text-body-12`,
  standard: tw.tw`text-body-16 sm:text-body-12`,
  large: tw.tw`text-body-16 sm:text-body-14`
};
const getDropdownMenuPopoverStyles = () => ({
  root: tw.tw`
			
		`,
  modalWrapper: tw.tw`
			z-[1220] #1
		`,
  // #1 This z-index is set to override the legacy layout of the Setmore application. This will be removed,
  // and a better solution will be implemented in the upcoming versions. Follow up ticket: https://anywhereworks.atlassian.net/browse/ADS-1471
  modalHeader: tw.tw`
		`,
  modalContent: tw.tw`
		`,
  modalHeading: tw.tw`
			text-center
		`
});
const getDropdownMenuListStyles = () => ({
  root: tw.tw`
			outline-none
			overflow-y-auto
		`
});

exports.getDropdownMenuListStyles = getDropdownMenuListStyles;
exports.getDropdownMenuPopoverStyles = getDropdownMenuPopoverStyles;
exports.getDropdownMenuTriggerStyles = getDropdownMenuTriggerStyles;
