// @adaptavant/eds-core/dist/styles-291d4f6c.cjs.prod.js
'use strict';

var tw = require('./tw-9929267e.cjs.prod.js');
var styles = require('./styles-3786e407.cjs.prod.js');

// -------------------------------- TextInput ----------------------------------

const getTextInputStyles = ({
  hasAdornmentEnd,
  hasAdornmentStart,
  size,
  isDisabled
}) => {
  const paddingEndStyle = getPaddingStyle({
    hasAdornment: hasAdornmentEnd,
    side: 'end',
    size
  });
  const paddingStartStyle = getPaddingStyle({
    hasAdornment: hasAdornmentStart,
    side: 'start',
    size
  });
  return {
    root: tw.tw`
			${paddingEndStyle}
			${paddingStartStyle}
			${isDisabled ? 'cursor-default' : ''}
		`,
    adornmentEnd: tw.tw`
			${styles.typography[size]}
			flex
			items-center
		`,
    adornmentStart: tw.tw`
			${styles.typography[size]}
			flex
			items-center
		`
  };
};
const inputPaddingStylesLookup = {
  start: {
    withAdornment: {
      standard: tw.tw`ps-1.5`,
      large: tw.tw`ps-2`
    },
    withoutAdornment: {
      standard: tw.tw`ps-2`,
      large: tw.tw`ps-3`
    }
  },
  end: {
    withAdornment: {
      standard: tw.tw`pe-1.5`,
      large: tw.tw`pe-2`
    },
    withoutAdornment: {
      standard: tw.tw`pe-2`,
      large: tw.tw`pe-3`
    }
  }
};
function getPaddingStyle({
  side,
  hasAdornment,
  size
}) {
  const adornmentKey = hasAdornment ? 'withAdornment' : 'withoutAdornment';
  return inputPaddingStylesLookup[side][adornmentKey][size];
}

// ---------------------------- TextInputPrimitive -----------------------------

const getTextInputPrimitiveStyles = ({
  size
}) => ({
  root: tw.tw`
		${inputHeightStylesLookup[size]}
		${inputLineHeightStylesLookup[size]}
		${styles.typography[size]}
		[outline:none]
		aria-disabled:placeholder:text-disabled
		aria-disabled:text-disabled-secondary
		bg-transparent
		flex
		placeholder:text-secondary
		text-primary
		w-full
	`
});
const inputHeightStylesLookup = {
  standard: tw.tw`h-8`,
  large: tw.tw`h-10`
};
const inputLineHeightStylesLookup = {
  standard: tw.tw`leading-8`,
  large: tw.tw`leading-10`
};

exports.getTextInputPrimitiveStyles = getTextInputPrimitiveStyles;
exports.getTextInputStyles = getTextInputStyles;
exports.inputHeightStylesLookup = inputHeightStylesLookup;
exports.inputPaddingStylesLookup = inputPaddingStylesLookup;
