// @adaptavant/eds-core/dist/styles-6ef6a787.cjs.prod.js
'use strict';

var tw = require('./tw-9929267e.cjs.prod.js');

// ------------------------------ Tabs ---------------------------------------
const getTabsStyles = () => ({
  root: tw.tw`
		w-full
	`
});

// ------------------------------ Tab List ---------------------------------------
const ScrollMaskBaseStyles = tw.tw`
	absolute
	inset-y-0
	pointer-events-none
	transition-opacity
	w-8
	z-10
`;
const tabHighlighterHeightStylesLookup = {
  small: tw.tw`h-[1.5px]`,
  standard: tw.tw`h-0.5`
};
const getTabListStyles = ({
  size
}) => ({
  root: tw.tw`
			overflow-hidden
			relative
	`,
  tabItemWrapper: tw.tw`
			flex
			gap-4
			overflow-x-auto
			px-1
			py-1.5
			relative
			scrollbar-hidden
	`,
  highlighter: tw.tw`
			${tabHighlighterHeightStylesLookup[size]}
			absolute
			bg-inverse
			bottom-0
			origin-left
			rounded-4px
			start-0
			transition-transform
	`,
  scrollMaskStart: tw.tw`
			${ScrollMaskBaseStyles}
			start-0
	`,
  scrollMaskEnd: tw.tw`
			${ScrollMaskBaseStyles}
			end-0
	`
});

// ------------------------------ Tab Item ---------------------------------------

const tabItemFontSizeLookup = {
  small: tw.tw`text-body-12`,
  standard: tw.tw`text-body-14`
};
const getTabItemStyles = ({
  size
}) => ({
  root: tw.tw`
		${tabItemFontSizeLookup[size]}
		active:text-tertiary
		aria-disabled:text-disabled
		aria-selected:text-primary
		focus-visible:focus-ring 
		font-strong 
		hover:text-secondary-hover
		outline-none 
		px-0.5
		rounded-6px
		text-secondary 
		whitespace-nowrap 
	`
});

// ------------------------------ Tab Content ---------------------------------------
const getTabContentStyles = () => ({
  root: tw.tw`
		aria-hidden:hidden
		focus-visible:focus-ring
		mt-1
	`
});

exports.getTabContentStyles = getTabContentStyles;
exports.getTabItemStyles = getTabItemStyles;
exports.getTabListStyles = getTabListStyles;
exports.getTabsStyles = getTabsStyles;
