// @adaptavant/eds-core/dist/styles-33ab6633.esm.js
import { t as tw } from './tw-c6a9a210.esm.js';

// ------------------------------ Tabs ---------------------------------------
const getTabsStyles = () => ({
  root: tw`
		w-full
	`
});

// ------------------------------ Tab List ---------------------------------------
const ScrollMaskBaseStyles = tw`
	absolute
	inset-y-0
	pointer-events-none
	transition-opacity
	w-8
	z-10
`;
const tabHighlighterHeightStylesLookup = {
  small: tw`h-[1.5px]`,
  standard: tw`h-0.5`
};
const getTabListStyles = ({
  size
}) => ({
  root: tw`
			overflow-hidden
			relative
	`,
  tabItemWrapper: tw`
			flex
			gap-4
			overflow-x-auto
			px-1
			py-1.5
			relative
			scrollbar-hidden
	`,
  highlighter: tw`
			${tabHighlighterHeightStylesLookup[size]}
			absolute
			bg-inverse
			bottom-0
			origin-left
			rounded-4px
			start-0
			transition-transform
	`,
  scrollMaskStart: tw`
			${ScrollMaskBaseStyles}
			start-0
	`,
  scrollMaskEnd: tw`
			${ScrollMaskBaseStyles}
			end-0
	`
});

// ------------------------------ Tab Item ---------------------------------------

const tabItemFontSizeLookup = {
  small: tw`text-body-12`,
  standard: tw`text-body-14`
};
const getTabItemStyles = ({
  size
}) => ({
  root: tw`
		${tabItemFontSizeLookup[size]}
		active:text-tertiary
		aria-disabled:text-disabled
		aria-selected:text-primary
		focus-visible:focus-ring 
		font-strong 
		hover:text-secondary-hover
		outline-none 
		px-0.5
		rounded-6px
		text-secondary 
		whitespace-nowrap 
	`
});

// ------------------------------ Tab Content ---------------------------------------
const getTabContentStyles = () => ({
  root: tw`
		aria-hidden:hidden
		focus-visible:focus-ring
		mt-1
	`
});

export { getTabItemStyles as a, getTabsStyles as b, getTabListStyles as c, getTabContentStyles as g };
