// @adaptavant/eds-core/dist/styles-015cb1bf.esm.js
import { t as tw } from './tw-c6a9a210.esm.js';

// ------------------------------ Dialog Wrapper ---------------------------------------

const dialogInlinePaddingClass = 'px-6';
const getDialogStyles = () => ({
  dialogWrapper: tw`
		bg-modal-overlay
		fixed
		flex
		h-full w-full #1
		inset-0
		items-center
		justify-center
		p-4
	`,
  // #1 can we replace with min-h-screen, w-screen?
  root: tw`
		[outline:none]
		bg-canvas
		flex
		flex-col
		gap-4
		high-contrast:outline
		max-h-full
		max-w-full
		pb-[calc(env(safe-area-inset-bottom)+24px)]
		pt-5
		rounded-8px
		shadow-30
	`
});

// ------------------------------ Dialog Header ---------------------------------------

const getDialogHeaderStyles = () => ({
  root: tw`
		${dialogInlinePaddingClass}
		flex
		flex-col
		gap-2
		pe-12 #2
		relative
		shrink-0
	`,
  // #2 additional padding end to ensure dialog title always has room for dialog clos
  closeButton: tw`
		absolute 
		right-3
		-top-2
	`
});

// ------------------------------ Dialog Content ---------------------------------------

const getDialogContentStyles = () => ({
  root: tw`
		${dialogInlinePaddingClass}
		overflow-auto
	`
});

// ------------------------------ Dialog Footer ---------------------------------------

const getDialogFooterStyles = () => ({
  root: tw`
		${dialogInlinePaddingClass}
		flex
		gap-2
		items-center
		justify-end
		pt-2
	`
});

export { getDialogFooterStyles as a, getDialogHeaderStyles as b, getDialogStyles as c, getDialogContentStyles as g };
