// @adaptavant/eds-core/dist/styles-12c293a4.esm.js
import { t as tw } from './tw-c6a9a210.esm.js';
import { t as typography } from './styles-267c9284.esm.js';

// -------------------------------- TextInput ----------------------------------

const getTextInputStyles = ({
  hasAdornmentEnd,
  hasAdornmentStart,
  size,
  isDisabled
}) => {
  const paddingEndStyle = getPaddingStyle({
    hasAdornment: hasAdornmentEnd,
    side: 'end',
    size
  });
  const paddingStartStyle = getPaddingStyle({
    hasAdornment: hasAdornmentStart,
    side: 'start',
    size
  });
  return {
    root: tw`
			${paddingEndStyle}
			${paddingStartStyle}
			${isDisabled ? 'cursor-default' : ''}
		`,
    adornmentEnd: tw`
			${typography[size]}
			flex
			items-center
		`,
    adornmentStart: tw`
			${typography[size]}
			flex
			items-center
		`
  };
};
const inputPaddingStylesLookup = {
  start: {
    withAdornment: {
      standard: tw`ps-1.5`,
      large: tw`ps-2`
    },
    withoutAdornment: {
      standard: tw`ps-2`,
      large: tw`ps-3`
    }
  },
  end: {
    withAdornment: {
      standard: tw`pe-1.5`,
      large: tw`pe-2`
    },
    withoutAdornment: {
      standard: tw`pe-2`,
      large: tw`pe-3`
    }
  }
};
function getPaddingStyle({
  side,
  hasAdornment,
  size
}) {
  const adornmentKey = hasAdornment ? 'withAdornment' : 'withoutAdornment';
  return inputPaddingStylesLookup[side][adornmentKey][size];
}

// ---------------------------- TextInputPrimitive -----------------------------

const getTextInputPrimitiveStyles = ({
  size
}) => ({
  root: tw`
		${inputHeightStylesLookup[size]}
		${inputLineHeightStylesLookup[size]}
		${typography[size]}
		[outline:none]
		aria-disabled:placeholder:text-disabled
		aria-disabled:text-disabled-secondary
		bg-transparent
		flex
		placeholder:text-secondary
		text-primary
		w-full
	`
});
const inputHeightStylesLookup = {
  standard: tw`h-8`,
  large: tw`h-10`
};
const inputLineHeightStylesLookup = {
  standard: tw`leading-8`,
  large: tw`leading-10`
};

export { getTextInputStyles as a, inputPaddingStylesLookup as b, getTextInputPrimitiveStyles as g, inputHeightStylesLookup as i };
