// @adaptavant/eds-core/dist/styles-9d8c2fa9.cjs.prod.js
'use strict';

var tw = require('./tw-9929267e.cjs.prod.js');
var styles = require('./styles-3786e407.cjs.prod.js');

// ----------------------------- Checkbox --------------------------------------

const getCheckboxStyles = ({
  isDisabled,
  size
}) => ({
  root: tw.tw`
	`,
  checkboxPrimitiveRoot: tw.tw`
	`,
  checkboxPrimitiveInput: tw.tw`
	`,
  checkboxPrimitiveControl: tw.tw`
	`,
  checkboxPrimitiveIcon: tw.tw`
	`,
  label: tw.tw`
		${isDisabled ? 'text-disabled' : ''}
		${styles.typography[size]}
		gap-2
		group
	`,
  descriptionTrack: tw.tw`
		gap-2
	`,
  description: tw.tw`
		${styles.typography[size]}
		${isDisabled ? 'text-disabled' : 'text-secondary'}
	`,
  spacer: tw.tw`
		${iconSizeMap[size]}
	`
});
const iconSizeMap = {
  large: tw.tw`
		h-4
		w-4
	`,
  standard: tw.tw`
		h-4
		w-4
	`
};

// ------------------------ CheckboxPrimitive ----------------------------------

const getCheckboxPrimitiveStyles = ({
  isInvalid,
  size
}) => ({
  root: tw.tw`
	`,
  input: tw.tw`
		peer
		sr-only
	`,
  control: tw.tw`
		${checkedStyles}
		${indeterminateStyles}
		${disabledStyles}
		${iconSizeMap[size]}
		${isInvalid ? invalidStyles : ''}
		${styles.typography[size]}
		[--visibility:hidden]
		[transition-timing-function:ease]
		bg-canvas
		border
		border-input
		duration-300
		group-hover:border-input-hover
		hover:border-input-hover
		inline-flex
		items-center
		justify-center
		peer-focus-visible:focus-ring
		rounded-4px
		text-transparent
		transition-colors
	`,
  icon: tw.tw`
		[visibility:--visibility]
		fill-current
		h-full
		w-full
	`
});
const checkedStyles = tw.tw`
	peer-checked:[--visibility:visible]
	peer-checked:bg-inverse
	peer-checked:border-input-active
	peer-checked:hover:border-input-active-hover
	peer-checked:text-inverse
`;
const indeterminateStyles = tw.tw`
	peer-indeterminate:[--visibility:visible]
	peer-indeterminate:bg-inverse
	peer-indeterminate:border-input-active
	peer-indeterminate:hover:border-input-active-hover
	peer-indeterminate:text-inverse
`;
const disabledStyles = tw.tw`
	peer-aria-disabled:bg-neutral-disabled
	peer-aria-disabled:border-input-disabled
	peer-aria-disabled:high-contrast:border-dotted
	peer-aria-disabled:hover:border-input-disabled
	peer-aria-disabled:peer-checked:text-disabled
	peer-aria-disabled:peer-indeterminate:text-disabled
	peer-aria-disabled:text-transparent
`;
const invalidStyles = tw.tw`
	peer-[[aria-invalid="true"]]:border-input-critical
	peer-[[aria-invalid="true"]]:group-hover:border-input-critical-hover
	peer-[[aria-invalid="true"]]:hover:border-input-critical-hover
	peer-[[aria-invalid="true"]]:peer-checked:border-input-critical
	peer-[[aria-invalid="true"]]:peer-indeterminate:border-input-critical
	peer-[[aria-invalid="true"]]:peer-checked:hover:border-input-critical-hover
	peer-[[aria-invalid="true"]]:peer-indeterminate:hover:border-input-critical-hover
	peer-[[aria-invalid="true"]]:peer-aria-disabled:border-input-disabled
`;

exports.getCheckboxPrimitiveStyles = getCheckboxPrimitiveStyles;
exports.getCheckboxStyles = getCheckboxStyles;
