// @adaptavant/eds-core/dist/styles-590374a3.cjs.dev.js
'use strict';

var styles = require('./styles-3e350289.cjs.dev.js');
var tw = require('./tw-cabb7e4c.cjs.dev.js');

// ----------------------------- FilterMenuTrigger -----------------------------

const getFilterMenuTriggerStyles = ({
  size
}) => ({
  root: tw.tw`
		${styles.typography[size]}
		${inlinePaddingStylesLookup[size]}
		active:bg-neutral-secondary-pressed
		aria-disabled:bg-neutral-disabled
		aria-disabled:text-disabled-secondary
		aria-expanded:bg-canvas
		border-transparent
		gap-2
		group
		high-contrast:outline
		hover:bg-canvas
		peer
		rounded-none
	`,
  chevron: tw.tw`
		fill-secondary
		group-aria-disabled:fill-tertiary
		me-0
	`,
  focusContainer: tw.tw`
	`,
  label: tw.tw`
		grow
		font-regular
		text-start
	`,
  labelTrack: tw.tw`
		w-full
	`
});
const inlinePaddingStylesLookup = {
  standard: 'px-2',
  large: 'px-3'
};

// --------------------------- FilterMenuSearchField ---------------------------

const getFilterMenuSearchFieldStyles = () => ({
  root: tw.tw`
			-mt-1
			-top-1
			bg-neutral
			inset-x-0
			py-1
			sticky
			z-20
		`
});

exports.getFilterMenuSearchFieldStyles = getFilterMenuSearchFieldStyles;
exports.getFilterMenuTriggerStyles = getFilterMenuTriggerStyles;
exports.inlinePaddingStylesLookup = inlinePaddingStylesLookup;
