// @adaptavant/eds-core/dist/styles-afa010e9.cjs.prod.js
'use strict';

var styles = require('./styles-3786e407.cjs.prod.js');
var tw = require('./tw-9929267e.cjs.prod.js');
var styles$1 = require('./styles-d86d95e2.cjs.prod.js');

// ----------------------------- SelectMenuTrigger -----------------------------

const getSelectMenuTriggerStyles = ({
  size
}) => ({
  root: tw.tw`
		${styles.typography[size]}
		${styles$1.inlinePaddingStylesLookup[size]}
		active:bg-neutral-secondary-pressed
		aria-disabled:bg-neutral-disabled
		aria-disabled:text-disabled-secondary
		border-transparent
		gap-2
		group
		high-contrast:outline
		hover:bg-canvas
		peer
		rounded-none
	`,
  chevron: tw.tw`
		fill-secondary
		group-aria-disabled:fill-tertiary
		me-0
	`,
  focusContainer: tw.tw`
	`,
  focusIndicator: tw.tw`
		peer-aria-expanded:border-2
		peer-aria-expanded:border-input-active
	`,
  label: tw.tw`
		grow
		font-regular
		text-start
	`,
  labelTrack: tw.tw`
		w-full
	`
});

// ----------------------------- SelectMenuPopover -----------------------------

const getSelectMenuPopoverStyles = () => ({
  modalWrapper: tw.tw`
	`,
  root: tw.tw`
	`
});

// ----------------------------- SelectMenuListbox -----------------------------

const getSelectMenuListboxStyles = () => ({
  root: tw.tw`
		overflow-y-auto
		outline-none
	`
});

exports.getSelectMenuListboxStyles = getSelectMenuListboxStyles;
exports.getSelectMenuPopoverStyles = getSelectMenuPopoverStyles;
exports.getSelectMenuTriggerStyles = getSelectMenuTriggerStyles;
