// @adaptavant/eds-core/dist/styles-ea402050.cjs.dev.js
'use strict';

var styles = require('./styles-3e350289.cjs.dev.js');
var tw = require('./tw-cabb7e4c.cjs.dev.js');

// ------------------------------ Toggle ----------------------------------------

const getToggleStyles = ({
  isDisabled,
  size
}) => ({
  root: tw.tw`
	`,
  togglePrimitiveRoot: tw.tw`
	`,
  togglePrimitiveInput: tw.tw`
	`,
  togglePrimitiveControl: tw.tw`
	`,
  togglePrimitiveIcon: tw.tw`
	`,
  label: tw.tw`
		${isDisabled ? 'text-disabled' : ''}
		${styles.typography[size]}
		gap-2
		group
	`,
  descriptionTrack: tw.tw`
		gap-2
	`,
  description: tw.tw`
		${styles.typography[size]}
		${isDisabled ? 'text-disabled' : 'text-secondary'}
	`,
  descriptionSpacer: tw.tw`
		${controlSizeMap[size]}
	`
});
const controlSizeMap = {
  standard: tw.tw`
		h-4
		w-7
	`,
  large: tw.tw`
		h-5
		w-9
	`
};

// -------------------------- TogglePrimitive -----------------------------------

const getTogglePrimitiveStyles = ({
  size
}) => ({
  root: tw.tw`
		relative
	`,
  input: tw.tw`
		peer
		sr-only
	`,
  control: tw.tw`
		${controlSizeMap[size]}
		[--translate-x:0px]
		bg-neutral-tertiary
		border
		border-transparent
		high-contrast:[outline:none]
		high-contrast:outline
		high-contrast:outline-transparent
		inline-flex
		items-center
		peer-aria-disabled:bg-neutral-disabled
		peer-aria-disabled:high-contrast:outline-dotted
		peer-aria-disabled:text-onPrimary-disabled
		peer-checked:[--translate-x:100%]
		peer-checked:bg-inverse
		peer-focus-visible:focus-ring
		px-px
		rounded-full
		text-onPrimary
		transition-colors
	`,
  icon: tw.tw`
		${thumbSizeMap[size]}
		duration-300
		fill-current
		high-contrast:outline
		high-contrast:outline-transparent
		transition-transform
		translate-x-[--translate-x]
	`
});
const thumbSizeMap = {
  standard: tw.tw`
		h-3
		w-3
	`,
  large: tw.tw`
		h-4
		w-4
	`
};

exports.getTogglePrimitiveStyles = getTogglePrimitiveStyles;
exports.getToggleStyles = getToggleStyles;
