// @adaptavant/eds-core/dist/styles-5ca722fa.esm.js
import { t as typography } from './styles-267c9284.esm.js';
import { t as tw } from './tw-c6a9a210.esm.js';

// ------------------------------ Toggle ----------------------------------------

const getToggleStyles = ({
  isDisabled,
  size
}) => ({
  root: tw`
	`,
  togglePrimitiveRoot: tw`
	`,
  togglePrimitiveInput: tw`
	`,
  togglePrimitiveControl: tw`
	`,
  togglePrimitiveIcon: tw`
	`,
  label: tw`
		${isDisabled ? 'text-disabled' : ''}
		${typography[size]}
		gap-2
		group
	`,
  descriptionTrack: tw`
		gap-2
	`,
  description: tw`
		${typography[size]}
		${isDisabled ? 'text-disabled' : 'text-secondary'}
	`,
  descriptionSpacer: tw`
		${controlSizeMap[size]}
	`
});
const controlSizeMap = {
  standard: tw`
		h-4
		w-7
	`,
  large: tw`
		h-5
		w-9
	`
};

// -------------------------- TogglePrimitive -----------------------------------

const getTogglePrimitiveStyles = ({
  size
}) => ({
  root: tw`
		relative
	`,
  input: tw`
		peer
		sr-only
	`,
  control: tw`
		${controlSizeMap[size]}
		[--translate-x:0px]
		bg-neutral-tertiary
		border
		border-transparent
		high-contrast:[outline:none]
		high-contrast:outline
		high-contrast:outline-transparent
		inline-flex
		items-center
		peer-aria-disabled:bg-neutral-disabled
		peer-aria-disabled:high-contrast:outline-dotted
		peer-aria-disabled:text-onPrimary-disabled
		peer-checked:[--translate-x:100%]
		peer-checked:bg-inverse
		peer-focus-visible:focus-ring
		px-px
		rounded-full
		text-onPrimary
		transition-colors
	`,
  icon: tw`
		${thumbSizeMap[size]}
		duration-300
		fill-current
		high-contrast:outline
		high-contrast:outline-transparent
		transition-transform
		translate-x-[--translate-x]
	`
});
const thumbSizeMap = {
  standard: tw`
		h-3
		w-3
	`,
  large: tw`
		h-4
		w-4
	`
};

export { getTogglePrimitiveStyles as a, getToggleStyles as g };
