// @adaptavant/eds-core/dist/styles-56e113a0.esm.js
import { t as tw } from './tw-c6a9a210.esm.js';

const getButtonStyles = ({
  buttonRadius,
  isDisabled,
  isLoading,
  size,
  variant
}) => ({
  root: tw`
		${buttonBaseStyles}
		${buttonDisabledStylesLookup[variant]}
		${buttonRadiusStylesLookup[buttonRadius]}
		${buttonSizeStylesLookup[size]}
		${buttonVariantStylesLookup[variant]}
		${isLoading || isDisabled ? '' : `${buttonHoverStylesLookup[variant]} ${buttonPressedStylesLookup[variant]}`}
		aria-disabled:cursor-default
		relative
		whitespace-nowrap
	`,
  labelTrack: tw`
		${isLoading && 'opacity-0'}
		gap-1
		min-w-0
		pointer-events-none #1
	`,
  // #1 Prevents Safari from displaying our :focus-visible style on .focus() https://bugs.webkit.org/show_bug.cgi?id=241408
  label: tw`
		-my-1
		font-strong
		line-clamp-1
		py-1
	`,
  iconStart: tw`
		-ms-1
	`,
  iconEnd: tw`
		-me-1
	`,
  loadingWrapper: tw`
		absolute
		flex
		inset-0
		items-center
		justify-center
		pointer-events-none
	`,
  loading: tw`
	`
});
const buttonBaseStyles = tw`
	[transition-timing-function:ease]
	border #1
	border-transparent
	cursor-pointer
	duration-300
	focus-visible:focus-ring
	inline-flex
	items-center
	justify-center
	transition-colors
`; // 1. All buttons should have the same border-width so they're all identical size

const buttonDisabledStylesLookup = {
  accentPrimary: tw`
		aria-disabled:bg-neutral-disabled
		aria-disabled:text-onPrimary-disabled
	`,
  neutralPrimary: tw`
		aria-disabled:border-secondary
		aria-disabled:text-disabled
	`,
  accentSecondary: tw`
		aria-disabled:text-disabled
		aria-disabled:border-secondary
	`,
  neutralSecondary: tw`
		aria-disabled:border-secondary
		aria-disabled:text-disabled
	`,
  neutralSecondaryIntense: tw`
		aria-disabled:border-secondary
		aria-disabled:text-disabled
	`,
  neutralTertiary: tw`
		aria-disabled:text-disabled
	`,
  criticalPrimary: tw`
		aria-disabled:bg-neutral-disabled
		aria-disabled:text-onPrimary-disabled
	`,
  criticalTertiary: tw`
		aria-disabled:text-disabled
	`
};
const buttonRadiusStylesLookup = {
  none: tw`rounded-none`,
  '4px': tw`rounded-4px`,
  '6px': tw`rounded-6px`,
  '8px': tw`rounded-8px`,
  '12px': tw`rounded-12px`,
  '16px': tw`rounded-16px`,
  full: tw`rounded-full`
};
const buttonSizeStylesLookup = {
  large: tw`
		h-10
		px-5
		text-body-14
	`,
  standard: tw`
		h-8
		px-4
		text-body-12
	`,
  small: tw`
		h-6
		px-3
		text-body-12
	`
};
const buttonVariantStylesLookup = {
  accentPrimary: tw`
		bg-accent
		text-onPrimary
	`,
  neutralPrimary: tw`
		bg-neutral
		border-secondary
		text-primary
	`,
  accentSecondary: tw`
		border-accent
		text-accent
	`,
  neutralSecondary: tw`
		bg-neutral
		border-secondary
		text-primary
	`,
  neutralSecondaryIntense: tw`
		bg-neutral
		border-primary
		text-primary
	`,
  neutralTertiary: tw`
		text-secondary
	`,
  criticalPrimary: tw`
		bg-critical
		text-critical-onPrimary
	`,
  criticalTertiary: tw`
		text-critical
	`
};
const buttonHoverStylesLookup = {
  accentPrimary: tw`
		hover:bg-accent-hover
	`,
  neutralPrimary: tw`
		hover:bg-inverse-hover
		hover:border-transparent
		hover:text-inverse
	`,
  accentSecondary: tw`
		hover:bg-accent-hover
		hover:border-transparent
		hover:text-onPrimary
	`,
  neutralSecondary: tw`
		hover:bg-neutral-hover
		hover:border-transparent
	`,
  neutralSecondaryIntense: tw`
		hover:bg-inverse-hover
		hover:border-transparent
		hover:text-inverse
	`,
  neutralTertiary: tw`
		hover:bg-neutral-hover
	`,
  criticalPrimary: tw`
		hover:bg-critical-hover
	`,
  criticalTertiary: tw`
		hover:bg-critical-secondary-hover
	`
};
const buttonPressedStylesLookup = {
  accentPrimary: tw`
		active:bg-accent-pressed
		aria-pressed:bg-accent-pressed
	`,
  neutralPrimary: tw`
		active:bg-inverse-pressed
		active:border-transparent
		active:text-inverse
		aria-pressed:bg-inverse-pressed
		aria-pressed:border-transparent
		aria-pressed:text-inverse
	`,
  accentSecondary: tw`
		active:bg-accent-pressed
		active:border-transparent
		active:text-onPrimary
		aria-pressed:bg-accent-pressed
		aria-pressed:border-transparent
		aria-pressed:text-onPrimary
	`,
  neutralSecondary: tw`
		active:bg-neutral-pressed
		active:border-transparent
		aria-pressed:bg-neutral-pressed
		aria-pressed:border-transparent
	`,
  neutralSecondaryIntense: tw`
		active:bg-inverse-pressed
		active:border-transparent
		active:text-inverse
		aria-pressed:bg-inverse-pressed
		aria-pressed:border-transparent
		aria-pressed:text-inverse
	`,
  neutralTertiary: tw`
		active:bg-neutral-pressed
		aria-pressed:bg-neutral-pressed
	`,
  criticalPrimary: tw`
		active:bg-critical-pressed
		aria-pressed:bg-critical-pressed
	`,
  criticalTertiary: tw`
		active:bg-critical-secondary-pressed
		aria-pressed:bg-critical-secondary-pressed
	`
};
const ariaExpandedButtonStylesLookup = {
  accentPrimary: tw`
		aria-expanded:bg-accent-pressed
	`,
  accentSecondary: tw`
		aria-expanded:bg-accent-pressed
		aria-expanded:border-transparent
		aria-expanded:text-onPrimary
	`,
  neutralPrimary: tw`
		aria-expanded:bg-inverse-pressed
		aria-expanded:border-transparent
		aria-expanded:text-inverse
	`,
  neutralSecondary: tw`
		aria-expanded:bg-neutral-pressed
		aria-expanded:border-transparent
	`,
  neutralTertiary: tw`		
		aria-expanded:bg-neutral-pressed
	`,
  criticalPrimary: tw`		
		aria-expanded:bg-critical-pressed
	`,
  criticalTertiary: tw`		
		aria-expanded:bg-critical-secondary-pressed
	`,
  neutralSecondaryIntense: tw`		
		aria-expanded:bg-inverse-pressed
		aria-expanded:border-transparent
		aria-expanded:text-inverse
	`
};

export { buttonDisabledStylesLookup as a, buttonBaseStyles as b, buttonHoverStylesLookup as c, buttonPressedStylesLookup as d, buttonRadiusStylesLookup as e, buttonSizeStylesLookup as f, buttonVariantStylesLookup as g, ariaExpandedButtonStylesLookup as h, getButtonStyles as i };
