// @adaptavant/eds-core/dist/tab-list-3e463566.cjs.dev.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var focus = require('@react-aria/focus');
var React = require('react');
var index = require('./index-3bdbdb42.cjs.dev.js');
var mergeProps = require('./merge-props-74bd38ff.cjs.dev.js');
var context = require('./context-1d4965bb.cjs.dev.js');
var styles = require('./styles-5c1faeec.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-88b49271.cjs.dev.js');
require('./tw-merge-70bc959c.cjs.dev.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js');
require('@react-aria/utils');
require('./cx-26784cbb.cjs.dev.js');
require('clsx');
require('./merge-ids-b19ec20e.cjs.dev.js');
require('./tw-cabb7e4c.cjs.dev.js');

/**
 * Represents a wrapper for TabItems.
 * It highlights the selected element with a line and displays a scroll mask when necessary
 */
function TabList({
  className = '',
  classNames = {},
  style = {},
  styles: styles$1 = {},
  children,
  ...consumerProps
}) {
  const {
    size
  } = context.useTabsContext();
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getTabListStyles({
      size
    })
  });
  const {
    selectedElement,
    tabListA11yProps
  } = context.useTabsContext();
  const {
    scrollRef,
    visibleMasks
  } = useScrollMask();
  const leftMaskIsVisible = ['both', 'end'].includes(visibleMasks);
  const rightMaskIsVisible = ['both', 'start'].includes(visibleMasks);
  const tabHighlighterStyles = useTabHighlighterStyles();
  return /*#__PURE__*/jsxRuntime.jsx(focus.FocusScope, {
    children: /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
      ...styleProps('root'),
      children: [/*#__PURE__*/jsxRuntime.jsxs(box.Box, {
        ...mergeProps.mergeProps(consumerProps, tabListA11yProps),
        ...styleProps('tabItemWrapper'),
        ref: scrollRef,
        children: [children, selectedElement ? /*#__PURE__*/jsxRuntime.jsx(box.Box, {
          as: "span",
          "data-item-highlighter": true,
          ...styleProps('highlighter'),
          style: {
            ...styleProps('highlighter').style,
            ...tabHighlighterStyles
          }
        }) : null]
      }), /*#__PURE__*/jsxRuntime.jsx(box.Box, {
        as: "span",
        ...styleProps('scrollMaskStart'),
        style: {
          ...styleProps('scrollMaskStart').style,
          background: `linear-gradient(90deg, var(--background-canvas) 0%, transparent 100%)`,
          opacity: leftMaskIsVisible ? 1 : 0
        }
      }), /*#__PURE__*/jsxRuntime.jsx(box.Box, {
        as: "span",
        ...styleProps('scrollMaskEnd'),
        style: {
          ...styleProps('scrollMaskEnd').style,
          background: `linear-gradient(90deg, transparent 0%, var(--background-canvas) 100%)`,
          opacity: rightMaskIsVisible ? 1 : 0
        }
      })]
    })
  });
}
TabList.displayName = 'TabList';

// This scroll mask functionality is already implemented for the booking app, but it is not exportable. Refer: https://github.com/Adaptavant/awds/pull/682
// May be in future, this hook will allow us to reuse the functionality in other parts of the monorepo (design system, booking app, etc.,) where it is required.

/**
 * Custom React hook that manages the visibility of scroll masks
 * based on the scroll position of the scrollable element.
 */
function useScrollMask() {
  const [visibleMasks, setVisibleMasks] = React.useState('none');
  const scrollRef = React.useRef(null);
  React.useEffect(() => {
    const scrollElement = scrollRef.current;
    if (scrollElement) {
      // Function to handle scroll events and update the visibility of scroll masks.
      const handleScroll = () => {
        const {
          clientWidth,
          scrollWidth,
          scrollLeft
        } = scrollElement;

        // Don't show masks if there's no scroll available
        if (scrollWidth === clientWidth) {
          setVisibleMasks('none');
          return;
        }

        // LTE/GTE to cater for Safari's bounce
        if (scrollLeft <= 0) {
          setVisibleMasks('start');
        } else if (Math.round(scrollLeft) >= scrollWidth - clientWidth) {
          setVisibleMasks('end');
        } else {
          setVisibleMasks('both');
        }
      };
      handleScroll();

      // For scroll events
      scrollElement.addEventListener('scroll', handleScroll, {
        passive: true
      }); // { passive: true } for better performance. Refer: https://github.com/WICG/EventListenerOptions/blob/gh-pages/explainer.md#passive-event-listeners.

      // For resize events
      const resizeObserver = new ResizeObserver(() => {
        handleScroll();
      });
      resizeObserver.observe(scrollElement);
      return () => {
        scrollElement.removeEventListener('scroll', handleScroll);
        resizeObserver.disconnect();
      };
    }
  }, []);
  return {
    scrollRef,
    visibleMasks
  };
}

/**
 * Custom React hook that calculates the width and offset for the tab highlighter
 */
function useTabHighlighterStyles() {
  const {
    selectedElement
  } = context.useTabsContext();
  const [highlighterStyles, setHighlighterStyles] = React.useState(undefined);

  // Calculate width and offset for the tab highlighter
  React.useMemo(() => {
    const selectedElementOriginalWidth = (selectedElement === null || selectedElement === void 0 ? void 0 : selectedElement.offsetWidth) ?? 0;
    const selectedElementLeft = (selectedElement === null || selectedElement === void 0 ? void 0 : selectedElement.offsetLeft) ?? 0;
    setHighlighterStyles({
      width: `${selectedElementOriginalWidth}px`,
      transform: `translateX(${selectedElementLeft}px)`
    });
  }, [selectedElement]);
  return highlighterStyles;
}

exports.TabList = TabList;
