// @adaptavant/eds-core/dist/table-a71917f2.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var reactTable = require('@tanstack/react-table');
var React = require('react');
var index = require('./index-7955fb8e.cjs.prod.js');
var tw = require('./tw-9929267e.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-e39057bd.cjs.prod.js');
var iconButton = require('./icon-button-42ea08f2.cjs.prod.js');
var chevronRightIcon = require('./chevron-right-icon-2a45d98c.cjs.prod.js');
var text = require('./text-a5ab226e.cjs.prod.js');
var translations = require('./translations-4a8a2311.cjs.prod.js');
var mergeProps = require('./merge-props-60701ce8.cjs.prod.js');
var stack = require('./stack-fa75d919.cjs.prod.js');
var dropdownUpIcon = require('./dropdown-up-icon-ef0561ac.cjs.prod.js');
var dropdownDownIcon = require('./dropdown-down-icon-677561cc.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');
require('./button-c6c568ca.cjs.prod.js');
require('./use-button-c05d68a1.cjs.prod.js');
require('./styles-3c34b522.cjs.prod.js');
require('./styles-4d7be1cb.cjs.prod.js');
require('./loading-8f0b3165.cjs.prod.js');
require('./grow-your-brand-icon-cce9ad51.cjs.prod.js');
require('./styles-30814c9d.cjs.prod.js');
require('./setmore-icon-24e17637.cjs.prod.js');
require('./config-7c4ee3f2.cjs.prod.js');
require('@react-aria/utils');
require('./cx-86353291.cjs.prod.js');
require('clsx');
require('./merge-ids-2d47bd24.cjs.prod.js');

const getTableStyles = () => ({
  baseContainer: tw.tw`
		flex 
		flex-col 
		text-body-12
	`,
  tableContainer: tw.tw`
		overflow-auto
 	`,
  table: tw.tw`
		w-full
  	`,
  row: tw.tw`
		border-b 
		border-tertiary
		h-12
	`,
  cell: tw.tw`
		p-2 
	`,
  footerContainer: tw.tw`
		flex 
		items-center 
		justify-between
		py-2
	`
});
const getPaginationStyles = () => ({
  root: tw.tw`
		flex 
		gap-1
		items-center 
		ml-auto	
  	`,
  text: tw.tw`
		shrink-0
		text-body-12 
		text-center 			
	`
});
const getHeaderStyles = () => ({
  root: tw.tw`
	`,
  row: tw.tw`
		border-b 
		border-secondary
		h-12
	`,
  cell: tw.tw`
	    font-regular
		px-1 
		py-3 
		text-left 
		text-secondary 			
	`,
  contentWrapper: tw.tw`
	    flex 
		group
		gap-1 
		items-center
		pl-1
	`,
  iconsWrapper: tw.tw`
	    group-focus:visible
	    group-hover:visible 
		invisible
	`
});

function Pagination({
  className = '',
  classNames = {},
  style = {},
  styles = {},
  tableInstance
}) {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getPaginationStyles()
  });
  const {
    experimental_table: translations$1
  } = translations.useTranslations();
  const GetCurrentVisibleRows = () => {
    const {
      pageIndex,
      pageSize
    } = tableInstance.getState().pagination;
    const totalRows = tableInstance.getRowCount();

    // Calculate start and end as strings
    const start = (pageIndex * pageSize + 1).toString();
    const end = Math.min((pageIndex + 1) * pageSize, totalRows).toString();

    // Replace placeholders with actual values
    const currentPageInfo = translations$1.paginationContent.replace('{start}', start).replace('{end}', end).replace('{totalRows}', totalRows.toString());
    return /*#__PURE__*/jsxRuntime.jsx(text.Text, {
      ...styleProps('text'),
      children: currentPageInfo
    });
  };
  return /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
    ...styleProps('root'),
    children: [/*#__PURE__*/jsxRuntime.jsx(GetCurrentVisibleRows, {}), /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
      children: [/*#__PURE__*/jsxRuntime.jsx(iconButton.IconButton, {
        "aria-label": translations$1.previousPageLabel,
        icon: chevronRightIcon.ChevronLeftIcon,
        isDisabled: !tableInstance.getCanPreviousPage(),
        onClick: () => tableInstance.previousPage(),
        variant: "neutralTertiary"
      }), /*#__PURE__*/jsxRuntime.jsx(iconButton.IconButton, {
        "aria-label": translations$1.nextPageLabel,
        icon: chevronRightIcon.ChevronRightIcon,
        isDisabled: !tableInstance.getCanNextPage(),
        onClick: () => tableInstance.nextPage(),
        variant: "neutralTertiary"
      })]
    })]
  });
}

function TableHeader({
  className = '',
  classNames = {},
  style = {},
  styles = {},
  tableInstance
}) {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getHeaderStyles()
  });

  // Check if there are any rows available in the table instance
  const isDataAvailable = tableInstance.getRowModel().rows.length > 0;

  // Function to render sorting icons based on the column's sorting state
  function SortingIcons({
    column
  }) {
    // Return null if sorting is not allowed or no data is available
    if (!column.getCanSort() || !isDataAvailable) return null;
    const sortingState = column.getIsSorted();
    const isAscending = sortingState === 'asc';
    const isDescending = sortingState === 'desc';
    return /*#__PURE__*/jsxRuntime.jsxs(stack.Stack, {
      ...mergeProps.mergeProps(styleProps('iconsWrapper'), {
        className: sortingState !== false ? 'visible' : ''
      }),
      children: [/*#__PURE__*/jsxRuntime.jsx(dropdownUpIcon.DropdownUpIcon, {
        ...mergeProps.mergeProps(styleProps('icon'), {
          className: '-mb-2.5'
        }),
        size: "16",
        tone: isDescending || sortingState === false ? 'neutralTertiary' : undefined
      }), /*#__PURE__*/jsxRuntime.jsx(dropdownDownIcon.DropdownDownIcon, {
        ...styleProps('icon'),
        size: "16",
        tone: isAscending || sortingState === false ? 'neutralTertiary' : undefined
      })]
    });
  }
  return /*#__PURE__*/jsxRuntime.jsx("thead", {
    role: "rowgroup",
    ...styleProps('root'),
    children: tableInstance.getHeaderGroups().map(({
      id: headerGroupId,
      headers
    }) => /*#__PURE__*/React.createElement("tr", {
      ...styleProps('row'),
      key: headerGroupId,
      role: "row"
    }, headers.map(({
      id: headerId,
      index,
      colSpan,
      column,
      isPlaceholder,
      getContext
    }) => {
      return /*#__PURE__*/React.createElement("th", {
        ...styleProps('cell'),
        "aria-colindex": index + 1
        // Set aria-sort when active sort direction is set
        ,
        ...(column.getIsSorted() ? {
          'aria-sort': column.getIsSorted() === 'asc' ? 'ascending' : 'descending'
        } : {}),
        colSpan: colSpan,
        key: headerId
        // Toggle sorting when the header cell is clicked, if data is available
        ,
        onClick: isDataAvailable ? column.getToggleSortingHandler() : undefined,
        role: "columnheader",
        scope: column.depth === 0 ? 'col' : 'colgroup'
      }, /*#__PURE__*/jsxRuntime.jsxs(box.Box
      // Use button if the column can be sorted, otherwise use div
      , {
        as: column.getCanSort() ? 'button' : 'div',
        ...mergeProps.mergeProps(styleProps('contentWrapper'), {
          className: column.getCanSort() ? 'focus-visible:focus-ring rounded-6px' : ''
        }),
        children: [/*#__PURE__*/jsxRuntime.jsx(box.Box, {
          as: "span",
          ...styleProps('content'),
          children: isPlaceholder ? null : reactTable.flexRender(column.columnDef.header, getContext())
        }), /*#__PURE__*/jsxRuntime.jsx(SortingIcons, {
          column: column
        })]
      }));
    })))
  });
}

/**
 * Table
 *
 * @description
 * An adaptable and accessible table component built using the `@tanstack/react-table` library.
 * Offers important features:
 * - Ensures accessibility with proper ARIA roles and attributes.
 * - Supports customizable columns for flexible data representation.
 * - Provides pagination for large datasets.
 * - Handles empty data gracefully with an empty template.
 */

function Table({
  columns,
  className = '',
  classNames = {},
  data,
  emptyTemplate,
  enableSorting,
  pagination,
  role = 'table',
  sortDescFirst,
  sortingFns,
  style = {},
  styles = {}
}) {
  const parts = index.useComponentStyleParts({
    className,
    classNames,
    style,
    styles,
    tw: getTableStyles()
  }, ['baseContainer', 'tableContainer', 'table', 'header', 'headerRow', 'headerCell', 'headerContentWrapper', 'headerContent', 'sortIconsWrapper', 'sortIcon', 'body', 'row', 'cell', 'footerContainer']);
  const headerStyleProps = {
    ...parts.header,
    classNames: {
      row: parts.headerRow.className,
      cell: parts.headerCell.className,
      contentWrapper: parts.headerContentWrapper.className,
      content: parts.headerContent.className,
      iconsWrapper: parts.sortIconsWrapper.className,
      icon: parts.sortIcon.className
    },
    styles: {
      row: parts.headerRow.style,
      cell: parts.headerCell.style,
      contentWrapper: parts.headerContentWrapper.style,
      content: parts.headerContent.style,
      iconsWrapper: parts.sortIconsWrapper.style,
      icon: parts.sortIcon.style
    }
  };

  // sorting state initialization
  const [sorting, setSorting] = React.useState([]);
  const tableInstance = reactTable.useReactTable({
    columns,
    data,
    enableSorting,
    getCoreRowModel: reactTable.getCoreRowModel(),
    getSortedRowModel: reactTable.getSortedRowModel(),
    onSortingChange: setSorting,
    sortDescFirst,
    sortingFns: sortingFns,
    state: {
      sorting
    },
    ...(pagination !== null && pagination !== void 0 && pagination.pageSize ? {
      getPaginationRowModel: reactTable.getPaginationRowModel()
    } : {}),
    initialState: pagination !== null && pagination !== void 0 && pagination.pageSize ? {
      pagination: {
        pageSize: pagination.pageSize,
        pageIndex: 0
      }
    } : {}
  });
  return /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
    ...parts.baseContainer,
    children: [/*#__PURE__*/jsxRuntime.jsx(box.Box, {
      ...parts.tableContainer,
      children: /*#__PURE__*/jsxRuntime.jsxs("table", {
        ...parts.table,
        "aria-colcount": tableInstance.getAllColumns().length,
        "aria-rowcount": tableInstance.getRowCount(),
        role: role,
        children: [/*#__PURE__*/jsxRuntime.jsx(TableHeader, {
          ...headerStyleProps,
          tableInstance: tableInstance
        }), /*#__PURE__*/jsxRuntime.jsx("tbody", {
          ...parts.body,
          role: "rowgroup",
          children: data.length === 0 ? /*#__PURE__*/jsxRuntime.jsx("tr", {
            children: /*#__PURE__*/jsxRuntime.jsx("td", {
              colSpan: columns.length,
              children: emptyTemplate || null
            })
          }) : tableInstance.getRowModel().rows.map(({
            id: rowId,
            index: rowIndex,
            getVisibleCells
          }) => /*#__PURE__*/React.createElement("tr", {
            ...parts.row,
            "aria-rowindex": rowIndex + 1,
            key: rowId,
            role: "row"
          }, getVisibleCells().map(({
            id: cellId,
            column,
            getContext
          }) => /*#__PURE__*/React.createElement("td", {
            ...parts.cell,
            "aria-colindex": column.getIndex() + 1,
            key: cellId,
            role: role === 'grid' ? 'gridcell' : 'cell'
          }, reactTable.flexRender(column.columnDef.cell, getContext())))))
        })]
      })
    }), pagination !== null && pagination !== void 0 && pagination.pageSize ? /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      ...parts.footerContainer,
      children: /*#__PURE__*/jsxRuntime.jsx(Pagination, {
        tableInstance: tableInstance
      })
    }) : null]
  });
}
Table.displayName = 'Table';

exports.Table = Table;
