// @adaptavant/eds-core/dist/tag-button-7091945f.esm.js
'use client';
import { forwardRef } from 'react';
import { g as getStyleProps } from './index-dd10a77e.esm.js';
import { a as getTagButtonStyles } from './styles-0248b764.esm.js';
import { jsx } from 'react/jsx-runtime';
import { useButton } from './use-button-36a8da62.esm.js';
import './tw-merge-91a8b6ee.esm.js';
import 'tailwind-merge';
import '../tokens/dist/adaptavant-eds-core-tokens.esm.js';

const TagButton = /*#__PURE__*/forwardRef(function TagButton({
  className = '',
  classNames = {},
  style = {},
  styles = {},
  icon: Icon,
  onClick,
  ...consumerProps
}, forwardedRef) {
  const {
    buttonProps
  } = useButton({
    isDisabled: false,
    isLoading: false,
    isPressed: false,
    onClick
  }, forwardedRef);
  const styleProps = getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getTagButtonStyles()
  });
  return /*#__PURE__*/jsx("button", {
    ...consumerProps,
    ...buttonProps,
    ...styleProps('button'),
    children: /*#__PURE__*/jsx(Icon, {
      size: "16",
      tone: "neutralTertiary",
      ...styleProps('icon')
    })
  });
});

export { TagButton };
