// @adaptavant/eds-core/dist/tag-fdb5d51b.esm.js
'use client';
import { g as getStyleProps } from './index-dd10a77e.esm.js';
import { g as getTagStyles } from './styles-0248b764.esm.js';
import { jsx } from 'react/jsx-runtime';
import { T as Text } from './text-3fc88ad6.esm.js';
import { Track } from './track-26dc4fe4.esm.js';
import './tw-merge-91a8b6ee.esm.js';
import 'tailwind-merge';
import '../tokens/dist/adaptavant-eds-core-tokens.esm.js';
import 'react';

/**
 * Tag
 *
 * @description A tag component is an item with the short label, which helps
 * in any sort of categorization or filtering. Usually, it appears in a group.
 * Each Tag can be interactive for a better user experience.
 */

function Tag({
  adornmentEnd,
  adornmentStart,
  children,
  className = '',
  classNames = {},
  style = {},
  styles = {},
  ...consumerProps
}) {
  const styleProps = getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: getTagStyles({
      hasAdornmentStart: Boolean(adornmentStart),
      hasAdornmentEnd: Boolean(adornmentEnd)
    })
  });
  return /*#__PURE__*/jsx(Track, {
    ...consumerProps,
    as: "span",
    classNames: {
      railEnd: `
					flex
					items-center
				`,
      railStart: `
					flex
					items-center
				`
    },
    railEnd: adornmentEnd,
    railStart: adornmentStart,
    ...styleProps('root'),
    children: /*#__PURE__*/jsx(Text, {
      ...styleProps('label'),
      children: children
    })
  });
}
Tag.displayName = 'Tag';

export { Tag };
