// @adaptavant/eds-core/dist/tag-button-ad1adc90.cjs.dev.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var index = require('./index-3bdbdb42.cjs.dev.js');
var styles = require('./styles-778f345d.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var use_button = require('./use-button-2241e53a.cjs.dev.js');
require('./tw-merge-70bc959c.cjs.dev.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js');

const TagButton = /*#__PURE__*/React.forwardRef(function TagButton({
  className = '',
  classNames = {},
  style = {},
  styles: styles$1 = {},
  icon: Icon,
  onClick,
  ...consumerProps
}, forwardedRef) {
  const {
    buttonProps
  } = use_button.useButton({
    isDisabled: false,
    isLoading: false,
    isPressed: false,
    onClick
  }, forwardedRef);
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getTagButtonStyles()
  });
  return /*#__PURE__*/jsxRuntime.jsx("button", {
    ...consumerProps,
    ...buttonProps,
    ...styleProps('button'),
    children: /*#__PURE__*/jsxRuntime.jsx(Icon, {
      size: "16",
      tone: "neutralTertiary",
      ...styleProps('icon')
    })
  });
});

exports.TagButton = TagButton;
