// @adaptavant/eds-core/dist/tag-button-093f7e0a.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var index = require('./index-7955fb8e.cjs.prod.js');
var styles = require('./styles-b479b378.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var use_button = require('./use-button-c05d68a1.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');

const TagButton = /*#__PURE__*/React.forwardRef(function TagButton({
  className = '',
  classNames = {},
  style = {},
  styles: styles$1 = {},
  icon: Icon,
  onClick,
  ...consumerProps
}, forwardedRef) {
  const {
    buttonProps
  } = use_button.useButton({
    isDisabled: false,
    isLoading: false,
    isPressed: false,
    onClick
  }, forwardedRef);
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getTagButtonStyles()
  });
  return /*#__PURE__*/jsxRuntime.jsx("button", {
    ...consumerProps,
    ...buttonProps,
    ...styleProps('button'),
    children: /*#__PURE__*/jsxRuntime.jsx(Icon, {
      size: "16",
      tone: "neutralTertiary",
      ...styleProps('icon')
    })
  });
});

exports.TagButton = TagButton;
