// @adaptavant/eds-core/dist/tag-1b40c136.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index = require('./index-7955fb8e.cjs.prod.js');
var styles = require('./styles-b479b378.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var text = require('./text-a5ab226e.cjs.prod.js');
var track = require('./track-f65e0bbd.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');
require('react');

/**
 * Tag
 *
 * @description A tag component is an item with the short label, which helps
 * in any sort of categorization or filtering. Usually, it appears in a group.
 * Each Tag can be interactive for a better user experience.
 */

function Tag({
  adornmentEnd,
  adornmentStart,
  children,
  className = '',
  classNames = {},
  style = {},
  styles: styles$1 = {},
  ...consumerProps
}) {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getTagStyles({
      hasAdornmentStart: Boolean(adornmentStart),
      hasAdornmentEnd: Boolean(adornmentEnd)
    })
  });
  return /*#__PURE__*/jsxRuntime.jsx(track.Track, {
    ...consumerProps,
    as: "span",
    classNames: {
      railEnd: `
					flex
					items-center
				`,
      railStart: `
					flex
					items-center
				`
    },
    railEnd: adornmentEnd,
    railStart: adornmentStart,
    ...styleProps('root'),
    children: /*#__PURE__*/jsxRuntime.jsx(text.Text, {
      ...styleProps('label'),
      children: children
    })
  });
}
Tag.displayName = 'Tag';

exports.Tag = Tag;
