// @adaptavant/eds-core/dist/text-input-primitive-06f88f66.cjs.dev.js
'use strict';

var React = require('react');
var index = require('./index-3bdbdb42.cjs.dev.js');
var styles = require('./styles-b5db22e5.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-88b49271.cjs.dev.js');

const TextInputPrimitive = /*#__PURE__*/React.forwardRef(function TextInputPrimitive({
  'aria-disabled': ariaDisabled,
  className = '',
  classNames = {},
  onKeyDown,
  size,
  style = {},
  styles: styles$1 = {},
  type = 'text',
  ...consumerProps
}, forwardedRef) {
  const handleKeyDown = React.useCallback(event => {
    if (ariaDisabled && event.key !== 'Tab') {
      event.preventDefault();
    }
    if (onKeyDown) {
      onKeyDown(event);
    }
  }, [ariaDisabled, onKeyDown]);
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getTextInputPrimitiveStyles({
      size
    })
  });
  return /*#__PURE__*/jsxRuntime.jsx(box.Box, {
    ...consumerProps,
    "aria-disabled": ariaDisabled,
    as: "input",
    autoComplete: ariaDisabled ? 'off' : consumerProps.autoComplete,
    onKeyDown: handleKeyDown,
    ref: forwardedRef,
    type: type,
    ...styleProps('root')
  });
});

exports.TextInputPrimitive = TextInputPrimitive;
