// @adaptavant/eds-core/dist/text-input-ca45a4ba.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var utils = require('@react-aria/utils');
var React = require('react');
var zeroWidthSpace = require('./zero-width-space-59cc78a9.cjs.prod.js');
var styles = require('./styles-291d4f6c.cjs.prod.js');
var textInputPrimitive = require('./text-input-primitive-20d226a2.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var text = require('./text-a5ab226e.cjs.prod.js');
var context = require('./context-008fcad0.cjs.prod.js');
var focus_container = require('./focus-container-be948f3e.cjs.prod.js');
require('./tw-9929267e.cjs.prod.js');
require('./styles-3786e407.cjs.prod.js');
require('./index-7955fb8e.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');
require('./box-e39057bd.cjs.prod.js');

/**
 * TextInput
 *
 * @description
 * The TextInput component provides a way for inputting text.
 * This component must be nested within a Field or InlineField.
 */
const TextInput = /*#__PURE__*/React.forwardRef(function TextInput({
  adornmentEnd,
  adornmentStart,
  className = '',
  classNames = {},
  focusContainerRef,
  style = {},
  styles: styles$1 = {},
  ...consumerProps
}, forwardedRef) {
  const {
    isDisabled,
    isInvalid,
    isRequired,
    size,
    variant
  } = context.useFieldContext();
  const a11yProps = context.useInputA11yProps({
    isDisabled,
    isInvalid,
    isRequired
  });
  const styleProps = focus_container.useCombinedFocusContainerInputStyles({
    className,
    classNames,
    isDisabled,
    isInvalid,
    style,
    styles: styles$1,
    tw: styles.getTextInputStyles({
      hasAdornmentEnd: Boolean(adornmentEnd),
      hasAdornmentStart: Boolean(adornmentStart),
      size,
      isDisabled
    })
  });
  return /*#__PURE__*/jsxRuntime.jsx(focus_container.FocusContainer, {
    adornmentEnd: adornmentEnd ? /*#__PURE__*/jsxRuntime.jsxs(text.Text, {
      ...styleProps('adornmentEnd'),
      children: [/*#__PURE__*/jsxRuntime.jsx(zeroWidthSpace.ZeroWidthSpace, {}), adornmentEnd]
    }) : undefined,
    adornmentStart: adornmentStart ? /*#__PURE__*/jsxRuntime.jsxs(text.Text, {
      ...styleProps('adornmentStart'),
      children: [/*#__PURE__*/jsxRuntime.jsx(zeroWidthSpace.ZeroWidthSpace, {}), adornmentStart]
    }) : undefined,
    classNames: {
      focusIndicator: {
        ...styleProps('focusIndicator')
      }.className
    },
    isDisabled: isDisabled,
    isGhost: variant === 'subtle',
    isInvalid: isInvalid,
    ref: focusContainerRef,
    styles: {
      focusIndicator: {
        ...styleProps('focusIndicator').style
      }
    },
    children: ({
      ref
    }) => /*#__PURE__*/jsxRuntime.jsx(textInputPrimitive.TextInputPrimitive, {
      ...consumerProps,
      ...a11yProps,
      ref: utils.mergeRefs(ref, forwardedRef),
      size: size,
      ...styleProps('root')
    })
  });
});
TextInput.displayName = 'TextInput';

exports.TextInput = TextInput;
