// @adaptavant/eds-core/dist/timepicker-input-2452d125.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var utils = require('@react-aria/utils');
var React = require('react');
var context$1 = require('./context-38cb595b.cjs.prod.js');
var mergeProps = require('./merge-props-60701ce8.cjs.prod.js');
var combobox_search_input = require('./combobox-search-input-090c1a94.cjs.prod.js');
var context = require('./context-3dc3ef27.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var text_input = require('./text-input-ca45a4ba.cjs.prod.js');
var context$2 = require('./context-008fcad0.cjs.prod.js');
require('./cx-86353291.cjs.prod.js');
require('clsx');
require('./merge-ids-2d47bd24.cjs.prod.js');

const TimePickerInput = /*#__PURE__*/React.forwardRef(function TimePickerInput(props, forwardedRef) {
  const {
    inputProps
  } = useTimePickerInput(props);
  return /*#__PURE__*/jsxRuntime.jsxs(React.Fragment, {
    children: [/*#__PURE__*/jsxRuntime.jsx(text_input.TextInput, {
      ...inputProps,
      ref: forwardedRef
    }), /*#__PURE__*/jsxRuntime.jsx(combobox_search_input.LiveRegion, {})]
  });
});
function useTimePickerInput(props) {
  const {
    onEnterInputHandler,
    ...restProps
  } = props;
  const {
    inputValue = '',
    // Set inputValue to an empty string to prevent uncontrolled to controlled input warning. More info: https://reactjs.org/link/uncontrolled-to-controlled
    menuTrigger,
    onInputChange,
    onSelectionChange
  } = context.useCombobox();
  const {
    activeDescendantId,
    activeDescendantIndex,
    isMenuOpen,
    listId,
    popover,
    openMenu,
    closeMenu,
    clearSelectedItem
  } = context$1.useDropdownContext();
  const {
    isDisabled
  } = context$2.useFieldContext();
  const NavigateToBlur = () => {
    if (activeDescendantIndex === -1) {
      onEnterInputHandler();
      if (isMenuOpen) closeMenu();
    }
  };
  const handleChange = React.useCallback(event => {
    if (!isMenuOpen) openMenu();
    if (onInputChange) onInputChange(event.target.value);
    if (event.target.value === '') {
      clearSelectedItem();
      if (onSelectionChange) {
        onSelectionChange(undefined);
      }
    }
  }, [clearSelectedItem, isMenuOpen, onInputChange, onSelectionChange, openMenu]);
  const handleFocus = React.useCallback(() => {
    if (menuTrigger === 'focus') openMenu();
  }, [menuTrigger, openMenu]);
  const referenceProps = popover.getReferenceProps();
  const handleKeyDown = combobox_search_input.useKeydownNavigation(NavigateToBlur);
  const isAppleDevice = utils.isAppleDevice();
  const a11yProps = React.useMemo(() => ({
    'aria-activedescendant': isAppleDevice ? undefined : activeDescendantId,
    'aria-autocomplete': 'list',
    'aria-controls': listId,
    'aria-expanded': isMenuOpen,
    'aria-haspopup': 'listbox',
    autoComplete: 'off',
    autoCorrect: 'off',
    spellCheck: 'false',
    role: 'combobox'
  }), [activeDescendantId, isAppleDevice, isMenuOpen, listId]);
  const comboboxProps = React.useMemo(() => ({
    onKeyDown: isDisabled ? undefined : handleKeyDown,
    onChange: handleChange,
    onFocus: isDisabled ? undefined : handleFocus,
    value: inputValue
  }), [isDisabled, handleKeyDown, handleChange, handleFocus, inputValue]);
  return {
    inputProps: React.useMemo(() => ({
      ...mergeProps.mergeProps(restProps, a11yProps, comboboxProps),
      focusContainerRef: referenceProps.ref
    }), [restProps, a11yProps, comboboxProps, referenceProps.ref])
  };
}

exports.TimePickerInput = TimePickerInput;
