// @adaptavant/eds-core/dist/timepicker-item-bd24c49d.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var utils = require('@react-aria/utils');
var React = require('react');
var context = require('./context-38cb595b.cjs.prod.js');
var use_dropdown_item_id = require('./use-dropdown-item-id-4f2bd16e.cjs.prod.js');
var useScrollIntoView = require('./use-scroll-into-view-af77842d.cjs.prod.js');
var index = require('./index-7955fb8e.cjs.prod.js');
var forwardRef = require('./forward-ref-0e64c63e.cjs.prod.js');
var context$1 = require('./context-3dc3ef27.cjs.prod.js');
var styles = require('./styles-18fee4e4.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var spacer = require('./spacer-388e98f9.cjs.prod.js');
var menu_item = require('./menu-item-5316e708.cjs.prod.js');
var align_child_to_text = require('./align-child-to-text-c75908d4.cjs.prod.js');
var checkIcon = require('./check-icon-19cfc34c.cjs.prod.js');
require('./utils-72a515c0.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');
require('./tw-9929267e.cjs.prod.js');
require('./box-e39057bd.cjs.prod.js');
require('./styles-30814c9d.cjs.prod.js');

/**
 * Item to be rendered inside of a TimePickerListBox.
 */
const TimePickerItem = forwardRef.forwardRefWithGenericComponent(function TimePickerItem({
  children,
  className = '',
  classNames = {},
  id: idProp,
  onClick,
  option,
  timeOptions,
  style = {},
  styles: styles$1 = {},
  ...consumerProps
}, forwardedRef) {
  const {
    activeDescendantId,
    closeMenu
  } = context.useDropdownContext();
  const {
    onSelectionChange,
    selectedKey
  } = context$1.useCombobox();
  const {
    selectedOption
  } = context$1.useCombobox();
  const id = use_dropdown_item_id.useDropdownItemId(idProp);
  const isHighlighted = id === activeDescendantId;
  const isSelected = Boolean(selectedOption && selectedOption[selectedKey] === option[selectedKey]);

  // check if selectedOption is from the timeoptionslist to remove
  // addditional space for tick
  const isSelectedFromOptions = React.useMemo(() => {
    return Boolean(selectedOption && timeOptions.some(option => option.time === selectedOption.time));
  }, [selectedOption, timeOptions]);

  // Ensure the active descendant is visible in long lists with overflow.
  // Without this, the active item may not be visible on the screen.
  const scrollIntoView = useScrollIntoView.useScrollIntoView(isHighlighted);
  const handleOnClick = React.useCallback(event => {
    if (typeof onClick === 'function') {
      onClick(event);
    }
    onSelectionChange(option);
    closeMenu();
  }, [closeMenu, onClick, onSelectionChange, option]);
  const railStart = isSelectedFromOptions ? /*#__PURE__*/jsxRuntime.jsx(align_child_to_text.AlignChildToText, {
    children: isSelected ? /*#__PURE__*/jsxRuntime.jsx(checkIcon.CheckIcon, {
      size: "16"
    }) : /*#__PURE__*/jsxRuntime.jsx(spacer.Spacer, {
      className: "w-4"
    })
  }) : null;
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getComboboxItemStyles(isSelectedFromOptions)
  });
  const menuItemTrackStyleProps = {
    className: styleProps('root').className,
    classNames: {
      center: styleProps('center').className,
      railStart: styleProps('railStart').className,
      railEnd: styleProps('railEnd').className
    },
    style: styleProps('root').style,
    styles: {
      center: styleProps('center').style,
      railStart: styleProps('railStart').style,
      railEnd: styleProps('railEnd').style
    }
  };
  return /*#__PURE__*/jsxRuntime.jsx(menu_item.MenuItem, {
    ...consumerProps,
    "aria-selected": isSelected,
    id: id,
    isHighlighted: isHighlighted,
    isSelected: isSelected,
    onClick: handleOnClick,
    railStart: railStart,
    ref: utils.mergeRefs(scrollIntoView, forwardedRef),
    role: "option",
    tabIndex: -1,
    ...menuItemTrackStyleProps,
    children: children
  });
});

exports.TimePickerItem = TimePickerItem;
