// @adaptavant/eds-core/dist/toggle-primitive-f323e941.cjs.dev.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var checkbox_primitive = require('./checkbox-primitive-f4324c66.cjs.dev.js');
var index = require('./index-3bdbdb42.cjs.dev.js');
var styles = require('./styles-ea402050.cjs.dev.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-88b49271.cjs.dev.js');
var align_child_to_text = require('./align-child-to-text-b4992541.cjs.dev.js');
require('./tw-merge-70bc959c.cjs.dev.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js');
require('./styles-3e350289.cjs.dev.js');
require('./tw-cabb7e4c.cjs.dev.js');

/**
 * TogglePrimitive
 *
 * @description
 * Component that renders a toggle input. Must be wrapped in a \<label\> element.
 *
 * @example
 * <Track
 *   as="label"
 *   gap="4"
 *   htmlFor={id}
 *   railStart={<Toggle {...props} id={id} />}
 * >
 *   <Text>Label</Text>
 * </Track>
 */
const TogglePrimitive = /*#__PURE__*/React.forwardRef(function TogglePrimitive({
  checked,
  className = '',
  classNames = {},
  id,
  isDisabled = false,
  name,
  onBlur,
  onChange,
  onClick,
  size = 'standard',
  style = {},
  styles: styles$1 = {},
  ...consumerProps
}, forwardedRef) {
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getTogglePrimitiveStyles({
      isDisabled,
      size
    })
  });
  const handleOnChange = checkbox_primitive.useCancellableHandler(onChange, isDisabled);
  const handleOnClick = checkbox_primitive.useCancellableHandler(onClick, isDisabled);
  return /*#__PURE__*/jsxRuntime.jsxs(align_child_to_text.AlignChildToText, {
    ...styleProps('root'),
    children: [/*#__PURE__*/jsxRuntime.jsx(box.Box, {
      ...consumerProps,
      "aria-disabled": isDisabled || undefined,
      as: "input",
      checked: checked,
      id: id,
      name: name,
      onBlur: onBlur,
      onChange: handleOnChange,
      onClick: handleOnClick,
      ref: forwardedRef,
      type: "checkbox",
      ...styleProps('input')
    }), /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      as: "span",
      ...styleProps('control'),
      children: /*#__PURE__*/jsxRuntime.jsx(box.Box, {
        "aria-hidden": true,
        as: "svg",
        fillRule: "evenodd",
        focusable: false,
        preserveAspectRatio: "xMidYMid meet",
        role: "img",
        viewBox: "0 0 480 480",
        xmlns: "http://www.w3.org/2000/svg",
        ...styleProps('icon'),
        children: /*#__PURE__*/jsxRuntime.jsx("circle", {
          cx: "240",
          cy: "240",
          r: "240"
        })
      })
    })]
  });
});

exports.TogglePrimitive = TogglePrimitive;
