// @adaptavant/eds-core/dist/toggle-1ef91408.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var index = require('./index-7955fb8e.cjs.prod.js');
var composeId = require('./compose-id-855ff9b7.cjs.prod.js');
var mergeIds = require('./merge-ids-2d47bd24.cjs.prod.js');
var use_provided_or_generated_id = require('./use-provided-or-generated-id-95445991.cjs.prod.js');
var styles = require('./styles-755e9607.cjs.prod.js');
var toggle_primitive = require('./toggle-primitive-13a235be.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var stack = require('./stack-fa75d919.cjs.prod.js');
var spacer = require('./spacer-388e98f9.cjs.prod.js');
var text = require('./text-a5ab226e.cjs.prod.js');
var context = require('./context-4fb6a635.cjs.prod.js');
var track = require('./track-f65e0bbd.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');
require('./styles-3786e407.cjs.prod.js');
require('./tw-9929267e.cjs.prod.js');
require('./box-e39057bd.cjs.prod.js');

/**
 * Toggle
 *
 * @description
 * Toggle is a UI element that allows the user to select option to
 * switch on or off.
 */

const Toggle = /*#__PURE__*/React.forwardRef(function Toggle({
  'aria-describedby': ariaDescribedByProp,
  checked,
  className = '',
  classNames = {},
  description,
  id: idProp,
  isDisabled: isDisabledProp = false,
  label,
  name,
  onBlur,
  onChange,
  onClick,
  size = 'standard',
  style = {},
  styles: styles$1 = {},
  value,
  ...consumerProps
}, forwardedRef) {
  const {
    isDisabled
  } = context.useControlInputState({
    isDisabled: isDisabledProp,
    isInvalid: false,
    isRequired: false
  });
  const a11yProps = context.useControlA11yProps({
    isDisabled
  });
  const {
    descriptionId,
    inputId
  } = useToggleIds(idProp);
  const ariaDescribedBy = mergeIds.mergeIds(a11yProps === null || a11yProps === void 0 ? void 0 : a11yProps['aria-describedby'], ariaDescribedByProp, description ? descriptionId : undefined);
  const parts = index.useComponentStyleParts({
    className,
    classNames,
    style,
    styles: styles$1,
    tw: styles.getToggleStyles({
      isDisabled,
      size
    })
  }, ['root', 'label', 'description', 'descriptionTrack', 'descriptionSpacer', 'togglePrimitiveRoot', 'togglePrimitiveInput', 'togglePrimitiveControl', 'togglePrimitiveIcon']);
  const togglePrimitiveStyleProps = {
    ...parts.togglePrimitiveRoot,
    classNames: {
      input: parts.togglePrimitiveInput.className,
      control: parts.togglePrimitiveControl.className,
      icon: parts.togglePrimitiveIcon.className
    },
    styles: {
      input: parts.togglePrimitiveInput.style,
      control: parts.togglePrimitiveControl.style,
      icon: parts.togglePrimitiveIcon.style
    }
  };
  return /*#__PURE__*/jsxRuntime.jsxs(stack.Stack, {
    ...parts.root,
    children: [/*#__PURE__*/jsxRuntime.jsx(track.Track, {
      as: "label",
      htmlFor: inputId,
      railStart: /*#__PURE__*/jsxRuntime.jsx(toggle_primitive.TogglePrimitive, {
        ...consumerProps,
        ...a11yProps,
        "aria-describedby": ariaDescribedBy,
        checked: checked,
        id: inputId,
        isDisabled: isDisabled,
        name: name,
        onBlur: onBlur,
        onChange: onChange,
        onClick: onClick,
        ref: forwardedRef,
        size: size,
        value: value,
        ...togglePrimitiveStyleProps
      }),
      verticalAlign: "top",
      ...parts.label,
      children: label
    }), description ? /*#__PURE__*/jsxRuntime.jsx(track.Track, {
      as: "span",
      railStart: /*#__PURE__*/jsxRuntime.jsx(spacer.Spacer, {
        ...parts.descriptionSpacer
      }),
      ...parts.descriptionTrack,
      children: /*#__PURE__*/jsxRuntime.jsx(text.Text, {
        id: descriptionId,
        ...parts.description,
        children: description
      })
    }) : null]
  });
});
Toggle.displayName = 'Toggle';
function useToggleIds(id) {
  const inputId = use_provided_or_generated_id.useProvidedOrGeneratedId(id);
  return {
    inputId,
    descriptionId: composeId.composeId(inputId, 'description')
  };
}

exports.Toggle = Toggle;
