// @adaptavant/eds-core/dist/track-f65e0bbd.cjs.prod.js
'use client';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var index = require('./index-7955fb8e.cjs.prod.js');
var tw = require('./tw-9929267e.cjs.prod.js');
var jsxRuntime = require('react/jsx-runtime');
var box = require('./box-e39057bd.cjs.prod.js');
require('./tw-merge-885aa7a8.cjs.prod.js');
require('tailwind-merge');
require('../tokens/dist/adaptavant-eds-core-tokens.cjs.prod.js');

/**
 * Track
 *
 * @description A layout component which allows you to layout elements with
 * 2 optional fixed "rails" and a fluid center (think railway track).
 */
const Track = /*#__PURE__*/React.forwardRef(function Track({
  as: RootEl = 'div',
  children,
  className = '',
  classNames = {},
  railEnd,
  railStart,
  style = {},
  styles = {},
  verticalAlign = 'middle',
  ...consumerProps
}, forwardedRef) {
  const ChildEl = ['label', 'p', 'span'].includes(RootEl) ? 'span' : 'div';
  const styleProps = index.getStyleProps({
    className,
    classNames,
    style,
    styles,
    tw: {
      center: tw.tw`
				flex-col
				grow
				min-w-0
			`,
      railEnd: tw.tw`
				inline-flex
				shrink-0
			`,
      railStart: tw.tw`
				inline-flex
				shrink-0
			`,
      root: tw.tw`
				${verticalAlignMap[verticalAlign]}
				flex
			`
    }
  });
  return /*#__PURE__*/jsxRuntime.jsxs(box.Box, {
    ...consumerProps,
    as: RootEl,
    ref: forwardedRef,
    ...styleProps('root'),
    children: [railStart ? /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      as: ChildEl,
      ...styleProps('railStart', styles.rail),
      children: railStart
    }) : null, /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      as: ChildEl,
      ...styleProps('center'),
      children: children
    }), railEnd ? /*#__PURE__*/jsxRuntime.jsx(box.Box, {
      as: ChildEl,
      ...styleProps('railEnd', styles.rail),
      children: railEnd
    }) : null]
  });
});
Track.displayName = 'Track';
const verticalAlignMap = {
  bottom: tw.tw`items-end`,
  middle: tw.tw`items-center`,
  top: tw.tw`items-start`
};

exports.Track = Track;
