// @adaptavant/eds-core/tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const global = {
  typography: {
    fontFamily: {
      heading: '',
      body: '',
      fallback: ''
    },
    fontWeight: {
      regular: '',
      strong: '',
      stronger: ''
    },
    heading: {
      '32': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '28': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '20': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '24': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '18': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '16': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '14': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '12': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      }
    },
    body: {
      '8': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '10': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '12': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '14': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '16': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      },
      '20': {
        fontSize: '',
        fontWeight: '',
        lineHeight: '',
        letterSpacing: ''
      }
    }
  },
  radius: {
    none: '',
    '4px': '',
    '6px': '',
    '8px': '',
    '12px': '',
    '16px': '',
    full: ''
  },
  breakpoints: {
    sm: '',
    md: '',
    lg: ''
  },
  borderWidth: {
    regular: '',
    large: ''
  },
  spacing: {
    '0': '',
    '2': '',
    '4': '',
    '6': '',
    '8': '',
    '12': '',
    '14': '',
    '16': '',
    '20': '',
    '24': '',
    '32': '',
    '40': '',
    '48': '',
    '56': '',
    '64': '',
    '72': '',
    '80': '',
    '88': '',
    '96': ''
  },
  sizing: {
    '0': '',
    '4': '',
    '6': '',
    '8': '',
    '12': '',
    '14': '',
    '16': '',
    '20': '',
    '24': '',
    '32': '',
    '40': '',
    '48': '',
    '56': '',
    '64': '',
    '72': '',
    '80': '',
    '88': '',
    '96': '',
    '1/2': '',
    '1/3': '',
    '2/3': '',
    '1/4': '',
    '2/4': '',
    '3/4': '',
    '1/5': '',
    '2/5': '',
    '3/5': '',
    '4/5': '',
    '1/6': '',
    '2/6': '',
    '3/6': '',
    '4/6': '',
    '5/6': '',
    full: ''
  },
  shadow: {
    '10': '',
    '20': '',
    '30': '',
    '40': '',
    '50': ''
  },
  animation: {
    duration: {
      '100': '',
      '200': '',
      '300': '',
      '400': '',
      '500': ''
    },
    easing: {
      regular: '',
      entrance: ''
    }
  }
};
const colorScheme = {
  background: {
    accent: '',
    'accent-hover': '',
    'accent-pressed': '',
    'accent-secondary': '',
    'accent-secondary-hover': '',
    'accent-secondary-pressed': '',
    'accent-tertiary': '',
    'accent-tertiary-hover': '',
    'accent-tertiary-pressed': '',
    critical: '',
    'critical-hover': '',
    'critical-pressed': '',
    'critical-secondary': '',
    'critical-secondary-hover': '',
    'critical-secondary-pressed': '',
    canvas: '',
    'canvas-secondary': '',
    'canvas-tertiary': '',
    menu: '',
    neutral: '',
    'neutral-hover': '',
    'neutral-pressed': '',
    'neutral-active': '',
    'neutral-disabled': '',
    'neutral-secondary': '',
    'neutral-secondary-hover': '',
    'neutral-secondary-pressed': '',
    'neutral-tertiary': '',
    'neutral-tertiary-hover': '',
    inverse: '',
    'inverse-hover': '',
    'inverse-pressed': '',
    'inverse-secondary': '',
    'inverse-tertiary': '',
    'inverse-tertiary-hover': '',
    'inverse-tertiary-pressed': '',
    positive: '',
    'positive-hover': '',
    'positive-pressed': '',
    'positive-secondary': '',
    'positive-secondary-hover': '',
    'positive-secondary-pressed': '',
    caution: '',
    'caution-hover': '',
    'caution-pressed': '',
    'caution-secondary': '',
    avatar: '',
    'avatar-hover': '',
    calendar: '',
    'calendar-disabled': '',
    'modal-overlay': ''
  },
  decoration: {
    brand: '',
    facebook: '',
    instagram: '',
    teleport: '',
    twitter: '',
    whatsapp: '',
    youtube: ''
  },
  text: {
    accent: '',
    'accent-hover': '',
    'accent-pressed': '',
    primary: '',
    secondary: '',
    tertiary: '',
    'secondary-hover': '',
    disabled: '',
    'disabled-secondary': '',
    onPrimary: '',
    'onPrimary-disabled': '',
    critical: '',
    'critical-onPrimary': '',
    inverse: '',
    'inverse-secondary': '',
    'inverse-caution': '',
    'inverse-positive': '',
    'inverse-critical': '',
    'inverse-accent': '',
    'critical-secondary': '',
    positive: '',
    'positive-secondary': '',
    caution: '',
    'caution-secondary': '',
    link: '',
    'link-hover': '',
    'link-pressed': ''
  },
  icon: {
    accent: '',
    critical: '',
    'critical-onPrimary': '',
    onPrimary: '',
    primary: '',
    secondary: '',
    tertiary: '',
    positive: '',
    'positive-hover': '',
    caution: '',
    inverse: ''
  },
  border: {
    accent: '',
    primary: '',
    focus: '',
    secondary: '',
    tertiary: '',
    hover: '',
    input: '',
    'input-hover': '',
    'input-active': '',
    'input-active-hover': '',
    'input-active-pressed': '',
    'input-disabled': '',
    'input-critical': '',
    'input-critical-hover': '',
    'input-critical-pressed': '',
    calendar: ''
  },
  status: {
    available: '',
    away: '',
    busy: '',
    offline: '',
    icon: '',
    border: ''
  },
  general: {
    'constant-white': '',
    'constant-black': ''
  },
  chart: {
    'accent-100': '',
    'accent-200': '',
    'accent-300': '',
    'accent-400': '',
    'accent-500': '',
    'accent-600': '',
    'accent-700': '',
    'accent-800': '',
    'neutral-100': '',
    'neutral-200': '',
    'neutral-300': '',
    'neutral-400': '',
    'neutral-500': '',
    'neutral-600': '',
    'neutral-700': '',
    'neutral-800': ''
  },
  palette: {
    'gray-background': '',
    'gray-background-hover': '',
    'gray-background-active': '',
    'gray-border': '',
    'gray-text': '',
    'gray-text-active': '',
    'gray-text-secondary': '',
    'dark-gray-background': '',
    'dark-gray-background-hover': '',
    'dark-gray-background-active': '',
    'dark-gray-border': '',
    'dark-gray-text': '',
    'dark-gray-text-active': '',
    'dark-gray-text-secondary': '',
    'light-pink-background': '',
    'light-pink-background-hover': '',
    'light-pink-background-active': '',
    'light-pink-border': '',
    'light-pink-text': '',
    'light-pink-text-active': '',
    'light-pink-text-secondary': '',
    'pink-background': '',
    'pink-background-hover': '',
    'pink-background-active': '',
    'pink-border': '',
    'pink-text': '',
    'pink-text-active': '',
    'pink-text-secondary': '',
    'red-background': '',
    'red-background-hover': '',
    'red-background-active': '',
    'red-border': '',
    'red-text': '',
    'red-text-active': '',
    'red-text-secondary': '',
    'maroon-background': '',
    'maroon-background-hover': '',
    'maroon-background-active': '',
    'maroon-border': '',
    'maroon-text': '',
    'maroon-text-active': '',
    'maroon-text-secondary': '',
    'orange-background': '',
    'orange-background-hover': '',
    'orange-background-active': '',
    'orange-border': '',
    'orange-text': '',
    'orange-text-active': '',
    'orange-text-secondary': '',
    'violet-background': '',
    'violet-background-hover': '',
    'violet-background-active': '',
    'violet-border': '',
    'violet-text': '',
    'violet-text-active': '',
    'violet-text-secondary': '',
    'yellow-background': '',
    'yellow-background-hover': '',
    'yellow-background-active': '',
    'yellow-border': '',
    'yellow-text': '',
    'yellow-text-active': '',
    'yellow-text-secondary': '',
    'green-background': '',
    'green-background-hover': '',
    'green-background-active': '',
    'green-border': '',
    'green-text': '',
    'green-text-active': '',
    'green-text-secondary': '',
    'dark-green-background': '',
    'dark-green-background-hover': '',
    'dark-green-background-active': '',
    'dark-green-border': '',
    'dark-green-text': '',
    'dark-green-text-active': '',
    'dark-green-text-secondary': '',
    'blue-background': '',
    'blue-background-hover': '',
    'blue-background-active': '',
    'blue-border': '',
    'blue-text': '',
    'blue-text-active': '',
    'blue-text-secondary': '',
    'dark-blue-background': '',
    'dark-blue-background-hover': '',
    'dark-blue-background-active': '',
    'dark-blue-border': '',
    'dark-blue-text': '',
    'dark-blue-text-active': '',
    'dark-blue-text-secondary': ''
  }
};
const tokenSchema = {
  global,
  colorSchemes: {
    light: colorScheme,
    dark: colorScheme
  }
};

exports.colorScheme = colorScheme;
exports.global = global;
exports.tokenSchema = tokenSchema;
