// @adaptavant/eds-core/utils/dist/adaptavant-eds-core-utils.cjs.dev.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var mergeProps = require('../../dist/merge-props-74bd38ff.cjs.dev.js');
var composeId = require('../../dist/compose-id-8f0cde18.cjs.dev.js');
var copy_to_clipboard = require('../../dist/copy-to-clipboard-90929ffd.cjs.dev.js');
var forwardRef = require('../../dist/forward-ref-6c731e6e.cjs.dev.js');
var mergeIds = require('../../dist/merge-ids-b19ec20e.cjs.dev.js');
var use_animate_visibility = require('../../dist/use-animate-visibility-6b555a5b.cjs.dev.js');
var use_click_away = require('../../dist/use-click-away-d103f9ae.cjs.dev.js');
var use_combined_refs = require('../../dist/use-combined-refs-f8d2ab7c.cjs.dev.js');
var use_debounce = require('../../dist/use-debounce-6838ef75.cjs.dev.js');
var use_debounce_function = require('../../dist/use-debounce-function-faa579f2.cjs.dev.js');
var use_deep_compare_memoize = require('../../dist/use-deep-compare-memoize-de2987aa.cjs.dev.js');
var use_isomorphic_layout_effect = require('../../dist/use-isomorphic-layout-effect-0b9afd7b.cjs.dev.js');
var use_provided_or_generated_id = require('../../dist/use-provided-or-generated-id-2bfa8150.cjs.dev.js');
var use_state_manager = require('../../dist/use-state-manager-4d162c0b.cjs.dev.js');
var use_state_ref = require('../../dist/use-state-ref-f7a866e1.cjs.dev.js');
var use_timeout = require('../../dist/use-timeout-8e83e1ab.cjs.dev.js');
var use_viewport_below_or_equal = require('../../dist/use-viewport-below-or-equal-a63c017b.cjs.dev.js');
require('@react-aria/utils');
require('../../dist/cx-26784cbb.cjs.dev.js');
require('clsx');
require('react');

const DOES_NOT_CONTAIN_HTML_ELEMENTS = /^((?!&lt;|&gt;|<|>).)*$/s;

exports.chainFunctions = mergeProps.chainFunctions;
exports.mergeProps = mergeProps.mergeProps;
exports.composeId = composeId.composeId;
Object.defineProperty(exports, 'useCopyToClipboard', {
	enumerable: true,
	get: function () { return copy_to_clipboard.useCopyToClipboard; }
});
exports.forwardRef = forwardRef.forwardRef;
exports.forwardRefWithGenericComponent = forwardRef.forwardRefWithGenericComponent;
exports.mergeIds = mergeIds.mergeIds;
Object.defineProperty(exports, 'useAnimateVisibility', {
	enumerable: true,
	get: function () { return use_animate_visibility.useAnimateVisibility; }
});
Object.defineProperty(exports, 'useClickAway', {
	enumerable: true,
	get: function () { return use_click_away.useClickAway; }
});
Object.defineProperty(exports, 'useCombinedRefs', {
	enumerable: true,
	get: function () { return use_combined_refs.useCombinedRefs; }
});
Object.defineProperty(exports, 'useDebounce', {
	enumerable: true,
	get: function () { return use_debounce.useDebounce; }
});
Object.defineProperty(exports, 'useDebounceFunction', {
	enumerable: true,
	get: function () { return use_debounce_function.useDebounceFunction; }
});
Object.defineProperty(exports, 'useDeepCompareCallback', {
	enumerable: true,
	get: function () { return use_deep_compare_memoize.useDeepCompareCallback; }
});
Object.defineProperty(exports, 'useDeepCompareEffect', {
	enumerable: true,
	get: function () { return use_deep_compare_memoize.useDeepCompareEffect; }
});
Object.defineProperty(exports, 'useDeepCompareMemo', {
	enumerable: true,
	get: function () { return use_deep_compare_memoize.useDeepCompareMemo; }
});
Object.defineProperty(exports, 'useDeepCompareMemoize', {
	enumerable: true,
	get: function () { return use_deep_compare_memoize.useDeepCompareMemoize; }
});
Object.defineProperty(exports, 'useIsomorphicLayoutEffect', {
	enumerable: true,
	get: function () { return use_isomorphic_layout_effect.useIsomorphicLayoutEffect; }
});
Object.defineProperty(exports, 'useProvidedOrGeneratedId', {
	enumerable: true,
	get: function () { return use_provided_or_generated_id.useProvidedOrGeneratedId; }
});
Object.defineProperty(exports, 'useStateManager', {
	enumerable: true,
	get: function () { return use_state_manager.useStateManager; }
});
Object.defineProperty(exports, 'useStateRef', {
	enumerable: true,
	get: function () { return use_state_ref.useStateRef; }
});
Object.defineProperty(exports, 'useTimeout', {
	enumerable: true,
	get: function () { return use_timeout.useTimeout; }
});
Object.defineProperty(exports, 'useIsViewportBelowOrEqual', {
	enumerable: true,
	get: function () { return use_viewport_below_or_equal.useIsViewportBelowOrEqual; }
});
exports.DOES_NOT_CONTAIN_HTML_ELEMENTS = DOES_NOT_CONTAIN_HTML_ELEMENTS;
