// @adaptavant/eds-core/utils/dist/adaptavant-eds-core-utils.cjs.prod.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var mergeProps = require('../../dist/merge-props-60701ce8.cjs.prod.js');
var composeId = require('../../dist/compose-id-855ff9b7.cjs.prod.js');
var copy_to_clipboard = require('../../dist/copy-to-clipboard-8f3fdff8.cjs.prod.js');
var forwardRef = require('../../dist/forward-ref-0e64c63e.cjs.prod.js');
var mergeIds = require('../../dist/merge-ids-2d47bd24.cjs.prod.js');
var use_animate_visibility = require('../../dist/use-animate-visibility-53a700c4.cjs.prod.js');
var use_click_away = require('../../dist/use-click-away-3152695d.cjs.prod.js');
var use_combined_refs = require('../../dist/use-combined-refs-a90d5bdf.cjs.prod.js');
var use_debounce = require('../../dist/use-debounce-811a61f0.cjs.prod.js');
var use_debounce_function = require('../../dist/use-debounce-function-0070fbe2.cjs.prod.js');
var use_deep_compare_memoize = require('../../dist/use-deep-compare-memoize-9916be91.cjs.prod.js');
var use_isomorphic_layout_effect = require('../../dist/use-isomorphic-layout-effect-94fa0923.cjs.prod.js');
var use_provided_or_generated_id = require('../../dist/use-provided-or-generated-id-95445991.cjs.prod.js');
var use_state_manager = require('../../dist/use-state-manager-2b411976.cjs.prod.js');
var use_state_ref = require('../../dist/use-state-ref-fe2b75f7.cjs.prod.js');
var use_timeout = require('../../dist/use-timeout-5313e527.cjs.prod.js');
var use_viewport_below_or_equal = require('../../dist/use-viewport-below-or-equal-8326f73f.cjs.prod.js');
require('@react-aria/utils');
require('../../dist/cx-86353291.cjs.prod.js');
require('clsx');
require('react');

const DOES_NOT_CONTAIN_HTML_ELEMENTS = /^((?!&lt;|&gt;|<|>).)*$/s;

exports.chainFunctions = mergeProps.chainFunctions;
exports.mergeProps = mergeProps.mergeProps;
exports.composeId = composeId.composeId;
Object.defineProperty(exports, 'useCopyToClipboard', {
	enumerable: true,
	get: function () { return copy_to_clipboard.useCopyToClipboard; }
});
exports.forwardRef = forwardRef.forwardRef;
exports.forwardRefWithGenericComponent = forwardRef.forwardRefWithGenericComponent;
exports.mergeIds = mergeIds.mergeIds;
Object.defineProperty(exports, 'useAnimateVisibility', {
	enumerable: true,
	get: function () { return use_animate_visibility.useAnimateVisibility; }
});
Object.defineProperty(exports, 'useClickAway', {
	enumerable: true,
	get: function () { return use_click_away.useClickAway; }
});
Object.defineProperty(exports, 'useCombinedRefs', {
	enumerable: true,
	get: function () { return use_combined_refs.useCombinedRefs; }
});
Object.defineProperty(exports, 'useDebounce', {
	enumerable: true,
	get: function () { return use_debounce.useDebounce; }
});
Object.defineProperty(exports, 'useDebounceFunction', {
	enumerable: true,
	get: function () { return use_debounce_function.useDebounceFunction; }
});
Object.defineProperty(exports, 'useDeepCompareCallback', {
	enumerable: true,
	get: function () { return use_deep_compare_memoize.useDeepCompareCallback; }
});
Object.defineProperty(exports, 'useDeepCompareEffect', {
	enumerable: true,
	get: function () { return use_deep_compare_memoize.useDeepCompareEffect; }
});
Object.defineProperty(exports, 'useDeepCompareMemo', {
	enumerable: true,
	get: function () { return use_deep_compare_memoize.useDeepCompareMemo; }
});
Object.defineProperty(exports, 'useDeepCompareMemoize', {
	enumerable: true,
	get: function () { return use_deep_compare_memoize.useDeepCompareMemoize; }
});
Object.defineProperty(exports, 'useIsomorphicLayoutEffect', {
	enumerable: true,
	get: function () { return use_isomorphic_layout_effect.useIsomorphicLayoutEffect; }
});
Object.defineProperty(exports, 'useProvidedOrGeneratedId', {
	enumerable: true,
	get: function () { return use_provided_or_generated_id.useProvidedOrGeneratedId; }
});
Object.defineProperty(exports, 'useStateManager', {
	enumerable: true,
	get: function () { return use_state_manager.useStateManager; }
});
Object.defineProperty(exports, 'useStateRef', {
	enumerable: true,
	get: function () { return use_state_ref.useStateRef; }
});
Object.defineProperty(exports, 'useTimeout', {
	enumerable: true,
	get: function () { return use_timeout.useTimeout; }
});
Object.defineProperty(exports, 'useIsViewportBelowOrEqual', {
	enumerable: true,
	get: function () { return use_viewport_below_or_equal.useIsViewportBelowOrEqual; }
});
exports.DOES_NOT_CONTAIN_HTML_ELEMENTS = DOES_NOT_CONTAIN_HTML_ELEMENTS;
