// @adaptavant/eds-core/utils/dist/adaptavant-eds-core-utils.esm.js
export { c as chainFunctions, m as mergeProps } from '../../dist/merge-props-5c13cba9.esm.js';
export { c as composeId } from '../../dist/compose-id-1f2e6139.esm.js';
export { useCopyToClipboard } from '../../dist/copy-to-clipboard-8b65f444.esm.js';
export { f as forwardRef, a as forwardRefWithGenericComponent } from '../../dist/forward-ref-a0e1d5d1.esm.js';
export { m as mergeIds } from '../../dist/merge-ids-c2d2d3d7.esm.js';
export { useAnimateVisibility } from '../../dist/use-animate-visibility-20f1f018.esm.js';
export { useClickAway } from '../../dist/use-click-away-a74e18f7.esm.js';
export { useCombinedRefs } from '../../dist/use-combined-refs-1e0ccd00.esm.js';
export { useDebounce } from '../../dist/use-debounce-8c5292b4.esm.js';
export { useDebounceFunction } from '../../dist/use-debounce-function-f581583e.esm.js';
export { useDeepCompareCallback, useDeepCompareEffect, useDeepCompareMemo, useDeepCompareMemoize } from '../../dist/use-deep-compare-memoize-70b8df88.esm.js';
export { useIsomorphicLayoutEffect } from '../../dist/use-isomorphic-layout-effect-354c38ea.esm.js';
export { useProvidedOrGeneratedId } from '../../dist/use-provided-or-generated-id-70d42050.esm.js';
export { useStateManager } from '../../dist/use-state-manager-d1acda47.esm.js';
export { useStateRef } from '../../dist/use-state-ref-caeea518.esm.js';
export { useTimeout } from '../../dist/use-timeout-9e505916.esm.js';
export { useIsViewportBelowOrEqual } from '../../dist/use-viewport-below-or-equal-b1c12ed0.esm.js';
import '@react-aria/utils';
import '../../dist/cx-c5d68be6.esm.js';
import 'clsx';
import 'react';

const DOES_NOT_CONTAIN_HTML_ELEMENTS = /^((?!&lt;|&gt;|<|>).)*$/s;

export { DOES_NOT_CONTAIN_HTML_ELEMENTS };
