// @adaptavant/eds-core/symbols/dist/adaptavant-eds-core-symbols.cjs.dev.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var zoomColorIcon = require('../../dist/zoom-color-icon-6067ec6d.cjs.dev.js');
require('react');
require('../../dist/index-3bdbdb42.cjs.dev.js');
require('../../dist/tw-merge-70bc959c.cjs.dev.js');
require('tailwind-merge');
require('../../tokens/dist/adaptavant-eds-core-tokens.cjs.dev.js');
require('react/jsx-runtime');
require('../../dist/tw-cabb7e4c.cjs.dev.js');



exports.AmexColorIcon = zoomColorIcon.AmexColorIcon;
exports.AnswerconnectColorIcon = zoomColorIcon.AnswerconnectColorIcon;
exports.AnywhereWorksColorIcon = zoomColorIcon.AnywhereWorksColorIcon;
exports.AppleColorIcon = zoomColorIcon.AppleColorIcon;
exports.ApplePayColorIcon = zoomColorIcon.ApplePayColorIcon;
exports.CashAppPayColorIcon = zoomColorIcon.CashAppPayColorIcon;
exports.ClioColorIcon = zoomColorIcon.ClioColorIcon;
exports.ConstantContactColorIcon = zoomColorIcon.ConstantContactColorIcon;
exports.DinersColorIcon = zoomColorIcon.DinersColorIcon;
exports.DiscoverColorIcon = zoomColorIcon.DiscoverColorIcon;
exports.DrupalColorIcon = zoomColorIcon.DrupalColorIcon;
exports.FacebookColorIcon = zoomColorIcon.FacebookColorIcon;
exports.FacetimeColorIcon = zoomColorIcon.FacetimeColorIcon;
exports.FigmaColorIcon = zoomColorIcon.FigmaColorIcon;
exports.FreshdeskColorIcon = zoomColorIcon.FreshdeskColorIcon;
exports.GithubColorIcon = zoomColorIcon.GithubColorIcon;
exports.GoogleAnalyticsColorIcon = zoomColorIcon.GoogleAnalyticsColorIcon;
exports.GoogleCalendarColorIcon = zoomColorIcon.GoogleCalendarColorIcon;
exports.GoogleColorIcon = zoomColorIcon.GoogleColorIcon;
exports.GoogleMeetingColorIcon = zoomColorIcon.GoogleMeetingColorIcon;
exports.GooglePayColorIcon = zoomColorIcon.GooglePayColorIcon;
exports.GoogleTagManagerColorIcon = zoomColorIcon.GoogleTagManagerColorIcon;
exports.HubspotColorIcon = zoomColorIcon.HubspotColorIcon;
exports.InsightlyColorIcon = zoomColorIcon.InsightlyColorIcon;
exports.InstagramColorIcon = zoomColorIcon.InstagramColorIcon;
exports.IntercomColorIcon = zoomColorIcon.IntercomColorIcon;
exports.JcbColorIcon = zoomColorIcon.JcbColorIcon;
exports.JimdoColorIcon = zoomColorIcon.JimdoColorIcon;
exports.JoomlaColorIcon = zoomColorIcon.JoomlaColorIcon;
exports.KeapColorIcon = zoomColorIcon.KeapColorIcon;
exports.LawpayColorIcon = zoomColorIcon.LawpayColorIcon;
exports.MaestroColorIcon = zoomColorIcon.MaestroColorIcon;
exports.MailchimpColorIcon = zoomColorIcon.MailchimpColorIcon;
exports.MailerliteColorIcon = zoomColorIcon.MailerliteColorIcon;
exports.MastercardColorIcon = zoomColorIcon.MastercardColorIcon;
exports.MicrosoftColorIcon = zoomColorIcon.MicrosoftColorIcon;
exports.Office365ColorIcon = zoomColorIcon.Office365ColorIcon;
exports.OpendentalColorIcon = zoomColorIcon.OpendentalColorIcon;
exports.OutlookColorIcon = zoomColorIcon.OutlookColorIcon;
exports.PaypalColorIcon = zoomColorIcon.PaypalColorIcon;
exports.QuickbooksColorIcon = zoomColorIcon.QuickbooksColorIcon;
exports.SalesforceColorIcon = zoomColorIcon.SalesforceColorIcon;
exports.ServiceminderColorIcon = zoomColorIcon.ServiceminderColorIcon;
exports.SetmoreColorIcon = zoomColorIcon.SetmoreColorIcon;
exports.ShopifyColorIcon = zoomColorIcon.ShopifyColorIcon;
exports.SlackColorIcon = zoomColorIcon.SlackColorIcon;
exports.SquareColorIcon = zoomColorIcon.SquareColorIcon;
exports.SquarespaceColorIcon = zoomColorIcon.SquarespaceColorIcon;
exports.StorybookColorIcon = zoomColorIcon.StorybookColorIcon;
exports.StripeColorIcon = zoomColorIcon.StripeColorIcon;
exports.TeleportColorIcon = zoomColorIcon.TeleportColorIcon;
exports.ThreadsColorIcon = zoomColorIcon.ThreadsColorIcon;
exports.VisaColorIcon = zoomColorIcon.VisaColorIcon;
exports.WeeblyColorIcon = zoomColorIcon.WeeblyColorIcon;
exports.WixColorIcon = zoomColorIcon.WixColorIcon;
exports.WordpressColorIcon = zoomColorIcon.WordpressColorIcon;
exports.XTwitterColorIcon = zoomColorIcon.XTwitterColorIcon;
exports.ZapierColorIcon = zoomColorIcon.ZapierColorIcon;
exports.ZendeskColorIcon = zoomColorIcon.ZendeskColorIcon;
exports.ZohoColorIcon = zoomColorIcon.ZohoColorIcon;
exports.ZoomColorIcon = zoomColorIcon.ZoomColorIcon;
